<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Wenjun Liu Dame Roma Mitchell Cancer Research Laboratories Adelaide Medical School University of Adelaide" />

<meta name="date" content="2023-01-18" />

<title>Differential_Expression</title>

<script src="site_libs/header-attrs-2.23/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.28/datatables.js"></script>
<link href="site_libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="site_libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet" />
<script src="site_libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="site_libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet" />
<script src="site_libs/selectize-0.12.0/selectize.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DHT_fibroblast_snRNAseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Preprocess_fibro_from10x.html">Pre-processing</a>
</li>
<li>
  <a href="Subtype_Prediction.html">Cell Type Annotation</a>
</li>
<li>
  <a href="Differential_Expression.html">Differential Expression &amp; Pathway Analysis</a>
</li>
<li>
  <a href="Data_Integration.html">Data Integration</a>
</li>
<li>
  <a href="RNA_velocity.html">RNA Velocity</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Differential_Expression</h1>
<h4 class="author">Wenjun Liu<br>Dame Roma Mitchell Cancer Research
Laboratories<br>Adelaide Medical School<br>University of Adelaide</h4>
<h4 class="date">2023-01-18</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-07-31
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>DHT_fibroblast_snRNAseq/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed12345code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(12345)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed12345code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(12345)</code> was run prior to running the
code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomWenjunLiuDHTfibroblastsnRNAseqtree766d02da0c943d4e0228cf1a615aedfb0c8e46cdtargetblank766d02da">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/Wenjun-Liu/DHT_fibroblast_snRNAseq/tree/766d02da0c943d4e0228cf1a615aedfb0c8e46cd" target="_blank">766d02d</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomWenjunLiuDHTfibroblastsnRNAseqtree766d02da0c943d4e0228cf1a615aedfb0c8e46cdtargetblank766d02da"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/Wenjun-Liu/DHT_fibroblast_snRNAseq/tree/766d02da0c943d4e0228cf1a615aedfb0c8e46cd" target="_blank">766d02d</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.ipynb_checkpoints/
    Ignored:    output/fibroblast_combined.rds
    Ignored:    output/fibroblast_seurat.rds

Untracked files:
    Untracked:  analysis/.DS_Store
    Untracked:  analysis/Data_integration.Rmd
    Untracked:  analysis/Trajectory_analysis/
    Untracked:  data/Wu_2020/
    Untracked:  data/mf43/
    Untracked:  data/mf61/
    Untracked:  data/sca.rds
    Untracked:  data/sca_ls.rds
    Untracked:  data/stem_sig.txt
    Untracked:  output/DE_global.xlsx
    Untracked:  output/DE_suppTable.xlsx
    Untracked:  output/cluster_marker_unintegrated.csv
    Untracked:  output/cluster_summary_ls.rds
    Untracked:  output/counts.mtx
    Untracked:  output/gene_names.csv
    Untracked:  output/metadata.csv
    Untracked:  output/path_suppTable.xlsx
    Untracked:  output/pca.csv
    Untracked:  output/pred_hpca_combined.rds
    Untracked:  output/pred_wu_combined.rds

Unstaged changes:
    Modified:   .DS_Store
    Deleted:    .Rprofile
    Deleted:    .gitattributes
    Modified:   .gitignore
    Deleted:    README.md
    Deleted:    _workflowr.yml
    Modified:   analysis/Differential_Expression.Rmd
    Modified:   analysis/Preprocess_fibro_from10x.Rmd
    Modified:   analysis/Subtype_Prediction.Rmd
    Modified:   analysis/_site.yml
    Deleted:    analysis/plot_commu_temp.R
    Deleted:    output/Table_Chap5_Supp3.xlsx

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/Differential_Expression.Rmd</code>) and HTML
(<code>docs/Differential_Expression.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Wenjun-Liu/DHT_fibroblast_snRNAseq/blob/766d02da0c943d4e0228cf1a615aedfb0c8e46cd/analysis/Differential_Expression.Rmd" target="_blank">766d02d</a>
</td>
<td>
Wenjun-Liu
</td>
<td>
2023-06-08
</td>
<td>
rebuilt html pages
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Wenjun-Liu/DHT_fibroblast_snRNAseq/766d02da0c943d4e0228cf1a615aedfb0c8e46cd/docs/Differential_Expression.html" target="_blank">766d02d</a>
</td>
<td>
Wenjun-Liu
</td>
<td>
2023-06-08
</td>
<td>
rebuilt html pages
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)
library(yaml)
library(scales)
library(pander)
library(glue)
library(edgeR)
library(AnnotationHub)
library(ensembldb)
library(cowplot)
library(ggfortify)
library(magrittr)
library(cqn)
library(ggrepel)
library(DT)
library(Seurat)
library(corrplot)
library(plotly)
library(patchwork)
library(colorspace)
library(ggpubr)
library(randomcoloR)
library(ggforce)
library(MAST)
library(pheatmap)
library(BiocParallel)
library(UpSetR)
library(msigdbr)
library(goseq)
library(sSNAPPY)
library(AUCell)
library(grid)</code></pre>
<pre class="r"><code>panderOptions(&quot;table.split.table&quot;, Inf)
panderOptions(&quot;big.mark&quot;, &quot;,&quot;)
theme_set(theme_bw())</code></pre>
<pre class="r"><code>formatP &lt;- function(p, m = 0.0001){
out &lt;- rep(&quot;&quot;, length(p))
out[p &lt; m] &lt;- sprintf(&quot;%.2e&quot;, p[p&lt;m])
out[p &gt;= m] &lt;- sprintf(&quot;%.4f&quot;, p[p&gt;=m])
out
}</code></pre>
<div id="gene-annotations" class="section level2">
<h2>Gene Annotations</h2>
<p>Firstly, gene annotations were retrieved from Ensembl Release
101.</p>
<pre class="r"><code>ah &lt;- AnnotationHub() %&gt;%
  subset(rdataclass == &quot;EnsDb&quot;) %&gt;%
  subset(str_detect(description, &quot;101&quot;)) %&gt;%
  subset(genome == &quot;GRCh38&quot;)
stopifnot(length(ah) == 1)</code></pre>
<pre class="r"><code>ensDb &lt;- ah[[1]]
genesGR &lt;- genes(ensDb)
transGR &lt;- transcripts(ensDb)</code></pre>
<pre class="r"><code>mcols(transGR) &lt;- mcols(transGR) %&gt;%
  cbind(
    transcriptLengths(ensDb)[rownames(.), c(&quot;nexon&quot;, &quot;tx_len&quot;)]
  )</code></pre>
<pre class="r"><code>mcols(genesGR) &lt;- mcols(genesGR) %&gt;%
  as.data.frame() %&gt;%
  dplyr::select(
    gene_id, gene_name, gene_biotype, entrezid
  ) %&gt;%
  left_join(
    mcols(transGR) %&gt;%
      as.data.frame() %&gt;%
      mutate(
        tx_support_level = case_when(
          is.na(tx_support_level) ~ 1L, 
          TRUE ~ tx_support_level
        )
      ) %&gt;%
      group_by(gene_id) %&gt;%
      summarise(
        n_tx = n(),
        longest_tx = max(tx_len),
        ave_tx_len = mean(tx_len),
        gc_content = sum(tx_len*gc_content) / sum(tx_len)
      ) %&gt;%
      mutate(
        bin_length = cut(
          x = ave_tx_len,
          labels = seq_len(10),
          breaks = quantile(ave_tx_len, probs = seq(0, 1, length.out = 11)),
          include.lowest = TRUE
        ),
        bin_gc = cut(
          x = gc_content,
          labels = seq_len(10),
          breaks = quantile(gc_content, probs = seq(0, 1, length.out = 11)),
          include.lowest = TRUE
        ),
        bin = paste(bin_gc, bin_length, sep = &quot;_&quot;)
      ),
    by = &quot;gene_id&quot;
  ) %&gt;%
  set_rownames(.$gene_id) %&gt;%
  as(&quot;DataFrame&quot;)</code></pre>
</div>
<div id="read-in-seurat-object" class="section level1">
<h1>Read in seurat object</h1>
<p>The pre-processed <code>seurat</code> object saved in the
pre-processing step was loaded in.</p>
<pre class="r"><code>fibroblast &lt;- readRDS(here::here(&quot;output/fibroblast_seurat.rds&quot;))
# fibroblast[[]] %&gt;%
#     rownames_to_column(&quot;cell&quot;) %&gt;%
#     mutate(
#         label = vapply(cell, function(x){
#             str_split(x, &quot;-&quot;)[[1]][2]
#         }, character(1))
#     ) %&gt;%
#     split(f = .$orig.ident) %&gt;%
#     lapply(pull, label) %&gt;%
#     lapply(unique)</code></pre>
<p>Color palettes for the patient and treatment levels are set up.</p>
<pre class="r"><code># Color palettes used by most scRNA-seq methods could be found [here](https://rdrr.io/bioc/scater/src/R/plot_colours.R). 
patient_col &lt;- c(
    &quot;MF61&quot; = &quot;#A71B4B&quot;, 
    &quot;MF43&quot; = &quot;#584B9F&quot;
)
# scanpy colors were found in https://github.com/scverse/scanpy/blob/034ca2823804645e0d4874c9b16ba2eb8c13ac0f/scanpy/plotting/palettes.py
treat_col &lt;- c(
  VEH = rgb(0.7, 0.7, 0.7),
  DHT = rgb(0.8, 0.2, 0.2))
sample_col &lt;- hcl.colors(n = 4) %&gt;%
    set_names(c(&quot;mf43DHT&quot;, &quot;mf43VEH&quot;, &quot;mf61DHT&quot;, &quot;mf61VEH&quot;))</code></pre>
<p>There are two different strategies that can be used to perform
differential expression analysis in scRNA-seq data: 1. Global DE:
performing DE on all DHT-treated cells against all vehicle treated
cells. 2. Cluster-specific DE: performing DE on DHT-treated cells
against vehicle treated cells within each cluster.</p>
<p>Both strategies will be applied to this dataset in this analysis.</p>
</div>
<div id="de-using-mast" class="section level1">
<h1>DE using MAST</h1>
<div id="global-de" class="section level2">
<h2>Global DE</h2>
<p>Firstly, differential expression analysis was perform using
<code>MAST</code> to compare all DHT-treated samples against the all
vehicle while nested within patients.</p>
<p><code>sca</code> objects were created using
<code>MAST::FromMatrix()</code> function and the actual linear mixed
hurdle model fittings were done on the DRMCRL workstation.</p>
<p>Note that the continuous part of the hurdle models is conditional of
a gene being expressed, so if a gene had non-zero count in only one Cell
Type + Type group, the models won’t work. The <code>FindMarkers()</code>
function in <code>Seruat</code> solves this by including a
<code>min.pct</code> parameter so only genes that were detected in a
minimum fraction of <code>min.pct</code> cells in either of the two
populations would be tested. <a
href="https://github.com/RGLab/MAST/issues/146">The developer of
<code>MAST</code> package also recommended this strategy</a> so only
genes that had non-zero log-norm counts in at least 10% of vehicle or
DHT-treated cells were tested.</p>
<pre class="r"><code># create SCA object
seurat_data &lt;- GetAssayData(object = fibroblast, slot = &#39;data&#39;)
VEH_cell &lt;- fibroblast[[]] %&gt;%
        dplyr::filter(Treatment == &quot;VEH&quot;) %&gt;%
        rownames()
gene2keep_VEH &lt;- rowSums(seurat_data[, colnames(seurat_data) %in% VEH_cell] &gt; 0) &gt; (0.1 * length(VEH_cell))
DHT_cell &lt;- fibroblast[[]] %&gt;%
        dplyr::filter(Treatment == &quot;DHT&quot;) %&gt;%
        rownames()
gene2keep_DHT &lt;- rowSums(seurat_data[, colnames(seurat_data) %in% DHT_cell] &gt; 0) &gt; (0.1 * length(DHT_cell))
seurat_data &lt;- seurat_data[gene2keep_DHT|gene2keep_VEH,]
# MF43_cell &lt;- fibroblast[[]] %&gt;%
#         dplyr::filter(patient == &quot;MF43&quot;) %&gt;%
#         rownames()
# gene2keep_MF43 &lt;- rowSums(seurat_data[, colnames(seurat_data) %in% MF43_cell] &gt; 0) &gt; (0.1 * length(MF43_cell))
# MF61_cell &lt;- fibroblast[[]] %&gt;%
#         dplyr::filter(Treatment == &quot;MF61&quot;) %&gt;%
#         rownames()
# gene2keep_MF61 &lt;- rowSums(seurat_data[, colnames(seurat_data) %in% MF61_cell] &gt; 0) &gt; (0.1 * length(MF61_cell))
# seurat_data &lt;- seurat_data[gene2keep_MF61|gene2keep_MF43,]
sca &lt;- MAST::FromMatrix(
    exprsArray = as.matrix(seurat_data), 
    cData = fibroblast[[]], 
    fData = data.frame(primerid = rownames(seurat_data)) %&gt;%
            set_rownames(rownames(seurat_data))
)</code></pre>
<p>Cell detection rate was calculated for each gene and added to the
<code>sca</code> object..</p>
<pre class="r"><code>cdr &lt;- colSums(SummarizedExperiment::assay(sca)&gt;0)
SummarizedExperiment::colData(sca)$ngeneson &lt;- scale(cdr)</code></pre>
<p>The reference level of treatment was set to VEH.</p>
<pre class="r"><code>SummarizedExperiment::colData(sca)$Treatment &lt;- factor(SummarizedExperiment::colData(sca)$Treatment, levels = c(&quot;VEH&quot;, &quot;DHT&quot;))
SummarizedExperiment::colData(sca)$patient &lt;- factor(SummarizedExperiment::colData(sca)$patient)
SummarizedExperiment::colData(sca)$patient &lt;- droplevels(SummarizedExperiment::colData(sca)$patient)</code></pre>
<pre class="r"><code># saveRDS(sca, here::here(&quot;data/sca.rds&quot;))</code></pre>
<p>Hurdle model was fitted with the patient as a random effect to nest
the comparisons in patient using linear mixed modelling. (This step was
performed on the DRMCRL workstation).</p>
<pre class="r"><code>zlm_summary_mixed &lt;- readRDS(here::here(&quot;output/zlm_summary_mixed.rds&quot;))$datatable
zlm_summary_mixed &lt;- zlm_summary_mixed %&gt;%
        dplyr::filter(contrast==&#39;TreatmentDHT&#39; &amp; component==&#39;H&#39;) %&gt;%
        dplyr::select(primerid, `Pr(&gt;Chisq)`) %&gt;%
        left_join(
            # logFC
            zlm_summary_mixed %&gt;%
                dplyr::filter(contrast==&#39;TreatmentDHT&#39; &amp; component==&#39;logFC&#39;) %&gt;%
                dplyr::select(primerid, logFC = coef, ci.hi, ci.lo) 
        )  %&gt;%
        mutate(FDR = p.adjust(`Pr(&gt;Chisq)`, &quot;fdr&quot;),
               DE = ifelse(FDR &lt; 0.01, TRUE, FALSE)
               ) %&gt;%
        dplyr::rename(gene_name = primerid)</code></pre>
<p>To my surprise, with the linear mixed model, even with a stringent
cut-off of FDR &lt; 0.01, There were still 4869 genes passing the
significance threshold and being considered differentially
expressed.</p>
<div id="knott-mast-results" class="section level3">
<h3>Knott MAST results</h3>
<p>In the study by <a
href="https://www.cell.com/cell-genomics/pdf/S2666-979X(23)00032-0.pdf">Rath
et al. (2023)</a>, breast tissues from 9 trans-male donors who have been
undertaking gender-affirming therapy (masculinizing dose of androgen)
were compared against 9 cis-females’ breast through snRNA-seq.</p>
<p>The pre-processed <code>Seurat</code> object for the snRNA-seq data
in that study was shared with us by the authors. In the original study,
differential expression analyses were performed for each detected cell
type without accounting for the confounding effect of patients.
Therefore, I re-analysed the data using MAST with donor as a random
effect as described above. Only genes that had non-zero log-norm counts
in at least 10% of CF or TF cells were tested. Results derived from
fibroblast cells were loaded.</p>
<pre class="r"><code>knott_mast_de &lt;- readRDS(
    &quot;~/20131906_HickeyT_JC_NormalBreast/output/knott_fibro_mast.rds&quot;
) %&gt;%
    mutate(knott = -sign(coef) * log10(`Pr(&gt;Chisq)`))</code></pre>
<p>783 out of the 4817 tested genes were considered as significantly
differentially expressed with a FDR cut-off of 0.05.</p>
<p>Ranking statistics for genes in both datasets were defined as <span
class="math display">\[-sign(logFC) \times \log_{10} (p{\text
-}value)\]</span>.</p>
<pre class="r"><code>knott_mast_de &lt;- readRDS(
    &quot;~/20131906_HickeyT_JC_NormalBreast/output/knott_fibro_mast.rds&quot;
) %&gt;%
    mutate(knott = -sign(coef) * log10(`Pr(&gt;Chisq)`))</code></pre>
<pre class="r"><code>fibro_2RS &lt;- zlm_summary_mixed %&gt;%
    mutate(pilot = -sign(logFC) * log10(`Pr(&gt;Chisq)`)) %&gt;%
    dplyr::select(gene_name, pilot) %&gt;%
    left_join(
        knott_mast_de%&gt;%
            dplyr::select(gene_name , knott)
    ) %&gt;%
    drop_na()
fibro_2RS_cor &lt;- cor.test(fibro_2RS$pilot, fibro_2RS$knott)</code></pre>
<p>Correlation between the ranking statistics derived from the two
independent studies was found to be 0.024 with a p-value of 0.168.</p>
<pre class="r"><code>pilotRK &lt;- zlm_summary_mixed %&gt;%
    mutate(pilot = -sign(logFC) * log10(`Pr(&gt;Chisq)`))  %&gt;%
    dplyr::filter(DE) %&gt;%
    pull(pilot)
knottRK &lt;- knott_mast_de  %&gt;%
    dplyr::filter(DE) %&gt;%
    pull(knott)</code></pre>
<pre class="r"><code>cor_plot &lt;- fibro_2RS %&gt;%
    mutate(aveRS = (pilot + knott)/2) %&gt;%
    .[order(abs(.$aveRS), decreasing = TRUE),] %&gt;%
    mutate(RANK = row_number()) %&gt;%
    ggplot(
        aes(pilot , knott)
    ) +
    geom_point() +
  geom_vline(
    xintercept = 0,
    linetype = 2,
    size = 1.5,
    colour = &quot;red&quot;
  ) +
    geom_vline(
    xintercept = c(max(knottRK[knottRK &lt; 0]), min(knottRK[knottRK &gt; 0])),
    linetype = 2,
    size = 1.5,
    colour = &quot;blue&quot;
  ) +
  geom_hline(
    yintercept = 0,
    linetype = 2,
    size = 1.5,
    colour = &quot;red&quot;
  ) +
    geom_hline(
    yintercept = c(max(pilotRK[pilotRK &lt; 0]), min(pilotRK[pilotRK &gt; 0])),
    linetype = 2,
    size = 1.5,
    colour = &quot;blue&quot;
  ) +
    geom_text_repel(
        aes(label = gene_name), 
        data = . %&gt;% 
            dplyr::filter(
                RANK &lt; 50
            ), 
        color = &quot;red&quot;, 
        size = 4
    ) +
    labs(
        x = &quot;Ranking Statistic (DHT vs Veh)&quot;, 
        y = &quot;Ranking Statistic (trans-male vs cis-female)&quot;, 
        # title = paste(&quot;Correlation between logFCs between the 2 study:&quot;, 
        #                 round(cor(fibro_2R$pilot, fibro_2FC$knott), digits = 2), 
        #                 sep = &quot; &quot;)
    ) 
cor_plot</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/Differential_Expression.Rmd/unnamed-chunk-11-1.png" alt="*Correlation between genes’ ranking statistics when (y-axis) comparing trans-male breast fibroblasts against cis-females’ vs (x-axis) comparing DHT-treated fibroblasts against vehicle-treated ones. The red dashed lines highlight the ranking statistic of zero. The signs of ranking statistics reflect genes’ directions of change. Blue dashed lines indicate the significance threshold for each study*" width="1344" />
<p class="caption">
<em>Correlation between genes’ ranking statistics when (y-axis)
comparing trans-male breast fibroblasts against cis-females’ vs (x-axis)
comparing DHT-treated fibroblasts against vehicle-treated ones. The red
dashed lines highlight the ranking statistic of zero. The signs of
ranking statistics reflect genes’ directions of change. Blue dashed
lines indicate the significance threshold for each study</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Wenjun-Liu/DHT_fibroblast_snRNAseq/blob/766d02da0c943d4e0228cf1a615aedfb0c8e46cd/docs/figure/Differential_Expression.Rmd/unnamed-chunk-11-1.png" target="_blank">766d02d</a>
</td>
<td>
Wenjun-Liu
</td>
<td>
2023-06-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="ar-signature" class="section level3">
<h3>AR signature</h3>
<pre class="r"><code>AR_sig &lt;- read.csv(here::here(&quot;data/AR_sig_down.csv&quot;)) %&gt;%
    mutate(AR_sig = &quot;Down&quot;) %&gt;%
    rbind(
        read.csv(here::here(&quot;data/AR_sig_up.csv&quot;)) %&gt;%
            mutate(AR_sig = &quot;Up&quot;)
    )
AR_sig_col &lt;- AR_sig %&gt;%
    mutate(
        color = ifelse(AR_sig == &quot;Down&quot;,&#39;blue&#39;, &quot;red&quot;)
    )</code></pre>
<p>A list of 142 AR signatures that were identified in ER-positive
breast cancer cell lines and xenograft models were loaded and the
following ones were considered differentially expressed in DHT-treated
fibroblasts:</p>
<pre class="r"><code>zlm_summary_mixed %&gt;%
    dplyr::filter(
        FDR &lt; 0.01, 
        gene_name %in% AR_sig$gene_name
    )  %&gt;%
    left_join(
        AR_sig 
    ) %&gt;%
    mutate(
        Align = case_when(
           logFC &lt; 0 &amp; AR_sig == &quot;Down&quot; ~ &quot;\u2714 Yes&quot;,
           logFC &gt; 0 &amp; AR_sig == &quot;Up&quot; ~ &quot;\u2714 Yes&quot;,
           logFC &lt; 0 &amp; AR_sig == &quot;Up&quot; ~ &quot;\u2716 No&quot;,
           logFC &gt; 0 &amp; AR_sig == &quot;Down&quot; ~ &quot;\u2716 No&quot;
        )
    ) %&gt;%
    mutate_if(is.character, as.factor) %&gt;%
    mutate_at(vars(c(&quot;Pr(&gt;Chisq)&quot;, &quot;FDR&quot;)), formatP) %&gt;%
    mutate(logFC = round(logFC, 3)) %&gt;%
    dplyr::select(
        `Gene Name` = gene_name,
        `Signature type` = AR_sig,
        logFC,
        PValue = `Pr(&gt;Chisq)`,
        FDR, Align
    ) %&gt;%
    datatable(
        filter = &quot;top&quot;, 
        caption = htmltools::tags$caption(
            htmltools::em(
                &quot;AR signatures that were differentially expressed in DHT-treated fibroblast cells. The align column shows whether directions of changes in this study align with AR signature type&quot;)
        ), 
        options = list(scrollY = &quot;450px&quot;)
    ) %&gt;%
    formatStyle(
        &#39;logFC&#39;,
        color = styleInterval(0, c(&#39;blue&#39;, &quot;red&quot;))
    ) %&gt;%
    formatStyle(
        &#39;Signature type&#39;,
            color = styleEqual(c(&quot;Down&quot;, &quot;Up&quot;), c(&#39;blue&#39;, &quot;red&quot;))
    ) %&gt;%
    formatStyle(
        &#39;Align&#39;,
            color = styleEqual(c( &quot;\u2714 Yes&quot;, &quot;\u2716 No&quot;), c(&quot;#008000&quot;, &quot;#e00000&quot;))
    )</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-b146b89b513bb0257623" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-b146b89b513bb0257623">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td><\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;ABI3BP&quot;,&quot;C1orf21&quot;,&quot;CKAP2L&quot;,&quot;CXCL12&quot;,&quot;ECHDC2&quot;,&quot;ECI2&quot;,&quot;ELOVL5&quot;,&quot;FAR2&quot;,&quot;GALNT5&quot;,&quot;HMGB2&quot;,&quot;IER3&quot;,&quot;IGFBP4&quot;,&quot;KNL1&quot;,&quot;LMNB1&quot;,&quot;MCCC2&quot;,&quot;ME3&quot;,&quot;PHF19&quot;,&quot;SCP2&quot;,&quot;SEMA3C&quot;,&quot;SMAD6&quot;,&quot;STC2&quot;,&quot;SVIL&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Down&quot;,&quot;Up&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"-0.199\" data-max=\"0.143\" data-scale=\"3\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;✔ Yes&quot;,&quot;✖ No&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n<\/tr>","caption":"<caption>\n  <em>AR signatures that were differentially expressed in DHT-treated fibroblast cells. The align column shows whether directions of changes in this study align with AR signature type<\/em>\n<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22"],["ABI3BP","C1orf21","CKAP2L","CXCL12","ECHDC2","ECI2","ELOVL5","FAR2","GALNT5","HMGB2","IER3","IGFBP4","KNL1","LMNB1","MCCC2","ME3","PHF19","SCP2","SEMA3C","SMAD6","STC2","SVIL"],["Down","Up","Down","Down","Up","Up","Up","Up","Down","Down","Down","Down","Down","Down","Up","Down","Down","Up","Down","Down","Down","Down"],[-0.112,-0.026,0.016,0.095,-0.031,0.052,-0,-0.024,-0.03,0.06,-0.16,0.143,0.028,0.034,-0.012,-0.025,0.024,0.034,-0.199,0.08599999999999999,-0.05,-0.064],["5.10e-24","0.0028","1.17e-05","2.68e-31","1.35e-08","3.39e-12","0.0017","7.64e-07","0.0009","4.54e-13","3.45e-41","1.72e-26","5.50e-07","6.81e-11","1.62e-05","7.02e-08","8.64e-15","2.83e-08","9.27e-55","7.42e-25","0.0007","6.26e-14"],["6.09e-23","0.0037","2.38e-05","4.41e-30","4.26e-08","1.66e-11","0.0023","1.90e-06","0.0013","2.42e-12","8.50e-40","2.29e-25","1.40e-06","2.89e-10","3.24e-05","2.02e-07","5.48e-14","8.62e-08","3.32e-53","9.19e-24","0.0010","3.62e-13"],["✔ Yes","✖ No","✖ No","✖ No","✖ No","✔ Yes","✖ No","✖ No","✔ Yes","✖ No","✔ Yes","✖ No","✖ No","✖ No","✖ No","✔ Yes","✖ No","✔ Yes","✔ Yes","✖ No","✔ Yes","✔ Yes"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Gene Name<\/th>\n      <th>Signature type<\/th>\n      <th>logFC<\/th>\n      <th>PValue<\/th>\n      <th>FDR<\/th>\n      <th>Align<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollY":"450px","columnDefs":[{"className":"dt-right","targets":3},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'color':isNaN(parseFloat(value)) ? '' : value <= 0 ? \"blue\" : \"red\"});\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'color':value == \"Down\" ? \"blue\" : value == \"Up\" ? \"red\" : null});\nvar value=data[6]; $(this.api().cell(row, 6).node()).css({'color':value == \"✔ Yes\" ? \"#008000\" : value == \"✖ No\" ? \"#e00000\" : null});\n}"},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
<p>Not all of the AR signatures that were differentially expressed in
DHT-treated cells showed directions of changes that aligned with their
signature types in ER-pos breast cancer.</p>
</div>
</div>
<div id="de-in-each-cluster" class="section level2">
<h2>DE in each cluster</h2>
<p>To perform DE in each cluster, the exact same methods described above
were followed, except that now one DE analysis was performed for each
clsuter.</p>
<pre class="r"><code># create SCA object
seurat_data &lt;- GetAssayData(object = fibroblast, slot = &#39;data&#39;)
sca_ls &lt;-  sapply(levels(fibroblast$cluster), function(x){
    cluster_sub &lt;- fibroblast[[]] %&gt;%
        dplyr::filter(cluster == x)
    VEH_cell &lt;- cluster_sub %&gt;%
        dplyr::filter(Treatment == &quot;VEH&quot;) %&gt;%
        rownames()
    gene2keep_VEH &lt;- rowSums(seurat_data[, colnames(seurat_data) %in% VEH_cell] &gt; 0) &gt; (0.1 * length(VEH_cell))
    DHT_cell &lt;- cluster_sub %&gt;%
        dplyr::filter(Treatment == &quot;DHT&quot;) %&gt;%
        rownames()
    gene2keep_DHT &lt;- rowSums(seurat_data[, colnames(seurat_data) %in% DHT_cell] &gt; 0) &gt; (0.1 * length(DHT_cell))
    count &lt;- seurat_data[gene2keep_DHT|gene2keep_VEH, colnames(seurat_data) %in% rownames(cluster_sub)]
    MAST::FromMatrix(
    exprsArray = as.matrix(count), 
    cData = cluster_sub, 
    fData = data.frame(primerid = rownames(count)) %&gt;%
            set_rownames(rownames(count))
)
}, simplify = FALSE)</code></pre>
<p>Cell detection rate was calculated for each gene within each clsuter
and added to the <code>sca</code> object..</p>
<pre class="r"><code>sca_ls &lt;-  sapply(levels(fibroblast$cluster), function(x){
    temp &lt;- sca_ls[[x]]
    cdr &lt;- colSums(SummarizedExperiment::assay(temp)&gt;0)
    SummarizedExperiment::colData(temp)$ngeneson &lt;- scale(cdr)
    SummarizedExperiment::colData(temp)$Treatment &lt;- factor(SummarizedExperiment::colData(temp)$Treatment, levels = c(&quot;VEH&quot;, &quot;DHT&quot;))
    SummarizedExperiment::colData(temp)$patient &lt;- factor(SummarizedExperiment::colData(temp)$patient)
    SummarizedExperiment::colData(temp)$patient &lt;- droplevels(SummarizedExperiment::colData(temp)$patient)
    temp
}, simplify = FALSE)</code></pre>
<pre class="r"><code># saveRDS(sca_ls, here::here(&quot;data/sca_ls.rds&quot;))</code></pre>
<p>Hurdle model was fitted with the patient as a random effect to nest
the comparisons in patient using linear mixed modelling within each
cluster, leveraging <code>Biocparrel</code> for parallisation.</p>
<pre class="r"><code>cluster_summary_ls &lt;- readRDS(here::here(&quot;output/cluster_summary_ls.rds&quot;)) 
cluster_summary_ls &lt;- sapply(names(cluster_summary_ls), function(x){
    cluster_summary_ls[[x]] %&gt;%
        dplyr::filter(contrast==&#39;TreatmentDHT&#39; &amp; component==&#39;H&#39;) %&gt;%
        dplyr::select(primerid, `Pr(&gt;Chisq)`) %&gt;%
        left_join(
            # logFC
            cluster_summary_ls[[x]] %&gt;%
                dplyr::filter(contrast==&#39;TreatmentDHT&#39; &amp; component==&#39;logFC&#39;) %&gt;%
                dplyr::select(primerid, logFC = coef, ci.hi, ci.lo) 
        )  %&gt;%
        mutate(FDR = p.adjust(`Pr(&gt;Chisq)`, &quot;fdr&quot;),
               DE = ifelse(FDR &lt; 0.01, TRUE, FALSE), 
               cluster = x
               ) %&gt;%
        dplyr::rename(gene_name = primerid)
}, simplify = FALSE)</code></pre>
<p>The numbers of DEGs detected in each cluster using a FDR cut-off of
0.01 are:</p>
<pre class="r"><code>cluster_summary_ls %&gt;%
    lapply(dplyr::filter, DE) %&gt;%
    vapply(nrow, numeric(1)) %&gt;%
    pander()</code></pre>
<table style="width:67%;">
<colgroup>
<col width="11%" />
<col width="11%" />
<col width="11%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">0</th>
<th align="center">1</th>
<th align="center">2</th>
<th align="center">3</th>
<th align="center">4</th>
<th align="center">5</th>
<th align="center">6</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">3,646</td>
<td align="center">3,119</td>
<td align="center">4,941</td>
<td align="center">181</td>
<td align="center">136</td>
<td align="center">217</td>
<td align="center">277</td>
</tr>
</tbody>
</table>
<p>Intersections between DEGs detected in each cluster were visualised
using an Upset plot. 32 consistent DEGs were detected.</p>
<pre class="r"><code>library(ComplexUpset)
cluster_summary_ls %&gt;%
    lapply(dplyr::filter, DE) %&gt;%
    lapply(dplyr::select, cluster, gene_name) %&gt;%
    bind_rows() %&gt;%
    mutate(Present = 1) %&gt;%
    pivot_wider(names_from = cluster, values_from = Present, values_fill = 0) %&gt;%
    column_to_rownames(&quot;gene_name&quot;) %&gt;%
    set_colnames(paste(&quot;Cluster&quot;, colnames(.), sep = &quot; &quot;)) %&gt;%
    ComplexUpset::upset(
        intersect = colnames(.), 
        group_by = &#39;degree&#39;, 
        sort_sets = FALSE,
        # set_sizes = FALSE, 
        queries=list(
            upset_query(intersect=colnames(.), color=&#39;orange&#39;, fill = &#39;orange&#39;)), 
        base_annotations=list(
            &#39;Intersection size&#39;=intersection_size(
                text_mapping=aes(
                    label=ifelse(
                    !!get_size_mode(&#39;exclusive_intersection&#39;) == 32,
                    !!get_size_mode(&#39;exclusive_intersection&#39;), NA), 
                    colour=ifelse(
                    !!get_size_mode(&#39;exclusive_intersection&#39;) == 32,
                    &#39;on_bar&#39;, &#39;on_background&#39;),
                ), 
                text_colors=c(
                     on_bar = &#39;orange&#39;, on_background=&quot;white&quot;
                ), 
                text = list(
                    size = 5
                )
            )
            + ylab(&#39;Intersection size&#39;)
        ), 
        # min_size = 2, 
        themes=upset_default_themes(
            text = element_text(size = 12), 
            axis.title.x = element_blank())
    )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/Differential_Expression.Rmd/unnamed-chunk-19-1.png" alt="*Upset plot showing the intersections between DHT-induced DEGs that were detected in each cluster using MAST. Genes with an FDR   extless 0.01 were considered to be differentially expressed. 32 genes that were consistently differentially expressed in all clusters were identified.*" width="1152" />
<p class="caption">
<em>Upset plot showing the intersections between DHT-induced DEGs that
were detected in each cluster using MAST. Genes with an FDR extless 0.01
were considered to be differentially expressed. 32 genes that were
consistently differentially expressed in all clusters were
identified.</em>
</p>
</div>
<p>LogFCs of the 32 consistent DEGs in each cluster were visaulised and
strong inter-patient heterogeneity was again evident.</p>
<pre class="r"><code>all_clu &lt;- cluster_summary_ls %&gt;%
    lapply(dplyr::filter, DE) %&gt;%
    lapply(pull, gene_name) %&gt;%
    Reduce(intersect,.)
deg32_df &lt;- cluster_summary_ls %&gt;% 
    bind_rows() %&gt;%
    dplyr::filter(gene_name %in% all_clu) %&gt;%
    dplyr::select(gene_name, logFC, cluster) %&gt;%
    mutate(cluster = paste(&quot;Cluster&quot;, cluster, sep = &quot; &quot;))
anno_df &lt;- data.frame(
    cluster = unique(deg32_df$cluster)
) %&gt;%
    mutate(
        `Dominant Patient` = ifelse(
        str_detect(cluster, &quot;2|5&quot;), &quot;MF43&quot;, &quot;MF61&quot;
    )
    )
deg32_df %&gt;%
    pivot_wider(
        names_from = &quot;cluster&quot;, 
        values_from = &quot;logFC&quot;
    ) %&gt;%
    column_to_rownames(&quot;gene_name&quot;) %&gt;%
    pheatmap(
        annotation_col = anno_df %&gt;%
            column_to_rownames(&quot;cluster&quot;), 
        annotation_colors = list(
           `Dominant Patient` = patient_col
        ),
        cutree_cols = 2, 
        cutree_rows = 2,
        legend_breaks = c(seq(-0.4, 0.4, length.out = 5), max(deg32_df$logFC)), 
        legend_labels = c(seq(-0.4, 0.4, length.out = 5), &quot;LogFC\n&quot;)
    )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/Differential_Expression.Rmd/unnamed-chunk-20-1.png" alt="*LogFC of the 32 consistent DEGs in each cluster. Opposite directions of changes between MF43- and MF61-dominant clusters were observed in all genes.*" width="960" />
<p class="caption">
<em>LogFC of the 32 consistent DEGs in each cluster. Opposite directions
of changes between MF43- and MF61-dominant clusters were observed in all
genes.</em>
</p>
</div>
<p>The top 10 global DEGs detected by comparing all DHT-treated cells
against vehicle-treated cells are inspected in cluster-specific DE
analysis results. Differences in those genes’ logFCs in MF-43 dominant
and MF61-dominant clusters are again observed, which was masked for
global DE analysis. These observations demonstrate that performing DE
analysis solely at global levels may obscure cluster-specific responses.
Conducting DE analysis within each cluster, on the other hand, is a more
refined approach that helps unveil cluster-specific transcriptional
changes.</p>
<pre class="r"><code>global_top10 &lt;- zlm_summary_mixed %&gt;%
    dplyr::filter(
        FDR &lt; 0.01
    ) %&gt;%
    arrange(FDR) %&gt;%
    .[1:10, ] %&gt;%
    pull(gene_name)
global_top10_df &lt;- cluster_summary_ls %&gt;% 
    bind_rows() %&gt;%
    dplyr::filter(
        gene_name %in% global_top10, 
        DE) %&gt;%
    dplyr::select(gene_name, logFC, cluster) %&gt;%
    mutate(cluster = paste(&quot;Cluster&quot;, cluster)) %&gt;%
    pivot_wider(
        names_from = &quot;cluster&quot;, 
        values_from = &quot;logFC&quot;
    ) %&gt;%
    column_to_rownames(&quot;gene_name&quot;) 
global_top10_df %&gt;%
    pheatmap(
        annotation_col = anno_df %&gt;%
            column_to_rownames(&quot;cluster&quot;), 
        annotation_colors = list(
           `Dominant Patient` = patient_col
        ),
        cutree_cols = 2, 
        legend_breaks = c(seq(-0.5, 0.5, length.out = 5), max(global_top10_df)), 
        legend_labels = c(seq(-0.5, 0.5, length.out = 5), &quot;LogFC\n&quot;), 
        # display_numbers = sig_anno
    )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/Differential_Expression.Rmd/unnamed-chunk-21-1.png" alt="*LogFC of the 10 most significant global DEGs that were detected by comparing all DHT-treated cells against all vehicle-treated cells in cluster-specific DE analyses. Grey cell indicates that the gene is not differentially expressed in a specific cluster*" width="960" />
<p class="caption">
<em>LogFC of the 10 most significant global DEGs that were detected by
comparing all DHT-treated cells against all vehicle-treated cells in
cluster-specific DE analyses. Grey cell indicates that the gene is not
differentially expressed in a specific cluster</em>
</p>
</div>
</div>
<div id="pathway-analysis" class="section level2">
<h2>Pathway Analysis</h2>
<div id="using-goseq" class="section level3">
<h3>Using <code>GOseq</code></h3>
<p>Enrichment of KEGG pathways among cluster-specific DEGs was tested
using <code>GOseq</code>.</p>
<pre class="r"><code>gsTopology &lt;- retrieve_topology(
    database = &quot;kegg&quot;, species = &quot;hsapiens&quot;)
# names(gsTopology) &lt;- str_remove_all(names(gsTopology), &quot;kegg.&quot;)
source(here::here(&quot;analysis/make_gsNetwork.R&quot;))
kg &lt;- get_GSgenelist(gsTopology)
genesGR &lt;- genesGR %&gt;%
    as.data.frame() %&gt;%
    unnest(entrezid)
kg &lt;- kg %&gt;%
    mutate(gene_id = str_remove_all(gene_id, &quot;ENTREZID:&quot;)) %&gt;%
    dplyr::rename(entrezid = gene_id) %&gt;%
    left_join(genesGR %&gt;%
                  dplyr::select(entrezid, gene_name, gene_id) %&gt;%
                  mutate(entrezid = as.character(entrezid)))
gsTokeep &lt;- readRDS(here::here(&quot;~/20131906_HickeyT_JC_NormalBreast/output/permutedscore_normal.rds&quot;)) %&gt;%
    names()
kg &lt;- kg %&gt;%
    dplyr::filter(gs_name %in% gsTokeep)
kgByGene &lt;- kg  %&gt;%
  split(f = .$gene_id) %&gt;%
  lapply(extract2, &quot;gs_name&quot;)
kgByGs &lt;- kg  %&gt;%
  split(f = .$gs_name) %&gt;%
  lapply(extract2, &quot;entrezid&quot;)
load(system.file(&quot;extdata&quot;, &quot;entrez2name.rda&quot;, package = &quot;sSNAPPY&quot;))</code></pre>
<p>235 KEGG pathways are tested for enrichment.</p>
<p>Enrichment of KEGG pathways among DEGs found with MAST in each
cluster was tested with <code>goseq</code>, using gene length as an
offset.</p>
<pre class="r"><code>de_ls &lt;- sapply(names(cluster_summary_ls), function(x){
    genesGR %&gt;%
        dplyr::filter(gene_name %in% (cluster_summary_ls[[x]] %&gt;%
                                          dplyr::filter(DE) %&gt;%
                                          pull(gene_name))) %&gt;%
        pull(gene_id)
}, simplify = FALSE)
lenPwf_ls &lt;- sapply(names(cluster_summary_ls), function(x){
    lenPwf &lt;-  transGR %&gt;%
        as.data.frame() %&gt;%
        distinct(gene_id, .keep_all = TRUE) %&gt;%
        dplyr::select(gene_id, tx_len) %&gt;%
        mutate(Status = case_when(gene_id %in% de_ls[[x]] ~ 1, !gene_id %in% de_ls[[x]]   ~ 0)) %&gt;%
        with(
            nullp(
                DEgenes = structure(
                    Status, names = gene_id
                ),
                genome = &quot;GRCh38.p16&quot;,
                id = &quot;ensGene&quot;,
                bias.data = tx_len,
                plot.fit = FALSE
            )
        )
}, simplify = FALSE)</code></pre>
<pre class="r"><code>goseq_kg_ls &lt;-  sapply(names(cluster_summary_ls), function(x){
    goseq(lenPwf_ls[[x]],
          gene2cat = kgByGene) %&gt;%
        as_tibble %&gt;%
        dplyr::filter(numDEInCat &gt; 0) %&gt;%
        mutate(
            adjP = p.adjust(over_represented_pvalue, method = &quot;bonf&quot;),
            FDR = as.numeric(p.adjust(over_represented_pvalue, method = &quot;fdr&quot;))
        ) %&gt;%
        dplyr::select(-contains(&quot;under&quot;)) %&gt;%
        dplyr::rename(
            gs_name = category,
            PValue = over_represented_pvalue,
            nDE = numDEInCat,
            nExpressed = numInCat
        ) %&gt;%
        left_join(kg) %&gt;%
        dplyr::select(
            gs_name, nExpressed, nDE, 
            contains(&quot;P&quot;, ignore.case = FALSE),
            FDR,
            gene_name, gene_id
        ) %&gt;%
        dplyr::filter(
            gene_id %in% de_ls[[x]]
        ) %&gt;%
        chop(c(&quot;gene_name&quot;, &quot;gene_id&quot;)) %&gt;%
        mutate(
            gene_name = vapply(.$gene_name, function(x){
                paste(x,collapse = &quot;; &quot;)
            }, character(1)),
            gene_id = vapply(.$gene_id, function(x){
                paste(x,collapse = &quot;; &quot;)
            }, character(1)), 
            cluster = x
        ) %&gt;%
        dplyr::rename(DE = gene_name) %&gt;%
        mutate_at(
            vars(one_of(c(&quot;PValue&quot;, &quot;adjP&quot;, &quot;FDR&quot;))),
            formatP
        )
}, simplify = FALSE)</code></pre>
<p>Using an FDR cut-off of 0.05, the number of pathways that were
significantly enriched in each pathway and their intersections are
displaced in the upset plot below:</p>
<pre class="r"><code>goseq_kg_ls %&gt;%
    lapply(dplyr::filter, FDR &lt; 0.05) %&gt;%
    lapply(pull, gs_name) %&gt;%
    fromList() %&gt;%
    set_colnames(paste(&quot;Cluster&quot;, colnames(.), sep = &quot; &quot;)) %&gt;%
    UpSetR::upset(
        sets = colnames(.), 
        nintersects = NA, 
        keep.order = TRUE,
        queries = list(
          list(
             query = intersects, 
             params = list(setdiff(colnames(.), &quot;Cluster 6&quot;)),
             color = &quot;orange&quot;,
             active = T),
          list(
             query = intersects, 
             params = list(&quot;Cluster 1&quot;),
             color = &#39;forestgreen&#39;,
             active = T), 
          list(
             query = intersects, 
             params = list(setdiff(colnames(.), c(&quot;Cluster 2&quot;, &quot;Cluster 6&quot;))),
             color = &#39;#9565CD&#39;,
             active = T)
        ), 
        text.scale = 2
    )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/Differential_Expression.Rmd/unnamed-chunk-24-1.png" alt="*Upset plot of KEGG pathways significantly over-represented among DEGs detected in each cluster.*" width="960" />
<p class="caption">
<em>Upset plot of KEGG pathways significantly over-represented among
DEGs detected in each cluster.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Wenjun-Liu/DHT_fibroblast_snRNAseq/blob/766d02da0c943d4e0228cf1a615aedfb0c8e46cd/docs/figure/Differential_Expression.Rmd/unnamed-chunk-24-1.png" target="_blank">766d02d</a>
</td>
<td>
Wenjun-Liu
</td>
<td>
2023-06-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>all_but6 &lt;- goseq_kg_ls[c(&quot;0&quot;,&quot;2&quot;, &quot;5&quot;, &quot;1&quot;, &quot;3&quot;, &quot;4&quot;)] %&gt;%
    lapply(dplyr::filter, FDR &lt; 0.05) %&gt;%
    lapply(pull, gs_name) %&gt;%
    Reduce(intersect, .) 
both_pat_gs &lt;-  goseq_kg_ls[c(&quot;0&quot;, &quot;5&quot;, &quot;1&quot;, &quot;3&quot;, &quot;4&quot;)] %&gt;%
    lapply(dplyr::filter, FDR &lt; 0.05) %&gt;%
    lapply(pull, gs_name) %&gt;%
    Reduce(intersect, .) %&gt;%
    setdiff(., all_but6)</code></pre>
<pre class="r"><code>clu1_specific_gs &lt;- setdiff(
    goseq_kg_ls$`1` %&gt;%
        dplyr::filter(FDR &lt; 0.05) %&gt;%
        pull(gs_name), 
    goseq_kg_ls[setdiff(names(goseq_kg_ls), &quot;1&quot;)] %&gt;%
        lapply(dplyr::filter, FDR &lt; 0.05) %&gt;%
        lapply(pull, gs_name) %&gt;%
        Reduce(union,.)
)</code></pre>
<p>The pathway that was significantly enriched in all clusters but
cluster 6 (highlighted in orange) is kegg.HIF-1 signaling pathway, and
the gene-set that was enriched in 5 clusters (hightlighted in purple),
both MF43-dominant and MF61-dominant ones, was kegg.Proteoglycans in
cancer.</p>
<p>There were also 8 pathways that were uniquely enriched in cluster 1,
which was predicted to be more stem-like than cells in other clusters.
The 8 cluster1-specific pathways were Mitophagy - animal, Prolactin
signaling pathway, C-type lectin receptor signaling pathway, Ras
signaling pathway, Wnt signaling pathway, T cell receptor signaling
pathway, Bladder cancer, Lipid and atherosclerosis, among which the WNT
pathway was identified.</p>
<p>The WNT Signalling Pathway is known to be a critical mediator of cell
stemness. Enrichment of this pathway exclusively in cluster 1 suggests
that AR activation might have an impact on the stemness of fibroblasts
assigned to cluster 1.</p>
</div>
<div id="using-aucell" class="section level3">
<h3>Using <code>AUCell</code></h3>
<p>To further explore whether cells in cluster 1 represent a more
stem-like state of fibroblasts and if AR activation can change the
stemness of cells, we performed a pathway scoring analysis using
<code>AUCell</code>, which is a method specifically designed for
sc/snRNA-seq data that allows scoring of pathway activity at the
individual cell level.</p>
<p>A list of 164 human stem cell signature was retrieved from the study
by Barata et al. </p>
<p>Using <code>AUCell</code>, genes are ranked based on their expression
within each cell, and one stemness score was computed for each cell
based on the ranking of the stemness genes.</p>
<pre class="r"><code>stem_gs &lt;- list(
    &quot;stemness&quot; = read_csv(
        here::here(&quot;data/stem_sig.txt&quot;), col_names = &quot;gene_name&quot;) %&gt;%
        pull(gene_name)
)</code></pre>
<pre class="r"><code>seurat_count &lt;- GetAssayData(object = fibroblast, slot = &#39;counts&#39;)
AUC_stemness &lt;- AUCell_run(
    geneSets = stem_gs, 
    exprMat = seurat_count, 
    aucMaxRank = ceiling(0.2 * nrow(seurat_count)),
    BPPARAM=BiocParallel::MulticoreParam(8)
)
AUC_matrix_stemness &lt;- getAUC(AUC_stemness)</code></pre>
<p>Distributions of the computed AUC stemness scores in cells of
different clusters and treatments were visualized. While the untreated
cells in cluster 5 had the highest median stemness scores, the median
stemness score of the vehicle-treated cells in cluster 1 was higher than
all the remaining clusters. The DHT-treated cells in cluster 1 also
exhibited the highest median stemness score when compared to the
DHT-treated cells in other clusters.</p>
<pre class="r"><code>AUC_matrix_stemness %&gt;%
        t() %&gt;%
    as.data.frame() %&gt;%
    rownames_to_column(&quot;cell&quot;) %&gt;%
    left_join(
        fibroblast[[]] %&gt;%
            rownames_to_column(&quot;cell&quot;)
    ) %&gt;%
    mutate(
        Treatment = factor(Treatment, levels = c(&quot;VEH&quot;, &quot;DHT&quot;)), 
        cluster = paste(&quot;Cluster&quot;, cluster, sep = &quot; &quot;)
    ) %&gt;%
    ggplot(
        aes(cluster, stemness, fill = Treatment)
    ) +
    geom_boxplot() +
    scale_fill_manual(values = treat_col) +
    labs(
        x = &quot;&quot;, 
        y = &quot;AUC Stemness Score&quot;
    ) +
    theme(axis.text.x = element_text(
        colour = c(&quot;black&quot;, &quot;orange&quot;, rep(&quot;black&quot;, 5))
    ))</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/Differential_Expression.Rmd/unnamed-chunk-29-1.png" alt="*Cell stemness scores that were computed by scoring individual cells against a list of 164 human stem cell signature genes using the AUCell method. Cells are grouped by their treatment and cluster allocation.*" width="960" />
<p class="caption">
<em>Cell stemness scores that were computed by scoring individual cells
against a list of 164 human stem cell signature genes using the AUCell
method. Cells are grouped by their treatment and cluster
allocation.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-29-1">
Past versions of unnamed-chunk-29-1.png
</button>
</p>
<div id="fig-unnamed-chunk-29-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Wenjun-Liu/DHT_fibroblast_snRNAseq/blob/766d02da0c943d4e0228cf1a615aedfb0c8e46cd/docs/figure/Differential_Expression.Rmd/unnamed-chunk-29-1.png" target="_blank">766d02d</a>
</td>
<td>
Wenjun-Liu
</td>
<td>
2023-06-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># ECM_gs &lt;- msigdbr(&quot;Homo sapiens&quot;, category = &quot;C2&quot;, subcategory = &quot;CP&quot;) %&gt;%
#     dplyr::filter(
#         str_detect(gs_name, &quot;MATRISOME|ECM&quot;)
#     ) %&gt;%
#   dplyr::rename(gene_id = ensembl_gene) %&gt;%
#   dplyr::filter(!is.na(gene_id)) %&gt;%
#   distinct(gs_name, gene_id, .keep_all = TRUE)
# ecmByGene &lt;- ECM_gs  %&gt;%
#   split(f = .$gene_id) %&gt;%
#   lapply(extract2, &quot;gs_name&quot;)
# ecmByID &lt;- ECM_gs  %&gt;%
#   split(f = .$gs_name) %&gt;%
#   lapply(extract2, &quot;gene_symbol&quot;)</code></pre>
</div>
</div>
</div>
<div id="thesis-figure" class="section level1">
<h1>Thesis figure</h1>
<p>Code for reproducing figures used in the theis:</p>
<!-- ```{r} -->
<!-- library(writexl) -->
<!-- # save global DEGs to excel spreadsheet -->
<!-- zlm_summary_mixed %>% -->
<!--     dplyr::filter( -->
<!--         FDR < 0.01 -->
<!--     ) %>% -->
<!--     dplyr::select(gene_name, Pvalue = `Pr(>Chisq)`,  -->
<!--            logFC, FDR) %>% -->
<!--     write_xlsx(here::here("output/DE_global.xlsx")) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- # Table 5.3 -->
<!-- zlm_summary_mixed %>% -->
<!--     dplyr::filter( -->
<!--         FDR < 0.01,  -->
<!--         gene_name %in% AR_sig$gene_name -->
<!--     )  %>% -->
<!--     left_join( -->
<!--         AR_sig  -->
<!--     ) %>% -->
<!--     mutate( -->
<!--         Align = case_when( -->
<!--             logFC < 0 & AR_sig == "Down" ~ "✓", -->
<!--             logFC > 0 & AR_sig == "Up" ~ "✓", -->
<!--             logFC < 0 & AR_sig == "Up" ~ "✗", -->
<!--             logFC > 0 & AR_sig == "Down" ~ "✗" -->
<!--         ),  -->
<!--         AR_sig = ifelse( -->
<!--             AR_sig == "Down", "\\color{blue}Down", "\\color{red}Up" -->
<!--         ),  -->
<!--         gene_name = paste("\\textit{", gene_name, "}", sep = "") -->
<!--     ) %>% -->
<!--     mutate_if(is.character, as.factor) %>% -->
<!--     .[order(.$FDR),] %>% -->
<!--     mutate_at(vars(c("Pr(>Chisq)", "FDR")), formatP) %>% -->
<!--     mutate(logFC = ifelse( -->
<!--         abs(logFC) < 0.01, sprintf("%.2e", logFC), sprintf("%.2f", logFC))) %>% -->
<!--     # mutate(logFC = as.character(round(logFC, 4))) %>% -->
<!--     dplyr::select( -->
<!--         `Gene Name` = gene_name, -->
<!--         `Expected Direction` = AR_sig, -->
<!--         logFC,FDR, Concordant = Align -->
<!--     ) %>% -->
<!--     xtable::xtable( -->
<!--         caption = "AR signatures defined in ER-positive BC\\cite{hickey_androgen_2021} that were also differentially expressed upon DHT treatment in mammary fibroblasts. The concordant column reflects whether the signs of logFCs in fibroblasts agreed with the signature types in malignant epithelial cells (ie. expected direction).",  -->
<!--         label = "chap5_table3" -->
<!--     )  %>% -->
<!--      xtable::print.xtable( -->
<!--         "~/PhD_thesis/draft_figure/chapter_05/table3.tex", type = "latex", -->
<!--         caption.placement = "top", -->
<!--         include.rownames = FALSE, -->
<!--         sanitize.text.function = function(x) x) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- chap5_fig7 <- cor_plot & -->
<!--     theme( -->
<!--         text = element_text(size = 50) -->
<!--     ) -->
<!-- # png( -->
<!-- #     "/Users/wenjunliu/PhD_thesis/Images/Chapter_05/chap5_fig7.png", -->
<!-- #     height = 1600, -->
<!-- #     width = 2000 -->
<!-- # ) -->
<!-- #chap5_fig7 -->
<!-- # dev.off() -->
<!-- ``` -->
<!-- ```{r} -->
<!-- # Table of num of DEGs in each cluster -->
<!-- cluster_summary_ls %>% -->
<!--     lapply(dplyr::filter, DE) %>% -->
<!--     vapply(nrow, numeric(1)) %>% -->
<!--     enframe( -->
<!--         name = "Cluster",  -->
<!--         value = "Number of DEGs" -->
<!--     ) %>% -->
<!--     left_join( -->
<!--         cluster_summary_ls %>% -->
<!--             vapply(nrow, numeric(1)) %>% -->
<!--             enframe( -->
<!--                 name = "Cluster",  -->
<!--                 value = "Number of tested genes" -->
<!--             )   -->
<!--     ) %>% -->
<!--     xtable::xtable( -->
<!--         caption = "The numbers of tested genes and detected DEGs in each cluster. DE analyses were performed using MAST with patient as a random effect. Only genes that were expressed in at least 10% of either DHT- or vehicle-treated cells in each cluster were testde for differential expression. Genes with an FDR smaller than 0.01 were considered to be DEGs",  -->
<!--         label = "chap5_tableDE",  -->
<!--         digits = 0 -->
<!--     )  %>% -->
<!--     xtable::print.xtable( -->
<!--         "~/PhD_thesis/draft_figure/chapter_05/clsuter_DE.tex", type = "latex", -->
<!--         caption.placement = "top", -->
<!--         include.rownames = FALSE, -->
<!--         sanitize.text.function = function(x) x) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- #DE upset plot -->
<!-- DE_upset <- cluster_summary_ls %>% -->
<!--     lapply(dplyr::filter, DE) %>% -->
<!--     lapply(dplyr::select, cluster, gene_name) %>% -->
<!--     bind_rows() %>% -->
<!--     mutate(Present = 1) %>% -->
<!--     pivot_wider(names_from = cluster, values_from = Present, values_fill = 0) %>% -->
<!--     column_to_rownames("gene_name") %>% -->
<!--     set_colnames(paste("Cluster", colnames(.), sep = " ")) %>% -->
<!--     ComplexUpset::upset( -->
<!--         intersect = colnames(.),  -->
<!--         group_by = 'degree',  -->
<!--         sort_sets = FALSE, -->
<!--         set_sizes = FALSE,  -->
<!--         queries=list( -->
<!--             upset_query(intersect=colnames(.), color='orange', fill = 'orange')),  -->
<!--         base_annotations=list( -->
<!--             'Intersection size'=intersection_size( -->
<!--                 text_mapping=aes( -->
<!--                     label=ifelse( -->
<!--                     !!get_size_mode('exclusive_intersection') == 32, -->
<!--                     !!get_size_mode('exclusive_intersection'), NA),  -->
<!--                     colour=ifelse( -->
<!--                     !!get_size_mode('exclusive_intersection') == 32, -->
<!--                     'on_bar', 'on_background'), -->
<!--                 ),  -->
<!--                 text_colors=c( -->
<!--                      on_bar = 'orange', on_background="white" -->
<!--                 ),  -->
<!--                 text = list( -->
<!--                     size = 5 -->
<!--                 ) -->
<!--             ) -->
<!--             + ylab('Intersection size') -->
<!--         ),  -->
<!--         # min_size = 2,  -->
<!--         themes=upset_default_themes( -->
<!--             text = element_text(size = 12),  -->
<!--             axis.title.x = element_blank()) -->
<!--     ) -->
<!-- # png( -->
<!-- #     "/Users/wenjunliu/PhD_thesis/Images/chapter_05/DE_upset.png", -->
<!-- #     width = 210, height =130, units='mm', res = 300 -->
<!-- # ) -->
<!-- # DE_upset -->
<!-- # dev.off() -->
<!-- ``` -->
<!-- ```{r} -->
<!-- # png( -->
<!-- #     "/Users/wenjunliu/PhD_thesis/Images/chapter_05/hp_32DEG.png", -->
<!-- #     width = 190, height =140, units='mm', res = 300 -->
<!-- # ) -->
<!-- # deg32_df %>% -->
<!-- #     pivot_wider( -->
<!-- #         names_from = "cluster",  -->
<!-- #         values_from = "logFC" -->
<!-- #     ) %>% -->
<!-- #     column_to_rownames("gene_name") %>% -->
<!-- #     pheatmap( -->
<!-- #         annotation_col = anno_df %>% -->
<!-- #             column_to_rownames("cluster"),  -->
<!-- #         annotation_colors = list( -->
<!-- #            `Dominant Patient` = patient_col -->
<!-- #         ), -->
<!-- #         cutree_cols = 2,  -->
<!-- #         cutree_rows = 2, -->
<!-- #         legend_breaks = c(seq(-0.4, 0.4, length.out = 5), max(deg32_df$logFC)),  -->
<!-- #         legend_labels = c(seq(-0.4, 0.4, length.out = 5), "LogFC\n") -->
<!-- #     ) -->
<!-- # dev.off() -->
<!-- ``` -->
<!-- ```{r} -->
<!-- # png( -->
<!-- #     "/Users/wenjunliu/PhD_thesis/Images/chapter_05/hp_global_top10.png", -->
<!-- #     width = 190, height =140, units='mm', res = 300 -->
<!-- # ) -->
<!-- # global_top10_df %>% -->
<!-- #     pheatmap( -->
<!-- #         annotation_col = anno_df %>% -->
<!-- #             column_to_rownames("cluster"),  -->
<!-- #         annotation_colors = list( -->
<!-- #            `Dominant Patient` = patient_col -->
<!-- #         ), -->
<!-- #         cutree_cols = 2,  -->
<!-- #         legend_breaks = c(seq(-0.5, 0.5, length.out = 5), max(global_top10_df)),  -->
<!-- #         legend_labels = c(seq(-0.5, 0.5, length.out = 5), "LogFC\n") -->
<!-- #     ) -->
<!-- # dev.off() -->
<!-- ``` -->
<!-- ```{r} -->
<!-- #save cluster-specific DE results to spreadsheet -->
<!-- library(writexl) -->
<!-- DE_suppTable <- cluster_summary_ls %>% -->
<!--     lapply(dplyr::filter, DE) %>% -->
<!--     lapply(dplyr::select, gene_name, Pvalue = `Pr(>Chisq)`,  -->
<!--            logFC, FDR,  cluster) -->
<!-- names(DE_suppTable) <- paste("Cluster", names(DE_suppTable), sep = "_") -->
<!-- # write_xlsx(DE_suppTable, here::here("output/DE_suppTable.xlsx")) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- goseq_kg_ls %>% -->
<!--     lapply(dplyr::filter, FDR < 0.05) %>% -->
<!--     vapply(nrow, numeric(1)) %>% -->
<!--     enframe( -->
<!--         name = "Cluster",  -->
<!--         value = "Number of significantly\nenriched pathways" -->
<!--     ) %>% -->
<!--     xtable::xtable( -->
<!--         caption = "The numbers of KEGG pathways that were significantly over-repsented in each cluster. The enrichment analysis was performed using GOseq\\cite{young_mgeethnode_2010}, with gene length as an offset term to account for any bias. Pathways with an FDR \\textless 0.05 were considered to be significantly enriched.",  -->
<!--         label = "path_numb",  -->
<!--         digits = 0 -->
<!--     )  %>% -->
<!--     xtable::print.xtable( -->
<!--         "~/PhD_thesis/draft_figure/chapter_05/path_num.tex", type = "latex", -->
<!--         caption.placement = "top", -->
<!--         include.rownames = FALSE, -->
<!--         sanitize.text.function = function(x) x) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- ## Supp Table: significantly enriched pathways in each cluster -->
<!-- path_suppTable <- goseq_kg_ls %>% -->
<!--     lapply(dplyr::filter, FDR < 0.05) %>% -->
<!--     lapply(function(x){mutate(x, gs_name = str_remove_all(x$gs_name, "kegg."))}) %>% -->
<!--     lapply(dplyr::select, gs_name, nExpressed, nDE, FDR) -->
<!-- names(path_suppTable) <- paste("Cluster", names(path_suppTable), sep = "_") -->
<!-- # write_xlsx(path_suppTable, here::here("output/path_suppTable.xlsx")) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- # png( -->
<!-- #     "/Users/wenjunliu/PhD_thesis/Images/chapter_05/pathway_upset.png", -->
<!-- #     width = 230, height =150, units='mm', res = 300 -->
<!-- # ) -->
<!-- # goseq_kg_ls %>% -->
<!-- #     lapply(dplyr::filter, FDR < 0.05) %>% -->
<!-- #     lapply(pull, gs_name) %>% -->
<!-- #     fromList() %>% -->
<!-- #     set_colnames(paste("Cluster", colnames(.), sep = " ")) %>% -->
<!-- #     UpSetR::upset( -->
<!-- #         sets = colnames(.),  -->
<!-- #         nintersects = NA,  -->
<!-- #         keep.order = TRUE, -->
<!-- #         queries = list( -->
<!-- #           list( -->
<!-- #              query = intersects,  -->
<!-- #              params = list(setdiff(colnames(.), "Cluster 6")), -->
<!-- #              color = "orange", -->
<!-- #              active = T), -->
<!-- #           list( -->
<!-- #              query = intersects,  -->
<!-- #              params = list("Cluster 1"), -->
<!-- #              color = 'forestgreen', -->
<!-- #              active = T),  -->
<!-- #           list( -->
<!-- #              query = intersects,  -->
<!-- #              params = list(setdiff(colnames(.), c("Cluster 2", "Cluster 6"))), -->
<!-- #              color = '#9565CD', -->
<!-- #              active = T) -->
<!-- #         ),  -->
<!-- #         text.scale = 2 -->
<!-- #     ) -->
<!-- # dev.off() -->
<!-- ``` -->
<!-- ```{r} -->
<!-- goseq_kg_ls[["1"]] %>% -->
<!--     dplyr::filter(gs_name %in% clu1_specific_gs) %>% -->
<!--     mutate( -->
<!--         gs_name = str_remove_all(gs_name, "kegg.") -->
<!--     ) %>% -->
<!--     dplyr::select( -->
<!--         `Gene-set name` = gs_name,  -->
<!--         nDE,  nExpressed, FDR -->
<!--     ) %>% -->
<!--     xtable::xtable( -->
<!--         caption = "KEGG pathways that were uniquely over-represented due to the DHT treatment in cluster 1",  -->
<!--         label = "clu1_unique",  -->
<!--         digits = 0 -->
<!--     )  %>% -->
<!--     xtable::print.xtable( -->
<!--         "~/PhD_thesis/draft_figure/chapter_05/clu1_unique.tex", type = "latex", -->
<!--         caption.placement = "top", -->
<!--         include.rownames = FALSE, -->
<!--         sanitize.text.function = function(x) x) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- stem_boxplot <- AUC_matrix_stemness %>% -->
<!--         t() %>% -->
<!--     as.data.frame() %>% -->
<!--     rownames_to_column("cell") %>% -->
<!--     left_join( -->
<!--         fibroblast[[]] %>% -->
<!--             rownames_to_column("cell") -->
<!--     ) %>% -->
<!--     mutate( -->
<!--         Treatment = factor(Treatment, levels = c("VEH", "DHT")) -->
<!--     ) %>% -->
<!--     ggplot( -->
<!--         aes(cluster, stemness, fill = Treatment) -->
<!--     ) + -->
<!--     geom_boxplot() + -->
<!--     scale_fill_manual(values = treat_col) + -->
<!--     labs( -->
<!--         x = "Cluster",  -->
<!--         y = "Cell Stemness Score" -->
<!--     ) + -->
<!--     theme( -->
<!--         text = element_text(size = 12), -->
<!--         axis.text.x = element_text( -->
<!--         colour = c("black","orange", rep("black", 5)),  -->
<!--         size = c(12, 20, rep(12, 5)) -->
<!--     )) -->
<!-- # png( -->
<!-- #     "/Users/wenjunliu/PhD_thesis/Images/chapter_05/stem_boxplot.png", -->
<!-- #     width = 180, height =100, units='mm', res = 300 -->
<!-- # ) -->
<!-- # stem_boxplot -->
<!-- # dev.off() -->
<!-- ``` -->
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21)
Platform: x86_64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.3.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Australia/Adelaide
tzcode source: internal

attached base packages:
 [1] grid      splines   stats4    stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] ComplexUpset_1.3.3          AUCell_1.22.0              
 [3] sSNAPPY_1.5.1               goseq_1.52.0               
 [5] geneLenDataBase_1.36.0      BiasedUrn_2.0.10           
 [7] msigdbr_7.5.1               UpSetR_1.4.0               
 [9] BiocParallel_1.34.2         pheatmap_1.0.12            
[11] MAST_1.26.0                 SingleCellExperiment_1.22.0
[13] SummarizedExperiment_1.30.2 MatrixGenerics_1.12.2      
[15] matrixStats_1.0.0           ggforce_0.4.1              
[17] randomcoloR_1.1.0.1         ggpubr_0.6.0               
[19] colorspace_2.1-0            patchwork_1.1.2            
[21] plotly_4.10.2               corrplot_0.92              
[23] SeuratObject_4.1.3          Seurat_4.3.0.1             
[25] DT_0.28                     ggrepel_0.9.3              
[27] cqn_1.46.0                  quantreg_5.95              
[29] SparseM_1.81                preprocessCore_1.62.1      
[31] nor1mix_1.3-0               mclust_6.0.0               
[33] magrittr_2.0.3              ggfortify_0.4.16           
[35] cowplot_1.1.1               ensembldb_2.24.0           
[37] AnnotationFilter_1.24.0     GenomicFeatures_1.52.1     
[39] AnnotationDbi_1.62.2        Biobase_2.60.0             
[41] GenomicRanges_1.52.0        GenomeInfoDb_1.36.1        
[43] IRanges_2.34.1              S4Vectors_0.38.1           
[45] AnnotationHub_3.8.0         BiocFileCache_2.8.0        
[47] dbplyr_2.3.3                BiocGenerics_0.46.0        
[49] edgeR_3.42.4                limma_3.56.2               
[51] glue_1.6.2                  pander_0.6.5               
[53] scales_1.2.1                yaml_2.3.7                 
[55] lubridate_1.9.2             forcats_1.0.0              
[57] stringr_1.5.0               dplyr_1.1.2                
[59] purrr_1.0.1                 readr_2.1.4                
[61] tidyr_1.3.0                 tibble_3.2.1               
[63] ggplot2_3.4.2               tidyverse_2.0.0            

loaded via a namespace (and not attached):
  [1] fs_1.6.2                      ProtGenerics_1.32.0          
  [3] spatstat.sparse_3.0-2         bitops_1.0-7                 
  [5] httr_1.4.6                    RColorBrewer_1.1-3           
  [7] backports_1.4.1               tools_4.3.0                  
  [9] sctransform_0.3.5             utf8_1.2.3                   
 [11] R6_2.5.1                      mgcv_1.9-0                   
 [13] lazyeval_0.2.2                uwot_0.1.16                  
 [15] withr_2.5.0                   sp_2.0-0                     
 [17] graphite_1.46.0               prettyunits_1.1.1            
 [19] gridExtra_2.3                 progressr_0.13.0             
 [21] cli_3.6.1                     spatstat.explore_3.2-1       
 [23] labeling_0.4.2                sass_0.4.6                   
 [25] spatstat.data_3.0-1           ggridges_0.5.4               
 [27] pbapply_1.7-2                 Rsamtools_2.16.0             
 [29] R.utils_2.12.2                parallelly_1.36.0            
 [31] rstudioapi_0.15.0             RSQLite_2.3.1                
 [33] generics_0.1.3                BiocIO_1.10.0                
 [35] vroom_1.6.3                   crosstalk_1.2.0              
 [37] gtools_3.9.4                  ica_1.0-3                    
 [39] spatstat.random_3.1-5         car_3.1-2                    
 [41] GO.db_3.17.0                  Matrix_1.6-0                 
 [43] fansi_1.0.4                   abind_1.4-5                  
 [45] R.methodsS3_1.8.2             lifecycle_1.0.3              
 [47] whisker_0.4.1                 carData_3.0-5                
 [49] Rtsne_0.16                    blob_1.2.4                   
 [51] promises_1.2.0.1              crayon_1.5.2                 
 [53] miniUI_0.1.1.1                lattice_0.21-8               
 [55] annotate_1.78.0               KEGGREST_1.40.0              
 [57] pillar_1.9.0                  knitr_1.43                   
 [59] rjson_0.2.21                  future.apply_1.11.0          
 [61] codetools_0.2-19              leiden_0.4.3                 
 [63] V8_4.3.2                      data.table_1.14.8            
 [65] vctrs_0.6.3                   png_0.1-8                    
 [67] gtable_0.3.3                  cachem_1.0.8                 
 [69] xfun_0.39                     S4Arrays_1.0.4               
 [71] mime_0.12                     tidygraph_1.2.3              
 [73] survival_3.5-5                interactiveDisplayBase_1.38.0
 [75] ellipsis_0.3.2                fitdistrplus_1.1-11          
 [77] ROCR_1.0-11                   nlme_3.1-162                 
 [79] bit64_4.0.5                   progress_1.2.2               
 [81] filelock_1.0.2                RcppAnnoy_0.0.21             
 [83] rprojroot_2.0.3               bslib_0.5.0                  
 [85] irlba_2.3.5.1                 KernSmooth_2.23-21           
 [87] DBI_1.1.3                     tidyselect_1.2.0             
 [89] bit_4.0.5                     compiler_4.3.0               
 [91] curl_5.0.1                    git2r_0.32.0                 
 [93] graph_1.78.0                  xml2_1.3.5                   
 [95] DelayedArray_0.26.6           rtracklayer_1.60.0           
 [97] lmtest_0.9-40                 rappdirs_0.3.3               
 [99] digest_0.6.33                 goftest_1.2-3                
[101] spatstat.utils_3.0-3          rmarkdown_2.23               
[103] XVector_0.40.0                htmltools_0.5.5              
[105] pkgconfig_2.0.3               sparseMatrixStats_1.12.2     
[107] highr_0.10                    fastmap_1.1.1                
[109] rlang_1.1.1                   htmlwidgets_1.6.2            
[111] DelayedMatrixStats_1.22.1     shiny_1.7.4.1                
[113] farver_2.1.1                  jquerylib_0.1.4              
[115] zoo_1.8-12                    jsonlite_1.8.7               
[117] R.oo_1.25.0                   RCurl_1.98-1.12              
[119] GenomeInfoDbData_1.2.10       munsell_0.5.0                
[121] Rcpp_1.0.11                   viridis_0.6.4                
[123] babelgene_22.9                reticulate_1.30              
[125] stringi_1.7.12                ggraph_2.1.0                 
[127] zlibbioc_1.46.0               MASS_7.3-60                  
[129] org.Hs.eg.db_3.17.0           plyr_1.8.8                   
[131] parallel_4.3.0                listenv_0.9.0                
[133] deldir_1.0-9                  graphlayouts_1.0.0           
[135] Biostrings_2.68.1             tensor_1.5                   
[137] hms_1.1.3                     locfit_1.5-9.8               
[139] igraph_1.5.0                  spatstat.geom_3.2-2          
[141] ggsignif_0.6.4                reshape2_1.4.4               
[143] biomaRt_2.56.1                BiocVersion_3.17.1           
[145] XML_3.99-0.14                 evaluate_0.21                
[147] BiocManager_1.30.21           tweenr_2.0.2                 
[149] tzdb_0.4.0                    httpuv_1.6.11                
[151] MatrixModels_0.5-2            RANN_2.6.1                   
[153] polyclip_1.10-4               future_1.33.0                
[155] scattermore_1.2               broom_1.0.5                  
[157] xtable_1.8-4                  restfulr_0.0.15              
[159] rstatix_0.7.2                 later_1.3.1                  
[161] viridisLite_0.4.2             memoise_2.0.1                
[163] GenomicAlignments_1.36.0      cluster_2.1.4                
[165] workflowr_1.7.0               timechange_0.2.0             
[167] globals_0.16.2                here_1.0.1                   
[169] GSEABase_1.62.0              </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
