<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Wenjun Liu Dame Roma Mitchell Cancer Research Laboratories Adelaide Medical School University of Adelaide" />

<meta name="date" content="2023-01-18" />

<title>Differential_Expression</title>

<script src="site_libs/header-attrs-2.21/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.27/datatables.js"></script>
<link href="site_libs/dt-core-1.12.1/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.12.1/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.12.1/js/jquery.dataTables.min.js"></script>
<link href="site_libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet" />
<script src="site_libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="site_libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet" />
<script src="site_libs/selectize-0.12.0/selectize.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DHT_fibroblast_snRNAseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Preprocess_fibro_from10x.html">Pre-processing</a>
</li>
<li>
  <a href="Subtype_Prediction.html">Cell Type Annotation</a>
</li>
<li>
  <a href="Differential_Expression.html">Differential Expression &amp; Pathway Analysis</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Differential_Expression</h1>
<h4 class="author">Wenjun Liu<br>Dame Roma Mitchell Cancer Research
Laboratories<br>Adelaide Medical School<br>University of Adelaide</h4>
<h4 class="date">2023-01-18</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-06-07
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>DHT_fibroblast_snRNAseq/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230528code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230528)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230528code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230528)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomWenjunLiuDHTfibroblastsnRNAseqtree2a242676e6a7843e55e195d7b05e45847b7a1b4btargetblank2a24267a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/Wenjun-Liu/DHT_fibroblast_snRNAseq/tree/2a242676e6a7843e55e195d7b05e45847b7a1b4b" target="_blank">2a24267</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomWenjunLiuDHTfibroblastsnRNAseqtree2a242676e6a7843e55e195d7b05e45847b7a1b4btargetblank2a24267a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/Wenjun-Liu/DHT_fibroblast_snRNAseq/tree/2a242676e6a7843e55e195d7b05e45847b7a1b4b" target="_blank">2a24267</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/mf43/
    Ignored:    data/mf61/
    Ignored:    data/sca.rds
    Ignored:    output/fibroblast_seurat.rds

Untracked files:
    Untracked:  .DS_Store
    Untracked:  analysis/Differential_Expression.Rmd
    Untracked:  analysis/Preprocess_fibro_from10x.Rmd
    Untracked:  analysis/Subtype_Prediction.Rmd
    Untracked:  analysis/make_gsNetwork.R
    Untracked:  analysis/plot_commu_temp.R
    Untracked:  data/.DS_Store
    Untracked:  data/AR_sig_down.csv
    Untracked:  data/AR_sig_up.csv
    Untracked:  data/Wu_2020/
    Untracked:  data/genesGR.rds
    Untracked:  output/Table_Chap5_Supp3.xlsx
    Untracked:  output/pred_hpca.rds
    Untracked:  output/pred_wu.rds
    Untracked:  output/zlm_summary_mixed.rds

Unstaged changes:
    Modified:   .gitignore
    Modified:   README.md
    Modified:   analysis/_site.yml
    Deleted:    analysis/about.Rmd
    Modified:   analysis/index.Rmd
    Deleted:    analysis/license.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)
library(yaml)
library(scales)
library(pander)
library(glue)
library(edgeR)
library(AnnotationHub)
library(ensembldb)
library(cowplot)
library(ggfortify)
library(magrittr)
library(cqn)
library(ggrepel)
library(DT)
library(Seurat)
library(corrplot)
library(plotly)
library(patchwork)
library(colorspace)
library(ggpubr)
library(randomcoloR)
library(ggforce)
library(MAST)
library(pheatmap)
library(BiocParallel)
library(UpSetR)
library(msigdbr)
library(goseq)
library(sSNAPPY)</code></pre>
<pre class="r"><code>panderOptions(&quot;table.split.table&quot;, Inf)
panderOptions(&quot;big.mark&quot;, &quot;,&quot;)
theme_set(theme_bw())</code></pre>
<pre class="r"><code>formatP &lt;- function(p, m = 0.0001){
out &lt;- rep(&quot;&quot;, length(p))
out[p &lt; m] &lt;- sprintf(&quot;%.2e&quot;, p[p&lt;m])
out[p &gt;= m] &lt;- sprintf(&quot;%.4f&quot;, p[p&gt;=m])
out
}</code></pre>
<div id="gene-annotations" class="section level2">
<h2>Gene Annotations</h2>
<pre class="r"><code>ah &lt;- AnnotationHub() %&gt;%
  subset(rdataclass == &quot;EnsDb&quot;) %&gt;%
  subset(str_detect(description, &quot;101&quot;)) %&gt;%
  subset(genome == &quot;GRCh38&quot;)
stopifnot(length(ah) == 1)</code></pre>
<pre class="r"><code>ensDb &lt;- ah[[1]]
genesGR &lt;- genes(ensDb)
transGR &lt;- transcripts(ensDb)</code></pre>
<pre class="r"><code>mcols(transGR) &lt;- mcols(transGR) %&gt;%
  cbind(
    transcriptLengths(ensDb)[rownames(.), c(&quot;nexon&quot;, &quot;tx_len&quot;)]
  )</code></pre>
<pre class="r"><code>mcols(genesGR) &lt;- mcols(genesGR) %&gt;%
  as.data.frame() %&gt;%
  dplyr::select(
    gene_id, gene_name, gene_biotype, entrezid
  ) %&gt;%
  left_join(
    mcols(transGR) %&gt;%
      as.data.frame() %&gt;%
      mutate(
        tx_support_level = case_when(
          is.na(tx_support_level) ~ 1L, 
          TRUE ~ tx_support_level
        )
      ) %&gt;%
      group_by(gene_id) %&gt;%
      summarise(
        n_tx = n(),
        longest_tx = max(tx_len),
        ave_tx_len = mean(tx_len),
        gc_content = sum(tx_len*gc_content) / sum(tx_len)
      ) %&gt;%
      mutate(
        bin_length = cut(
          x = ave_tx_len,
          labels = seq_len(10),
          breaks = quantile(ave_tx_len, probs = seq(0, 1, length.out = 11)),
          include.lowest = TRUE
        ),
        bin_gc = cut(
          x = gc_content,
          labels = seq_len(10),
          breaks = quantile(gc_content, probs = seq(0, 1, length.out = 11)),
          include.lowest = TRUE
        ),
        bin = paste(bin_gc, bin_length, sep = &quot;_&quot;)
      ),
    by = &quot;gene_id&quot;
  ) %&gt;%
  set_rownames(.$gene_id) %&gt;%
  as(&quot;DataFrame&quot;)</code></pre>
</div>
<div id="read-in-seurat-object" class="section level1">
<h1>Read in seurat object</h1>
<p>The pre-processed <code>seurat</code> object was loaded in.</p>
<pre class="r"><code>fibroblast &lt;- readRDS(here::here(&quot;output/fibroblast_seurat.rds&quot;))</code></pre>
<pre class="r"><code># Color palettes used by most scRNA-seq methods could be found [here](https://rdrr.io/bioc/scater/src/R/plot_colours.R). 
patient_col &lt;- c(
    &quot;MF61&quot; = &quot;#A71B4B&quot;, 
    &quot;MF43&quot; = &quot;#584B9F&quot;
)
# scanpy colors were found in https://github.com/scverse/scanpy/blob/034ca2823804645e0d4874c9b16ba2eb8c13ac0f/scanpy/plotting/palettes.py
treat_col &lt;- c(
  VEH = rgb(0.7, 0.7, 0.7),
  DHT = rgb(0.8, 0.2, 0.2))
sample_col &lt;- hcl.colors(n = 4) %&gt;%
    set_names(c(&quot;mf43DHT&quot;, &quot;mf43VEH&quot;, &quot;mf61DHT&quot;, &quot;mf61VEH&quot;))</code></pre>
<p>There are two different strategies that can be used to perform
differential expression analysis in scRNA-seq data: 1. Using methods
specifically designed for scRNA-seq data, such as <code>MAST</code>. 2.
Pseudo-bulking cells from the same groups together and use methods
designed for bulk RNA-seq data, such as <code>edgeR</code></p>
<p>Both strategies will be applied to this dataset in this analysis.</p>
</div>
<div id="de-using-mast" class="section level1">
<h1>DE using MAST</h1>
<p>Firstly, differential expression analysis was perform using
<code>MAST</code> to compare DHT-treated samples against the vehicle
while nested within patients.</p>
<p><code>sca</code> objects were created using
<code>MAST::FromMatrix()</code> function and the actual linear mixed
hurdle model fittings were done on the DRMCRL workstation.</p>
<p>Note that the continuous part of the hurdle models is conditional of
a gene being expressed, so if a gene had non-zero count in only one Cell
Type + Type group, the models won’t work. The <code>FindMarkers()</code>
function in <code>Seruat</code> solves this by including a
<code>min.pct</code> parameter so only genes that were detected in a
minimum fraction of <code>min.pct</code> cells in either of the two
populations would be tested. <a
href="https://github.com/RGLab/MAST/issues/146">The developer of
<code>MAST</code> package also recommended this strategy</a> so only
genes that had non-zero log-norm counts in at least 10% of vehicle or
DHT-treated cells were tested.</p>
<pre class="r"><code># create SCA object
seurat_data &lt;- GetAssayData(object = fibroblast, slot = &#39;data&#39;)
VEH_cell &lt;- fibroblast[[]] %&gt;%
        dplyr::filter(Treatment == &quot;VEH&quot;) %&gt;%
        rownames()
gene2keep_VEH &lt;- rowSums(seurat_data[, colnames(seurat_data) %in% VEH_cell] &gt; 0) &gt; (0.1 * length(VEH_cell))
DHT_cell &lt;- fibroblast[[]] %&gt;%
        dplyr::filter(Treatment == &quot;DHT&quot;) %&gt;%
        rownames()
gene2keep_DHT &lt;- rowSums(seurat_data[, colnames(seurat_data) %in% DHT_cell] &gt; 0) &gt; (0.1 * length(DHT_cell))
seurat_data &lt;- seurat_data[gene2keep_DHT|gene2keep_VEH,]
# MF43_cell &lt;- fibroblast[[]] %&gt;%
#         dplyr::filter(patient == &quot;MF43&quot;) %&gt;%
#         rownames()
# gene2keep_MF43 &lt;- rowSums(seurat_data[, colnames(seurat_data) %in% MF43_cell] &gt; 0) &gt; (0.1 * length(MF43_cell))
# MF61_cell &lt;- fibroblast[[]] %&gt;%
#         dplyr::filter(Treatment == &quot;MF61&quot;) %&gt;%
#         rownames()
# gene2keep_MF61 &lt;- rowSums(seurat_data[, colnames(seurat_data) %in% MF61_cell] &gt; 0) &gt; (0.1 * length(MF61_cell))
# seurat_data &lt;- seurat_data[gene2keep_MF61|gene2keep_MF43,]
sca &lt;- MAST::FromMatrix(
    exprsArray = as.matrix(seurat_data), 
    cData = fibroblast[[]], 
    fData = data.frame(primerid = rownames(seurat_data)) %&gt;%
            set_rownames(rownames(seurat_data))
)</code></pre>
<p>Cell detection rate was calculated for each gene and added to the
<code>sca</code> object..</p>
<pre class="r"><code>cdr &lt;- colSums(SummarizedExperiment::assay(sca)&gt;0)
SummarizedExperiment::colData(sca)$ngeneson &lt;- scale(cdr)</code></pre>
<p>The reference level of treatment was set to VEH.</p>
<pre class="r"><code>SummarizedExperiment::colData(sca)$Treatment &lt;- factor(SummarizedExperiment::colData(sca)$Treatment, levels = c(&quot;VEH&quot;, &quot;DHT&quot;))
SummarizedExperiment::colData(sca)$patient &lt;- factor(SummarizedExperiment::colData(sca)$patient)
SummarizedExperiment::colData(sca)$patient &lt;- droplevels(SummarizedExperiment::colData(sca)$patient)</code></pre>
<pre class="r"><code># saveRDS(sca, here::here(&quot;data/sca.rds&quot;))</code></pre>
<p>Hurdle model was fitted with the patient as a random effect to nest
the comparisons in patient using linear mixed modelling. (This step was
performed on the DRMCRL workstation).</p>
<pre class="r"><code>zlm_summary_mixed &lt;- readRDS(here::here(&quot;output/zlm_summary_mixed.rds&quot;))$datatable
zlm_summary_mixed &lt;- zlm_summary_mixed %&gt;%
        dplyr::filter(contrast==&#39;TreatmentDHT&#39; &amp; component==&#39;H&#39;) %&gt;%
        dplyr::select(primerid, `Pr(&gt;Chisq)`) %&gt;%
        left_join(
            # logFC
            zlm_summary_mixed %&gt;%
                dplyr::filter(contrast==&#39;TreatmentDHT&#39; &amp; component==&#39;logFC&#39;) %&gt;%
                dplyr::select(primerid, logFC = coef, ci.hi, ci.lo) 
        )  %&gt;%
        mutate(FDR = p.adjust(`Pr(&gt;Chisq)`, &quot;fdr&quot;),
               DE = ifelse(FDR &lt; 0.05, TRUE, FALSE)
               ) %&gt;%
        dplyr::rename(gene_name = primerid)</code></pre>
<p>To my surprise, with the linear mixed model, even with a stringent
cut-off of FDR &lt; 0.01, There were still 4869 genes passing the
significance threshold and being considered differentially
expressed.</p>
<div id="knott-mast-results" class="section level3">
<h3>Knott MAST results</h3>
<p>In the study by <a
href="https://www.cell.com/cell-genomics/pdf/S2666-979X(23)00032-0.pdf">Rath
et al. (2023)</a>, breast tissues from 9 trans-male donors who have been
undertaking gender-affirming therapy (masculinizing dose of androgen)
were compared against 9 cis-females’ breast through snRNA-seq.</p>
<p>The pre-processed <code>Seurat</code> object for the snRNA-seq data
in that study was shared with us by the authors. In the original study,
differential expression analyses were performed for each detected cell
type withouth accounting for the confounding effect of patients.
Therefore, I re-analysed the data using MAST with donor as a random
effect as described above. Only genes that had non-zero log-norm counts
in at least 10% of CF or TF cells were tested. Results derived from
fibroblast cells were loaded.</p>
<pre class="r"><code>knott_mast_de &lt;- readRDS(
    &quot;~/20131906_HickeyT_JC_NormalBreast/output/knott_fibro_mast.rds&quot;
) %&gt;%
    mutate(knott = -sign(coef) * log10(`Pr(&gt;Chisq)`))</code></pre>
<p>783 out of the 4817 tested genes were considered as significantly
differentially expressed with a FDR cut-off of 0.05.</p>
<p>Ranking statistics for genes in both datasets were defined as <span
class="math display">\[-sign(logFC) \times \log_{10} (p{\text
-}value)\]</span>.</p>
<pre class="r"><code>knott_mast_de &lt;- readRDS(
    &quot;~/20131906_HickeyT_JC_NormalBreast/output/knott_fibro_mast.rds&quot;
) %&gt;%
    mutate(knott = -sign(coef) * log10(`Pr(&gt;Chisq)`))</code></pre>
<pre class="r"><code>fibro_2RS &lt;- zlm_summary_mixed %&gt;%
    mutate(pilot = -sign(logFC) * log10(`Pr(&gt;Chisq)`)) %&gt;%
    dplyr::select(gene_name, pilot) %&gt;%
    left_join(
        knott_mast_de%&gt;%
            dplyr::select(gene_name , knott)
    ) %&gt;%
    drop_na()
fibro_2RS_cor &lt;- cor.test(fibro_2RS$pilot, fibro_2RS$knott)</code></pre>
<p>Correlation between the ranking statistics derived from the two
independent studies was found to be 0.024 with a p-value of 0.168.</p>
<pre class="r"><code>pilotRK &lt;- zlm_summary_mixed %&gt;%
    mutate(pilot = -sign(logFC) * log10(`Pr(&gt;Chisq)`))  %&gt;%
    dplyr::filter(DE) %&gt;%
    pull(pilot)
knottRK &lt;- knott_mast_de  %&gt;%
    dplyr::filter(DE) %&gt;%
    pull(knott)</code></pre>
<pre class="r"><code>cor_plot &lt;- fibro_2RS %&gt;%
    mutate(aveRS = (pilot + knott)/2) %&gt;%
    .[order(abs(.$aveRS), decreasing = TRUE),] %&gt;%
    mutate(RANK = row_number()) %&gt;%
    ggplot(
        aes(pilot , knott)
    ) +
    geom_point() +
  geom_vline(
    xintercept = 0,
    linetype = 2,
    size = 1.5,
    colour = &quot;red&quot;
  ) +
    geom_vline(
    xintercept = c(max(knottRK[knottRK &lt; 0]), min(knottRK[knottRK &gt; 0])),
    linetype = 2,
    size = 1.5,
    colour = &quot;blue&quot;
  ) +
  geom_hline(
    yintercept = 0,
    linetype = 2,
    size = 1.5,
    colour = &quot;red&quot;
  ) +
    geom_hline(
    yintercept = c(max(pilotRK[pilotRK &lt; 0]), min(pilotRK[pilotRK &gt; 0])),
    linetype = 2,
    size = 1.5,
    colour = &quot;blue&quot;
  ) +
    geom_text_repel(
        aes(label = gene_name), 
        data = . %&gt;% 
            dplyr::filter(
                RANK &lt; 50
            ), 
        color = &quot;red&quot;, 
        size = 10
    ) +
    labs(
        x = &quot;Ranking Statistic (DHT vs Veh)&quot;, 
        y = &quot;Ranking Statistic (trans-male vs cis-female)&quot;, 
        # title = paste(&quot;Correlation between logFCs between the 2 study:&quot;, 
        #                 round(cor(fibro_2R$pilot, fibro_2FC$knott), digits = 2), 
        #                 sep = &quot; &quot;)
    ) 
cor_plot</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/Differential_Expression.Rmd/unnamed-chunk-11-1.png" alt="*Comparison of ranking statistics derived from the knott fibroblast cells against this pilot study. Genes with top 50 ranked mean absolute ranking statistics were labelled. The blue lines indicate the significance threshold for both studies.*" width="1344" />
<p class="caption">
<em>Comparison of ranking statistics derived from the knott fibroblast
cells against this pilot study. Genes with top 50 ranked mean absolute
ranking statistics were labelled. The blue lines indicate the
significance threshold for both studies.</em>
</p>
</div>
</div>
<div id="ar-signature" class="section level3">
<h3>AR signature</h3>
<pre class="r"><code>AR_sig &lt;- read.csv(here::here(&quot;data/AR_sig_down.csv&quot;)) %&gt;%
    mutate(AR_sig = &quot;Down&quot;) %&gt;%
    rbind(
        read.csv(here::here(&quot;data/AR_sig_up.csv&quot;)) %&gt;%
            mutate(AR_sig = &quot;Up&quot;)
    )
AR_sig_col &lt;- AR_sig %&gt;%
    mutate(
        color = ifelse(AR_sig == &quot;Down&quot;,&#39;blue&#39;, &quot;red&quot;)
    )</code></pre>
<p>A list of 142 AR signatures that were identified in ER-positive
breast cancer cell lines and xenograft models were loaded and the
following ones were considered differentially expressed in DHT-treated
fibroblasts:</p>
<pre class="r"><code>zlm_summary_mixed %&gt;%
    dplyr::filter(
        FDR &lt; 0.01, 
        gene_name %in% AR_sig$gene_name
    )  %&gt;%
    left_join(
        AR_sig 
    ) %&gt;%
    mutate(
        Align = case_when(
           logFC &lt; 0 &amp; AR_sig == &quot;Down&quot; ~ &quot;\u2714 Yes&quot;,
           logFC &gt; 0 &amp; AR_sig == &quot;Up&quot; ~ &quot;\u2714 Yes&quot;,
           logFC &lt; 0 &amp; AR_sig == &quot;Up&quot; ~ &quot;\u2716 No&quot;,
           logFC &gt; 0 &amp; AR_sig == &quot;Down&quot; ~ &quot;\u2716 No&quot;
        )
    ) %&gt;%
    mutate_if(is.character, as.factor) %&gt;%
    mutate_at(vars(c(&quot;Pr(&gt;Chisq)&quot;, &quot;FDR&quot;)), formatP) %&gt;%
    mutate(logFC = round(logFC, 3)) %&gt;%
    dplyr::select(
        `Gene Name` = gene_name,
        `Signature type` = AR_sig,
        logFC,
        PValue = `Pr(&gt;Chisq)`,
        FDR, Align
    ) %&gt;%
    datatable(
        filter = &quot;top&quot;, 
        caption = htmltools::tags$caption(
            htmltools::em(
                &quot;AR signatures that were differentially expressed in DHT-treated fibroblast cells. The align column shows whether directions of changes in this study align with AR signature type&quot;)
        ), 
        options = list(scrollY = &quot;450px&quot;)
    ) %&gt;%
    formatStyle(
        &#39;logFC&#39;,
        color = styleInterval(0, c(&#39;blue&#39;, &quot;red&quot;))
    ) %&gt;%
    formatStyle(
        &#39;Signature type&#39;,
            color = styleEqual(c(&quot;Down&quot;, &quot;Up&quot;), c(&#39;blue&#39;, &quot;red&quot;))
    ) %&gt;%
    formatStyle(
        &#39;Align&#39;,
            color = styleEqual(c( &quot;\u2714 Yes&quot;, &quot;\u2716 No&quot;), c(&quot;#008000&quot;, &quot;#e00000&quot;))
    )</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-b9dde7cce5fada53d320" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-b9dde7cce5fada53d320">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td><\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;ABI3BP&quot;,&quot;C1orf21&quot;,&quot;CKAP2L&quot;,&quot;CXCL12&quot;,&quot;ECHDC2&quot;,&quot;ECI2&quot;,&quot;ELOVL5&quot;,&quot;FAR2&quot;,&quot;GALNT5&quot;,&quot;HMGB2&quot;,&quot;IER3&quot;,&quot;IGFBP4&quot;,&quot;KNL1&quot;,&quot;LMNB1&quot;,&quot;MCCC2&quot;,&quot;ME3&quot;,&quot;PHF19&quot;,&quot;SCP2&quot;,&quot;SEMA3C&quot;,&quot;SMAD6&quot;,&quot;STC2&quot;,&quot;SVIL&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Down&quot;,&quot;Up&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"-0.199\" data-max=\"0.143\" data-scale=\"3\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;✔ Yes&quot;,&quot;✖ No&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n<\/tr>","caption":"<caption>\n  <em>AR signatures that were differentially expressed in DHT-treated fibroblast cells. The align column shows whether directions of changes in this study align with AR signature type<\/em>\n<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22"],["ABI3BP","C1orf21","CKAP2L","CXCL12","ECHDC2","ECI2","ELOVL5","FAR2","GALNT5","HMGB2","IER3","IGFBP4","KNL1","LMNB1","MCCC2","ME3","PHF19","SCP2","SEMA3C","SMAD6","STC2","SVIL"],["Down","Up","Down","Down","Up","Up","Up","Up","Down","Down","Down","Down","Down","Down","Up","Down","Down","Up","Down","Down","Down","Down"],[-0.112,-0.026,0.016,0.095,-0.031,0.052,-0,-0.024,-0.03,0.06,-0.16,0.143,0.028,0.034,-0.012,-0.025,0.024,0.034,-0.199,0.086,-0.05,-0.064],["5.10e-24","0.0028","1.17e-05","2.68e-31","1.35e-08","3.39e-12","0.0017","7.64e-07","0.0009","4.54e-13","3.45e-41","1.72e-26","5.50e-07","6.81e-11","1.62e-05","7.02e-08","8.64e-15","2.83e-08","9.27e-55","7.42e-25","0.0007","6.26e-14"],["6.09e-23","0.0037","2.38e-05","4.41e-30","4.26e-08","1.66e-11","0.0023","1.90e-06","0.0013","2.42e-12","8.50e-40","2.29e-25","1.40e-06","2.89e-10","3.24e-05","2.02e-07","5.48e-14","8.62e-08","3.32e-53","9.19e-24","0.0010","3.62e-13"],["✔ Yes","✖ No","✖ No","✖ No","✖ No","✔ Yes","✖ No","✖ No","✔ Yes","✖ No","✔ Yes","✖ No","✖ No","✖ No","✖ No","✔ Yes","✖ No","✔ Yes","✔ Yes","✖ No","✔ Yes","✔ Yes"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Gene Name<\/th>\n      <th>Signature type<\/th>\n      <th>logFC<\/th>\n      <th>PValue<\/th>\n      <th>FDR<\/th>\n      <th>Align<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollY":"450px","columnDefs":[{"className":"dt-right","targets":3},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'color':isNaN(parseFloat(value)) ? '' : value <= 0 ? \"blue\" : \"red\"});\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'color':value == \"Down\" ? \"blue\" : value == \"Up\" ? \"red\" : null});\nvar value=data[6]; $(this.api().cell(row, 6).node()).css({'color':value == \"✔ Yes\" ? \"#008000\" : value == \"✖ No\" ? \"#e00000\" : null});\n}"},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
<p>Not all of the AR signaturs that were differentially expressed in
DHT-treated cells showed directions of changes that aligned with their
signature types in ER-pos breast cancer.</p>
</div>
<div id="pathway-analysis-using-goseq" class="section level3">
<h3>Pathway analysis using GOseq</h3>
<pre class="r"><code>DE_mast &lt;- zlm_summary_mixed %&gt;%
    dplyr::filter(
        FDR &lt; 0.01
    ) %&gt;%
    left_join(
        genesGR %&gt;%
            as.data.frame() %&gt;% 
            dplyr::select(gene_name, gene_id)
    )</code></pre>
<div id="kegg" class="section level4 tabset">
<h4 class="tabset">KEGG</h4>
<pre class="r"><code>gsTopology &lt;- retrieve_topology(
    database = &quot;kegg&quot;, species = &quot;hsapiens&quot;)
# names(gsTopology) &lt;- str_remove_all(names(gsTopology), &quot;kegg.&quot;)
source(here::here(&quot;analysis/make_gsNetwork.R&quot;))
kg &lt;- get_GSgenelist(gsTopology)
genesGR &lt;- genesGR %&gt;%
    as.data.frame() %&gt;%
    unnest(entrezid)
kg &lt;- kg %&gt;%
    mutate(gene_id = str_remove_all(gene_id, &quot;ENTREZID:&quot;)) %&gt;%
    dplyr::rename(entrezid = gene_id) %&gt;%
    left_join(genesGR %&gt;%
                  dplyr::select(entrezid, gene_name, gene_id) %&gt;%
                  mutate(entrezid = as.character(entrezid)))
kgByGene &lt;- kg  %&gt;%
  split(f = .$gene_id) %&gt;%
  lapply(extract2, &quot;gs_name&quot;)
kgByGs &lt;- kg  %&gt;%
  split(f = .$gs_name) %&gt;%
  lapply(extract2, &quot;entrezid&quot;)
load(system.file(&quot;extdata&quot;, &quot;entrez2name.rda&quot;, package = &quot;sSNAPPY&quot;))</code></pre>
<p>315 KEGG pathways were retrieved from <code>msigdbr</code>.</p>
<p>Enrichment of KEGG pathways among DEGs found with MAST was tested
with <code>goseq</code>, using gene length as an offset.</p>
<pre class="r"><code>lenPwf &lt;-  transGR %&gt;%
    as.data.frame() %&gt;%
    distinct(gene_id, .keep_all = TRUE) %&gt;%
    dplyr::select(gene_id, tx_len) %&gt;%
    mutate(Status = case_when(gene_id %in% DE_mast$gene_id  ~ 1, !gene_id %in% DE_mast$gene_id   ~ 0)) %&gt;%
    with(
      nullp(
        DEgenes = structure(
          Status, names = gene_id
        ),
        genome = &quot;GRCh38.p16&quot;,
        id = &quot;ensGene&quot;,
        bias.data = tx_len,
        plot.fit = FALSE
      )
    )</code></pre>
<pre class="r"><code>goseq_kg &lt;- goseq(lenPwf,
        gene2cat = kgByGene) %&gt;%
  as_tibble %&gt;%
  dplyr::filter(numDEInCat &gt; 0) %&gt;%
  mutate(
    adjP = p.adjust(over_represented_pvalue, method = &quot;bonf&quot;),
    FDR = as.numeric(p.adjust(over_represented_pvalue, method = &quot;fdr&quot;))
  ) %&gt;%
  dplyr::select(-contains(&quot;under&quot;)) %&gt;%
  dplyr::rename(
    gs_name = category,
    PValue = over_represented_pvalue,
    nDE = numDEInCat,
    nExpressed = numInCat
  ) %&gt;%
  left_join(kg) %&gt;%
    dplyr::select(
    gs_name, nExpressed, nDE, 
    contains(&quot;P&quot;, ignore.case = FALSE),
    FDR,
    gene_name, gene_id
  ) %&gt;%
  dplyr::filter(
      gene_name %in% DE_mast$gene_name
  ) %&gt;%
    chop(c(&quot;gene_name&quot;, &quot;gene_id&quot;)) %&gt;%
  mutate(
    gene_name = vapply(.$gene_name, function(x){
      paste(x,collapse = &quot;; &quot;)
    }, character(1)),
    gene_id = vapply(.$gene_id, function(x){
      paste(x,collapse = &quot;; &quot;)
    }, character(1))
  ) %&gt;%
  dplyr::rename(DE = gene_name) %&gt;%
  mutate_at(
    vars(one_of(c(&quot;PValue&quot;, &quot;adjP&quot;, &quot;FDR&quot;))),
    formatP
  )</code></pre>
<p>KEGG pathways that were defined to be significantly enriched (FDR
&lt; 0.05) by <code>goseq</code> were:</p>
<pre class="r"><code>goseq_kg %&gt;%
    dplyr::filter(FDR &lt; 0.05) %&gt;%
    mutate_if(is.character, as.factor) %&gt;%
    datatable(
        filter = &quot;top&quot;, 
        options = list(scrollY = &#39;450px&#39;), 
        caption = htmltools::tags$caption(
            htmltools::em(
                &quot;KEGG pathways that were significantly enriced among differentially expressed genes detected using MAST. The DE column contains pathways genes that were also differentially expressed genes.&quot;)
        )
    ) </code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-ee7a5eb5d707b0a3eb95" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-ee7a5eb5d707b0a3eb95">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td><\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;kegg.2-Oxocarboxylic acid metabolism&quot;,&quot;kegg.Amino sugar and nucleotide sugar metabolism&quot;,&quot;kegg.Apoptosis&quot;,&quot;kegg.Autophagy - animal&quot;,&quot;kegg.Autophagy - other&quot;,&quot;kegg.Bacterial invasion of epithelial cells&quot;,&quot;kegg.Biosynthesis of amino acids&quot;,&quot;kegg.Biosynthesis of nucleotide sugars&quot;,&quot;kegg.Central carbon metabolism in cancer&quot;,&quot;kegg.Drug metabolism - other enzymes&quot;,&quot;kegg.Endocytosis&quot;,&quot;kegg.Epithelial cell signaling in Helicobacter pylori infection&quot;,&quot;kegg.Fatty acid elongation&quot;,&quot;kegg.Fatty acid metabolism&quot;,&quot;kegg.Fluid shear stress and atherosclerosis&quot;,&quot;kegg.Focal adhesion&quot;,&quot;kegg.Glycolysis / Gluconeogenesis&quot;,&quot;kegg.Glyoxylate and dicarboxylate metabolism&quot;,&quot;kegg.HIF-1 signaling pathway&quot;,&quot;kegg.Human cytomegalovirus infection&quot;,&quot;kegg.Human immunodeficiency virus 1 infection&quot;,&quot;kegg.Huntington disease&quot;,&quot;kegg.Lysine degradation&quot;,&quot;kegg.N-Glycan biosynthesis&quot;,&quot;kegg.Neurotrophin signaling pathway&quot;,&quot;kegg.Nucleotide metabolism&quot;,&quot;kegg.Pathogenic Escherichia coli infection&quot;,&quot;kegg.Pentose phosphate pathway&quot;,&quot;kegg.Propanoate metabolism&quot;,&quot;kegg.Protein processing in endoplasmic reticulum&quot;,&quot;kegg.Pyrimidine metabolism&quot;,&quot;kegg.Pyruvate metabolism&quot;,&quot;kegg.Regulation of actin cytoskeleton&quot;,&quot;kegg.Renal cell carcinoma&quot;,&quot;kegg.Shigellosis&quot;,&quot;kegg.SNARE interactions in vesicular transport&quot;,&quot;kegg.Synaptic vesicle cycle&quot;,&quot;kegg.Valine, leucine and isoleucine degradation&quot;,&quot;kegg.Yersinia infection&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"integer\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"16\" data-max=\"341\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"integer\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"11\" data-max=\"111\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;0.0002&quot;,&quot;0.0003&quot;,&quot;0.0005&quot;,&quot;0.0006&quot;,&quot;0.0009&quot;,&quot;0.0010&quot;,&quot;0.0011&quot;,&quot;0.0013&quot;,&quot;0.0014&quot;,&quot;0.0019&quot;,&quot;0.0026&quot;,&quot;0.0029&quot;,&quot;0.0030&quot;,&quot;0.0032&quot;,&quot;0.0037&quot;,&quot;0.0045&quot;,&quot;0.0056&quot;,&quot;0.0057&quot;,&quot;1.06e-05&quot;,&quot;1.09e-05&quot;,&quot;1.23e-05&quot;,&quot;1.35e-05&quot;,&quot;1.46e-05&quot;,&quot;2.26e-05&quot;,&quot;5.84e-05&quot;,&quot;7.23e-05&quot;,&quot;8.55e-06&quot;,&quot;8.87e-06&quot;,&quot;9.36e-06&quot;,&quot;9.97e-06&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;0.0027&quot;,&quot;0.0028&quot;,&quot;0.0029&quot;,&quot;0.0031&quot;,&quot;0.0033&quot;,&quot;0.0034&quot;,&quot;0.0038&quot;,&quot;0.0042&quot;,&quot;0.0046&quot;,&quot;0.0070&quot;,&quot;0.0182&quot;,&quot;0.0226&quot;,&quot;0.0580&quot;,&quot;0.0623&quot;,&quot;0.0747&quot;,&quot;0.0887&quot;,&quot;0.0949&quot;,&quot;0.1046&quot;,&quot;0.1078&quot;,&quot;0.1460&quot;,&quot;0.1554&quot;,&quot;0.1754&quot;,&quot;0.2824&quot;,&quot;0.3248&quot;,&quot;0.3525&quot;,&quot;0.4093&quot;,&quot;0.4231&quot;,&quot;0.5830&quot;,&quot;0.8157&quot;,&quot;0.8226&quot;,&quot;0.8256&quot;,&quot;0.9077&quot;,&quot;0.9472&quot;,&quot;0.9848&quot;,&quot;1.0000&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;0.0002&quot;,&quot;0.0003&quot;,&quot;0.0007&quot;,&quot;0.0008&quot;,&quot;0.0020&quot;,&quot;0.0021&quot;,&quot;0.0024&quot;,&quot;0.0028&quot;,&quot;0.0029&quot;,&quot;0.0031&quot;,&quot;0.0041&quot;,&quot;0.0042&quot;,&quot;0.0046&quot;,&quot;0.0072&quot;,&quot;0.0081&quot;,&quot;0.0086&quot;,&quot;0.0097&quot;,&quot;0.0098&quot;,&quot;0.0132&quot;,&quot;0.0176&quot;,&quot;0.0189&quot;,&quot;0.0193&quot;,&quot;0.0197&quot;,&quot;0.0224&quot;,&quot;0.0267&quot;,&quot;0.0323&quot;,&quot;0.0326&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;AASS; ECHS1; EZH1; SETD2; SETD2; HADHA; KMT5A; KMT2A; KMT2A; ALDH7A1; KMT5B; NSD3; ASH1L; KMT2E; SMYD2; KMT2C; NSD1; NSD1; SMYD3; NSD2; PRDM2; EHMT1; CAMKMT; KMT2D; SETD7; DOT1L; PHYKPL; PLOD2; PLOD3; ACAT1; ACAT1; ACAT2; SETDB1&quot;,&quot;ABI1; WASF3; TAB2; NCKAP1; BRK1; NFKB1; NFKB1; CYFIP1; CYFIP1; CYFIP1; IKBKB; IKBKB; ARPC1B; ARPC1B; ARPC4; ARPC3; ARF1; ARPC5; ARPC2; LPAR1; GNA13; ARHGEF12; F2R; MAPK8; WASF1; GAPDH; GNA12; NCK2; FAS; FAS; RAC1; IL1R1; CDC42; TRAF6; NCK1; MAP3K7; ARPC5L; IRAK4; IRAK4; CYCS; CYCS; ARHGEF1; IKBKG; IKBKG; RIPK1; MAPK9; TNFRSF10B; MAPK10; IRAK1; CYTH3; CYTH1; WIPF2; WIPF1; WIPF1; CASP4; MYO1B; MYO6; MYO10; MYO1D; MYO1E; NFKBIA; NFKBIA; ABCF2; TMBIM6; TUBB6; TUBA1B; TUBA1C; TUBB2A; ACTB; ACTB; ACTG1; CASP3; ROCK1&quot;,&quot;ABI2; CRK; NCKAP1; PTK2; BRK1; CYFIP1; CYFIP1; CYFIP1; AKT3; AKT3; AKT3; FGF7; ITGAV; ITGB8; DIAPH1; GNA13; ITGA11; EGFR; EGFR; FGF5; MRAS; FN1; RRAS2; FGF2; LPAR1; DOCK1; GNA12; ARHGEF7; FGFR1; FGFR1; ITGA6; ITGA1; ROCK1; F2R; RAC1; SOS2; NRAS; NRAS; RRAS; CXCL12; TIAM1; WASF1; VAV2; PDGFRA; PDGFRA; PDGFRB; AKT1; AKT1; PDGFC; ARHGEF12; CDC42; ARHGEF1; DIAPH3; BCAR1; BCAR1; SOS1; SOS1; ACTG1; ACTB; ACTB; PPP1R12C; SSH1; SSH2; PIKFYVE; PIP5K1C; PIP4K2A; VCL; VCL; PIP5K1A; ARHGAP35; MYLK; RAF1; RAF1; MAP2K2; MAP2K2; BRAF; BRAF; ARPC5L; ARPC5; ARPC4; PIK3CA; PIK3CA; PFN1; PIK3R1; MYL9; PFN2; ARPC3; ARPC1B; ARPC1B; ARPC2; IQGAP1; MYL12B; MYL12A; MSN; RDX; TMSB4X; CFL2; CFL2; CFL1; ACTN4; ACTN4; ACTN1; ACTN1; PXN; GSN&quot;,&quot;ACAA2; ECHS1; HACD2; ACSL4; ACAA1; HADHA; ACACA; ACACA; ACADVL; HSD17B12; HACD3; PPT1; PPT1; ELOVL6; ACOX3; HACD1; TECR; CPT1A; SCP2; ACAT1; ACAT1; ACAT2&quot;,&quot;ACAA2; ECHS1; HACD2; HADHA; HSD17B12; HACD3; PPT1; PPT1; ELOVL5; ELOVL1; ELOVL6; HACD1; TECR; ACOT7; THEM4&quot;,&quot;ACAA2; HIBADH; ECHS1; HIBCH; ACAA1; HSD17B10; HSD17B10; HADHA; IVD; ACAT1; ACAT1; ACAT2; MMUT; ALDH7A1; PCCA; AUH; AUH; MCCC1; BCKDHB; MCCC2; MCCC2; MCCC2; AACS&quot;,&quot;AK6; AK6; ADK; ADSL; DUT; TYMP; TYMP; NT5C2; AK5; GUK1; NME7; NT5C; HPRT1; APRT; IMPDH1; IMPDH2; ITPA; NME1; NME2; NME3; NME4; PNP; PNP; HDDC2; NT5C3A; ENPP1; CMPK1; TK1; TK2; UCK2; ASMTL; GMPS; ENTPD6; ENTPD4&quot;,&quot;AKR1A1; ADH5; ENO1; ENO1; GAPDH; GPI; GPI; GPI; LDHA; LDHA; LDHB; ALDH7A1; PCK2; PCK2; PDHA1; PDHB; PFKL; PFKP; PGAM1; PGK1; PGM1; PKM; PGM2; TPI1; ADPGK; MINPP1&quot;,&quot;AKR1A1; ME3; ADH5; FH; FH; GLO1; HAGH; ACACA; ACACA; ACAT1; ACAT1; ACAT2; LDHA; LDHA; LDHB; MDH1; MDH2; ALDH7A1; PCK2; PCK2; PDHA1; PDHB; PKM; ACYP2; GRHPR&quot;,&quot;AKT3; AKT3; AKT3; EGFR; EGFR; MAP2K2; MAP2K2; RAF1; RAF1; AKT1; AKT1; PDGFRA; PDGFRA; PDGFRB; FGFR1; FGFR1; MET; MET; PIK3CA; PIK3CA; PIK3R1; NRAS; NRAS; PDK1; SCO2; SLC7A5; LDHA; LDHA; PGAM1; PKM; LDHB; SLC2A1; SLC2A1; PFKP; SLC1A5; GLS; PFKL; PDHA1; PDHB; G6PD; G6PD&quot;,&quot;AP2M1; CLTA; SH3KBP1; AP2S1; CLTB; EPN1; EPN2; AP2A1; GRK5; SH3GL1; AP2A2; AP2A2; AP2A2; EPS15; VTA1; DAB2; EPS15L1; RAB11FIP2; CBLB; CBLB; VPS45; VPS45; MVB12A; MVB12B; TGFBR2; TGFBR2; RNF41; TRAF6; WWP1; NEDD4; SMURF2; SMURF1; SMURF1; ITCH; ITCH; IGF1R; RUFY1; RUFY1; EHD2; EGFR; EGFR; PDGFRA; PDGFRA; STAM2; STAM; SMAD2; CAV1; CAV2&quot;,&quot;ARPC5; ARPC4; ARPC3; ARPC1B; ARPC1B; ARPC2; CRK; GAB1; DOCK1; FN1; RHOG; PXN; RAC1; PIK3CA; PIK3CA; PIK3R1; PTK2; ARHGAP10; ARPC5L; WASF1; VCL; VCL; CDC42; BCAR1; BCAR1; MET; MET; ACTB; ACTB; ACTG1; DNM3; CTNNA1&quot;,&quot;ATF6; ERO1A; EIF2S1; EIF2AK1; NFE2L2; CAPN2; CAPN1; EIF2AK3; SEC31A; UBXN2A; NPLOC4; UBXN1; NSFL1C; UBXN6; SAR1B; SAR1A; PDIA4; PDIA6; ERP29; TXNDC5; MAPK9; MAPK10; MAPK8; VCP; VCP; SEC24D; SEC23B; SEC23B; SEC24A; SEC24B; UFD1&quot;,&quot;CLTA; CLTB; STX2; VAMP2; AP2M1; AP2S1; AP2A1; AP2A2; AP2A2; AP2A2; RIMS1&quot;,&quot;CREB3; AP2A1; ADRM1; AP2M1; AP2S1; CLTA; CLTB; CREB1; AP2A2; AP2A2; AP2A2; HDAC1; HDAC2; ATG14; HIP1; RCOR1; IFT57; SIN3A; NRBF2; PIK3R4; CREB3L2; SEM1; ULK2; ATG101; ATG13; CYCS; CYCS; GNAQ; AMBRA1; APAF1; ATG2B; PLCB4; MAPK9; MAPK8; MAPK10; CREB5; REST; CASP3; WIPI1; WIPI2; MT-CYB; UQCR10; SDHA; UQCRB; KIF5B; CYC1; KLC1; UQCRQ; UQCRC2; UQCRH; SDHB; SDHB; SDHC; SDHC; UQCRC1; UQCRFS1; SOD2; NRF1; GPX8; SOD1; SOD1&quot;,&quot;CREBBP; ELOB; ELOC; AKT3; AKT3; AKT3; EGLN2; INSR; IGF1R; AKT1; AKT1; PIK3CA; PIK3CA; EGFR; EGFR; MKNK2; PIK3R1; IFNGR2; IFNGR2; IFNGR2; MAP2K2; MAP2K2; EGLN1; NFKB1; NFKB1; STAT3; STAT3; PLCG2; PLCG2; PDK1; RBX1; EIF4E2; PFKL; VEGFA; PFKP; PGK1; LTBR; SLC2A1; SLC2A1; ANGPT1; GAPDH; ANGPT2; HMOX1; LDHA; LDHA; LDHB; ENO1; ENO1; TEK; PDHA1; PDHB&quot;,&quot;CRK; TAB2; DOCK1; IKBKB; IKBKB; IRAK1; ARPC1B; ARPC1B; MAP2K2; MAP2K2; ARPC5; ARPC3; ARPC4; ARHGEF12; PXN; MAP2K3; WIPF2; GNAQ; FN1; IRAK4; IRAK4; MAPK8; MAPK9; MAPK10; TBK1; TBK1; MAP3K7; NFKBIA; NFKBIA; ARPC2; VAV2; PTK2; RHOG; RAC1; ARHGEF1; MAP2K4; BCAR1; BCAR1; TRAF6; IKBKG; IKBKG; WIPF1; WIPF1; ARHGEF7; ARPC5L; GIT2; CDC42; NFATC3; IRF3; NFKB1; NFKB1; AKT3; AKT3; AKT3; AKT1; AKT1; PIK3CA; PIK3CA; PIK3R1; PKN1; RPS6KA2; ACTB; ACTB; ACTG1; PIP5K1C; PIP5K1A; ROCK1; SKAP2&quot;,&quot;CS; ENO1; ENO1; GAPDH; GOT2; PSAT1; IDH1; IDH1; IDH2; IDH2; IDH3A; IDH3B; IDH3G; ASL; MTR; ACO1; ACO2; PFKL; PFKP; PGAM1; PGK1; PKM; SHMT2; TKT; GLUL; TPI1; PRPS1; PRPS1; MAT2B; MAT2A; TALDO1&quot;,&quot;CS; GCSH; GCSH; PGP; MDH1; MDH2; ACO1; ACO2; PCCA; SHMT2; HYI; CAT; GRHPR; ACAT1; ACAT1; ACAT2; MMUT&quot;,&quot;CS; GOT2; IDH1; IDH1; IDH2; IDH2; IDH3A; IDH3B; IDH3G; ACO1; ACO2&quot;,&quot;CYBA; CYBA; HSP90AB1; SUMO2; SUMO3; HSP90B1; CAV2; RAC1; CAV1; GPC1; NFE2L2; SDC1; PIK3R1; ITGAV; MAPK8; PTK2; MAP2K4; AKT1; AKT1; AKT3; AKT3; AKT3; BMPR1B; ACVR2A; BMPR2; BMPR2; IKBKB; IKBKB; IL1R1; MAP3K7; SDC4; MAPK9; MAPK10; IKBKG; IKBKG; PIK3CA; PIK3CA; BMPR1A; BMPR1A; MAP2K5; KEAP1; MEF2A; NFKB1; NFKB1; SUMO1; ACTB; ACTB; ACTG1; SQSTM1; SQSTM1; GSTO1; PLAT; PLAT; CTSL; GSTP1; GSTP1; TXN; HMOX1; MGST1; GSTM3; MGST3; VEGFA; NQO1; TXN2; MMP2&quot;,&quot;DAD1; STT3B; DPM3; RPN1; RPN2; DPM1; ALG3; MGAT4B; MGAT4B; FUT8; B4GALT1; MAN2A1; MGAT1; MGAT2; MGAT5; MOGS; MOGS; ALG13; TUSC3; ALG2; B4GALT2; MAN1A2; MAN1B1; ALG8; MAN1A1&quot;,&quot;DPYD; DPYD; DUT; TYMP; TYMP; GUSB; NME7; HPRT1; IMPDH1; IMPDH2; ITPA; NME1; NME2; NME3; NME4; CMPK1; TK1; TK2; TPMT; UCK2; UMPS; GMPS; GSTM3; GSTP1; GSTP1; MGST1; MGST3; GSTO1&quot;,&quot;DPYD; DPYD; DUT; TYMP; TYMP; NT5C2; NME7; NT5C; NUDT2; NME1; NME2; NME3; NME4; HDDC2; NT5C3A; ENPP1; CMPK1; TK1; TK2; UCK2; UMPS; ASMTL; ENTPD6; ENTPD4&quot;,&quot;ECHS1; HIBCH; HADHA; ACACA; ACACA; LDHA; LDHA; LDHB; MMUT; PCCA; BCKDHB; ACOX3; SUCLG2; SUCLG1; SUCLA2&quot;,&quot;EGLN2; CRK; GAB1; RAPGEF1; RAC1; MET; MET; EGLN1; MAP2K2; MAP2K2; BRAF; BRAF; CDC42; RAF1; RAF1; CREBBP; EPAS1; PIK3CA; PIK3CA; PIK3R1; AKT3; AKT3; AKT3; AKT1; AKT1; SOS1; SOS1; SOS2; RAP1A; ETS1; SLC2A1; SLC2A1; VEGFA&quot;,&quot;ELOC; TAB2; TRAF5; ELOB; IRAK1; PDIA3; AKT3; AKT3; AKT3; PPP3CB; GNB2; NFKB1; NFKB1; TBK1; TBK1; NRAS; NRAS; AKT1; AKT1; RAC1; PIK3CA; PIK3CA; PIK3R1; GNA11; FAS; FAS; MAP3K7; GNAQ; MAPK8; GNG5; GNG10; GNG11; CYCS; CYCS; TRAF6; IRAK4; IRAK4; RNF7; MAPK9; PPP3CC; PPP3R1; MAP2K3; MAP2K2; MAP2K2; RAF1; RAF1; MAPK10; PTK2; RIPK1; DCAF1; TAPBP; TAPBP; TAPBP; TAPBP; TAPBP; TAPBP; HLA-A; HLA-A; HLA-A; HLA-A; HLA-A; HLA-A; HLA-A; HLA-A; HLA-B; HLA-B; HLA-B; HLA-B; HLA-B; HLA-B; HLA-E; HLA-E; HLA-E; HLA-E; HLA-E; HLA-E; HLA-E; IRF3; PLCG2; PLCG2; RBX1; NFATC3; IKBKB; IKBKB; IKBKG; IKBKG; CFL2; CFL2; CASP3; CFL1; APOBEC3C; AP1G1; AP1S1; B2M; B2M; B2M; AP1B1; PXN; CRK; ITPR2&quot;,&quot;FAS; FAS; MAPK9; APAF1; CTSC; CTSC; MAPK10; CTSL; AKT1; AKT1; RAF1; RAF1; CAPN1; AKT3; AKT3; AKT3; TNFRSF10B; MAP2K2; MAP2K2; MAP3K14; MAP3K14; MAP3K14; CYCS; CYCS; NGF; NGF; NRAS; NRAS; EIF2S1; MAPK8; CTSB; CTSB; CTSD; CAPN2; IKBKB; IKBKB; EIF2AK3; IKBKG; IKBKG; NFKBIA; NFKBIA; NFKB1; NFKB1; PIK3CA; PIK3CA; PIK3R1; CASP3; MCL1; CFLAR; RIPK1; PTPN13; GADD45B; PARP4; PARP1; ACTG1; DFFA; TUBA1C; LMNB2; ACTB; ACTB; TUBA1B; LMNB1&quot;,&quot;G6PD; G6PD; PGLS; GPI; GPI; GPI; DERA; PFKL; PFKP; PGD; PGM1; PGM2; PRPS1; PRPS1; TALDO1; TKT; H6PD&quot;,&quot;GABARAP; GABARAPL2; ULK2; ATG12; ATG101; ATG7; IGBP1; ATG10; ATG3; ATG4C; ATG5; ATG13&quot;,&quot;GALK1; GALK2; GFPT1; GFPT1; GMDS; GPI; GPI; GPI; GMPPB; GMPPA; PGM1; PGM3; PGM3; PMM1; PMM2; NANS; PGM2; NAGK; GNPNAT1; UGDH; GFPT2; UXS1&quot;,&quot;GNPDA2; GALK1; GALK2; GFPT1; GFPT1; GMDS; GPI; GPI; GPI; GMPPB; GMPPA; HEXA; HEXB; PGM1; PGM3; PGM3; PMM1; PMM2; NANS; PGM2; NAGK; GNPNAT1; UGDH; UXS1; GFPT2&quot;,&quot;IGF1R; ATG12; TANK; DEPTOR; ATG14; CAMKK2; ATG3; WIPI2; RRAS2; WIPI1; IRS1; HMGB1; NRBF2; ATG7; NRAS; NRAS; WDR41; PRKACB; IRS2; SUPT20H; DDIT4; MRAS; MAP2K2; MAP2K2; IGBP1; EIF2AK3; AKT1S1; TBK1; TBK1; AMBRA1; ULK2; ATG4C; RAF1; RAF1; ATG13; RRAS; MAP3K7; ATG5; TRAF6; PIK3R4; VMP1; ATG10; SNAP29; ATG2B; PIK3CA; PIK3CA; PIK3R1; MTMR3; AKT3; AKT3; AKT3; AKT1; AKT1; MAPK8; MAPK9; MAPK10; BNIP3; CFLAR; GABARAP; GABARAPL2; MAP1LC3B; MAP1LC3A; SQSTM1; SQSTM1; EIF2S1; ATG101&quot;,&quot;IKBKB; IKBKB; ADAM10; MAPK10; ADAM17; ADAM17; MAPK8; MAP3K14; MAP3K14; MAP3K14; MAPK9; MAP2K4; JAM3; NFKBIA; NFKBIA; NFKB1; NFKB1; NOD1; IKBKG; IKBKG; EGFR; EGFR; TJP1; TJP1&quot;,&quot;IRAK1; IRAK4; IRAK4; IRAK3; RAPGEF1; NRAS; NRAS; IRS1; AKT3; AKT3; AKT3; MAP2K5; PLCG2; PLCG2; RAC1; NGF; NGF; PIK3CA; PIK3CA; AKT1; AKT1; NTF3; SOS2; RAP1A; MAPK8; IKBKB; IKBKB; MAP3K1; MAPK9; ARHGDIA; MAP2K2; MAP2K2; RAF1; RAF1; PIK3R1; SOS1; SOS1; BRAF; BRAF; TRAF6; RPS6KA2; CDC42; MAPK10; GAB1; SH2B3; SH2B3; KIDINS220; BEX3; FRS2; CRK; NFKBIA; NFKBIA; FOXO3; PRDM4; RIPK2; MAGED1; NFKB1; NFKB1&quot;,&quot;ITGA6; ITGA11; ITGAV; ITGA1; ITGB8; COL4A1; COL6A1; COL6A1; MYL12B; MYL9; COL4A2; COL6A3; COL6A3; COL6A2; COL6A2; COL1A1; COL1A1; COL1A2; COL1A2; COMP; CRK; DOCK1; MYL12A; MET; MET; MYLK; RAP1A; FN1; FYN; LAMB1; RAPGEF1; LAMB2; LAMA2; LAMA2; TNC; EGFR; EGFR; PTK2; VEGFA; LAMC1; THBS2; PDGFRA; PDGFRA; PDGFRB; IGF1R; VAV2; LAMA4; LAMA4; CDC42; RAC1; VEGFB; VEGFC; BCAR1; BCAR1; TNXB; TNXB; TNXB; TNXB; TNXB; TNXB; TNXB; TNXB; SOS1; SOS1; PDGFC; CAV2; SOS2; CAV1; FLNC; FLNC; PARVB; TLN2; FLNB; PARVA; VCL; VCL; EMP2; PXN; ACTN4; ACTN4; ACTN1; ACTN1; VASP; ZYX; ZYX; TLN1; PPP1R12C; DIAPH1; PIP5K1C; PIK3CA; PIK3CA; PIK3R1; PIP5K1A; AKT3; AKT3; AKT3; ROCK1; CAPN2; ARHGAP35; AKT1; AKT1; RAF1; RAF1; MAPK8; MAPK9; BRAF; BRAF; MAPK10; ACTB; ACTB; ACTG1&quot;,&quot;PTK2; ITGAV; CRK; TRAF5; PXN; PDIA3; GNB2; GNA13; EGFR; EGFR; AKT3; AKT3; AKT3; TBK1; TBK1; GNA11; GNA12; AKAP13; ADCY7; GNG10; ADCY9; GNG11; GNAQ; GNAS; PIK3R1; AKT1; AKT1; ARHGEF12; GNG5; PRKACB; PDGFRA; PDGFRA; FAS; FAS; IKBKB; IKBKB; IL10RB; IL10RB; NRAS; NRAS; RAF1; RAF1; PPP3CC; PPP3R1; CYCS; CYCS; IL1R1; PPP3CB; PIK3CA; PIK3CA; MAP2K2; MAP2K2; RIPK1; SOS2; SOS1; SOS1; ROCK1; ARHGEF1; CXCL12; IKBKG; IKBKG; TAPBP; TAPBP; TAPBP; TAPBP; TAPBP; TAPBP; HLA-E; HLA-E; HLA-E; HLA-E; HLA-E; HLA-E; HLA-E; HLA-B; HLA-B; HLA-B; HLA-B; HLA-B; HLA-B; HLA-A; HLA-A; HLA-A; HLA-A; HLA-A; HLA-A; HLA-A; HLA-A; NFKBIA; NFKBIA; IRF3; NFKB1; NFKB1; STAT3; STAT3; NFATC3; PLCB4; RAC1; BCAR1; BCAR1; CREB3; CREB1; CREB3L2; CREB5; CASP3; B2M; B2M; B2M; VEGFA; ITPR2&quot;,&quot;STX6; VTI1B; STX2; STX18; VTI1A; STX4; STX7; SEC22B; BET1; YKT6; VAMP2; VAMP3; STX16; STX8; GOSR2&quot;,&quot;UBE2N; RBCK1; DOCK1; TAB2; CRK; TRAF5; CAPN1; NOD1; NFKB1; NFKB1; VDAC1; IKBKB; IKBKB; TLN1; ATG12; MALT1; MALT1; VCL; VCL; UBE2V1; CAPN2; ARPC5; ARPC4; ARPC3; MYL12B; AKT3; AKT3; AKT3; ARPC2; EGFR; EGFR; ARPC1B; ARPC1B; AKT1; AKT1; MAPK10; MYL9; ARPC5L; ROCK1; PTK2; TLN2; ACTN4; ACTN4; DIAPH1; ARF1; MYL12A; RIPK2; IL1R1; PXN; RAC1; WASF1; TRAF6; UBE2V2; CYTH1; MAP3K7; TBK1; TBK1; CASP4; ATG5; IKBKG; IKBKG; MAPK8; MAPK9; RIPK1; BCAR1; BCAR1; RBX1; ACTN1; ACTN1; BNIP3; CYTH3; SQSTM1; SQSTM1; WIPI1; CDC42; MAP1LC3B; MAP1LC3A; WIPI2; NFKBIA; NFKBIA; CBX3; IRF3; PIK3R1; PLCE1; PLCG2; PLCG2; PLCB4; FNBP1; ITPR2; FNBP1L; PIK3CA; PIK3CA; CAPNS1; ACTG1; ACTB; ACTB; AKT1S1; CYCS; CYCS; PPID; SEPTIN2; SEPTIN7; PFN2; PFN1; FOXO1; FOXO3&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;ENSG00000005339; ENSG00000103363; ENSG00000154582; ENSG00000117020; LRG_1396; ENSG00000275199; ENSG00000269858; ENSG00000171105; ENSG00000140443; ENSG00000142208; LRG_721; ENSG00000121879; LRG_310; ENSG00000146648; LRG_304; ENSG00000099875; ENSG00000145675; LRG_67; ENSG00000159128; ENSG00000262795; ENSG00000126934; LRG_750; ENSG00000135766; ENSG00000109320; LRG_1316; ENSG00000168610; LRG_112; ENSG00000197943; LRG_376; ENSG00000152256; ENSG00000100387; ENSG00000135930; ENSG00000141959; ENSG00000112715; ENSG00000067057; ENSG00000102144; ENSG00000111321; ENSG00000117394; LRG_1132; ENSG00000154188; ENSG00000111640; ENSG00000091879; ENSG00000100292; ENSG00000134333; ENSG00000288299; ENSG00000111716; ENSG00000074800; LRG_1165; ENSG00000120156; ENSG00000131828; ENSG00000168291&quot;,&quot;ENSG00000008311; ENSG00000127884; ENSG00000108799; LRG_775; ENSG00000181555; ENSG00000084754; ENSG00000183955; ENSG00000118058; LRG_613; ENSG00000164904; ENSG00000110066; ENSG00000147548; ENSG00000116539; ENSG00000005483; ENSG00000143499; ENSG00000055609; ENSG00000165671; LRG_512; ENSG00000185420; ENSG00000109685; ENSG00000116731; ENSG00000181090; ENSG00000143919; ENSG00000167548; ENSG00000145391; ENSG00000104885; ENSG00000175309; ENSG00000152952; ENSG00000106397; ENSG00000075239; LRG_1400; ENSG00000120437; ENSG00000143379&quot;,&quot;ENSG00000026103; LRG_134; ENSG00000050748; ENSG00000120868; ENSG00000109861; LRG_50; ENSG00000109339; ENSG00000135047; ENSG00000142208; LRG_721; ENSG00000132155; LRG_413; ENSG00000014216; ENSG00000117020; LRG_1396; ENSG00000275199; ENSG00000120889; ENSG00000126934; LRG_750; ENSG00000006062; LRG_1222; ENSG00000282637; LRG_876; ENSG00000172115; LRG_260; ENSG00000134259; LRG_92; ENSG00000213281; ENSG00000134001; ENSG00000107643; ENSG00000164733; ENSG00000285132; ENSG00000117984; ENSG00000162909; ENSG00000104365; LRG_1241; ENSG00000172071; ENSG00000269335; LRG_70; LRG_89; ENSG00000100906; ENSG00000109320; LRG_1316; ENSG00000121879; LRG_310; ENSG00000145675; ENSG00000164305; ENSG00000143384; ENSG00000003402; ENSG00000137275; ENSG00000163629; ENSG00000099860; ENSG00000102699; ENSG00000143799; ENSG00000184009; ENSG00000160049; ENSG00000167553; ENSG00000176619; ENSG00000075624; LRG_132; ENSG00000123416; ENSG00000113368&quot;,&quot;ENSG00000051523; LRG_52; ENSG00000096384; ENSG00000188612; ENSG00000184900; ENSG00000166598; ENSG00000105971; ENSG00000136238; ENSG00000105974; ENSG00000063660; ENSG00000116044; ENSG00000115884; ENSG00000145675; ENSG00000138448; ENSG00000107643; ENSG00000169398; ENSG00000065559; ENSG00000142208; LRG_721; ENSG00000117020; LRG_1396; ENSG00000275199; ENSG00000138696; ENSG00000121989; ENSG00000204217; LRG_712; ENSG00000104365; LRG_1241; ENSG00000115594; ENSG00000135341; ENSG00000124145; ENSG00000050748; ENSG00000109339; ENSG00000269335; LRG_70; ENSG00000121879; LRG_310; ENSG00000107779; LRG_298; ENSG00000137764; ENSG00000079999; ENSG00000068305; ENSG00000109320; LRG_1316; ENSG00000116030; ENSG00000075624; LRG_132; ENSG00000184009; ENSG00000161011; ENSG00000284099; ENSG00000148834; ENSG00000104368; LRG_570; ENSG00000135047; LRG_723; ENSG00000084207; ENSG00000136810; ENSG00000100292; ENSG00000008394; ENSG00000134202; ENSG00000143198; ENSG00000112715; ENSG00000181019; ENSG00000100348; ENSG00000087245&quot;,&quot;ENSG00000062485; ENSG00000074800; LRG_1165; ENSG00000111640; ENSG00000125166; ENSG00000135069; LRG_610; ENSG00000138413; ENSG00000182054; LRG_611; ENSG00000166411; ENSG00000101365; ENSG00000067829; ENSG00000126522; ENSG00000116984; ENSG00000122729; ENSG00000100412; ENSG00000141959; ENSG00000067057; ENSG00000171314; ENSG00000102144; ENSG00000067225; ENSG00000182199; ENSG00000163931; ENSG00000135821; ENSG00000111669; LRG_264; ENSG00000147224; ENSG00000038274; ENSG00000168906; ENSG00000177156&quot;,&quot;ENSG00000062485; ENSG00000125166; LRG_610; ENSG00000138413; ENSG00000182054; LRG_611; ENSG00000166411; ENSG00000101365; ENSG00000067829; ENSG00000122729; ENSG00000100412&quot;,&quot;ENSG00000062485; ENSG00000140905; LRG_541; ENSG00000184207; ENSG00000014641; ENSG00000146701; ENSG00000122729; ENSG00000100412; ENSG00000175198; ENSG00000182199; ENSG00000178922; ENSG00000121691; ENSG00000137106; ENSG00000075239; LRG_1400; ENSG00000120437; ENSG00000146085&quot;,&quot;ENSG00000085231; ENSG00000279247; ENSG00000156110; ENSG00000239900; ENSG00000128951; ENSG00000025708; LRG_727; ENSG00000076685; ENSG00000154027; ENSG00000143774; ENSG00000143156; ENSG00000125458; ENSG00000165704; ENSG00000198931; ENSG00000106348; ENSG00000178035; ENSG00000125877; ENSG00000239672; ENSG00000243678; ENSG00000103024; ENSG00000103202; ENSG00000198805; LRG_91; ENSG00000111906; ENSG00000122643; ENSG00000197594; ENSG00000162368; ENSG00000167900; ENSG00000166548; ENSG00000143179; ENSG00000169093; ENSG00000163655; ENSG00000197586; ENSG00000197217&quot;,&quot;ENSG00000091409; ENSG00000137809; ENSG00000138448; ENSG00000213949; ENSG00000105855; ENSG00000187498; LRG_475; ENSG00000142156; ENSG00000118680; ENSG00000101335; ENSG00000134871; ENSG00000163359; LRG_473; ENSG00000142173; LRG_476; LRG_1; ENSG00000108821; LRG_2; ENSG00000164692; ENSG00000105664; ENSG00000167193; ENSG00000150760; ENSG00000101608; ENSG00000105976; LRG_662; ENSG00000065534; ENSG00000116473; ENSG00000115414; ENSG00000010810; ENSG00000091136; ENSG00000107263; ENSG00000172037; ENSG00000196569; LRG_409; ENSG00000041982; ENSG00000146648; LRG_304; ENSG00000169398; ENSG00000112715; ENSG00000135862; ENSG00000186340; LRG_309; ENSG00000134853; ENSG00000113721; ENSG00000140443; ENSG00000160293; ENSG00000112769; LRG_433; ENSG00000070831; ENSG00000136238; ENSG00000173511; ENSG00000150630; ENSG00000050820; ENSG00000285460; ENSG00000168477; ENSG00000229341; ENSG00000233323; ENSG00000231608; ENSG00000236221; ENSG00000206258; ENSG00000229353; ENSG00000236236; ENSG00000115904; LRG_754; ENSG00000145431; ENSG00000105971; ENSG00000100485; ENSG00000105974; ENSG00000128591; LRG_870; ENSG00000188677; ENSG00000171914; ENSG00000136068; ENSG00000197702; ENSG00000035403; LRG_383; ENSG00000213853; ENSG00000089159; ENSG00000130402; ENSG00000282844; LRG_886; ENSG00000072110; ENSG00000125753; ENSG00000159840; ENSG00000285443; ENSG00000137076; ENSG00000125503; ENSG00000131504; ENSG00000186111; ENSG00000121879; LRG_310; ENSG00000145675; ENSG00000143398; ENSG00000117020; LRG_1396; ENSG00000275199; ENSG00000067900; ENSG00000162909; ENSG00000160007; ENSG00000142208; LRG_721; ENSG00000132155; LRG_413; ENSG00000107643; ENSG00000050748; ENSG00000157764; LRG_299; ENSG00000109339; ENSG00000075624; LRG_132; ENSG00000184009&quot;,&quot;ENSG00000104365; LRG_1241; ENSG00000137845; ENSG00000109339; ENSG00000151694; LRG_1203; ENSG00000107643; ENSG00000006062; LRG_1222; ENSG00000282637; ENSG00000050748; ENSG00000065559; ENSG00000166086; LRG_89; ENSG00000100906; ENSG00000109320; LRG_1316; ENSG00000106100; ENSG00000269335; LRG_70; ENSG00000146648; LRG_304; ENSG00000104067; ENSG00000277401&quot;,&quot;ENSG00000107175; ENSG00000196961; ENSG00000130706; ENSG00000161203; ENSG00000042753; ENSG00000122705; ENSG00000175416; ENSG00000118260; ENSG00000183020; ENSG00000280759; ENSG00000281385; ENSG00000116478; ENSG00000196591; ENSG00000126775; ENSG00000127946; ENSG00000089902; ENSG00000114446; ENSG00000169375; ENSG00000148572; ENSG00000196455; ENSG00000182158; ENSG00000127922; ENSG00000083290; ENSG00000123395; ENSG00000175224; LRG_876; ENSG00000172115; ENSG00000156052; ENSG00000110497; ENSG00000120868; ENSG00000066739; ENSG00000101333; ENSG00000050748; ENSG00000107643; ENSG00000109339; ENSG00000146592; ENSG00000084093; ENSG00000164305; ENSG00000070540; ENSG00000157954; ENSG00000198727; ENSG00000184076; ENSG00000073578; ENSG00000156467; ENSG00000170759; ENSG00000179091; ENSG00000126214; ENSG00000164405; ENSG00000140740; ENSG00000173660; ENSG00000117118; LRG_316; LRG_317; ENSG00000143252; ENSG00000010256; ENSG00000169021; ENSG00000112096; ENSG00000106459; ENSG00000164294; LRG_652; ENSG00000142168&quot;,&quot;ENSG00000108479; ENSG00000156958; LRG_787; ENSG00000198380; ENSG00000112699; ENSG00000105220; LRG_1178; ENSG00000282019; ENSG00000173540; ENSG00000144591; ENSG00000079739; ENSG00000013375; LRG_1312; ENSG00000100417; ENSG00000140650; ENSG00000095380; ENSG00000169299; ENSG00000124357; ENSG00000100522; ENSG00000109814; ENSG00000131459; ENSG00000115652&quot;,&quot;ENSG00000117020; LRG_1396; ENSG00000275199; ENSG00000146648; LRG_304; ENSG00000126934; LRG_750; ENSG00000132155; LRG_413; ENSG00000142208; LRG_721; LRG_309; ENSG00000134853; ENSG00000113721; ENSG00000077782; LRG_993; ENSG00000105976; LRG_662; ENSG00000121879; LRG_310; ENSG00000145675; LRG_92; ENSG00000213281; ENSG00000152256; ENSG00000284194; ENSG00000103257; ENSG00000134333; ENSG00000288299; ENSG00000171314; ENSG00000067225; ENSG00000111716; ENSG00000117394; LRG_1132; ENSG00000067057; ENSG00000105281; ENSG00000115419; ENSG00000141959; ENSG00000131828; ENSG00000168291; ENSG00000160211; LRG_148&quot;,&quot;ENSG00000117448; ENSG00000151376; ENSG00000197894; LRG_504; ENSG00000091483; ENSG00000124767; ENSG00000063854; ENSG00000278540; ENSG00000275176; ENSG00000075239; LRG_1400; ENSG00000120437; ENSG00000134333; ENSG00000288299; ENSG00000111716; ENSG00000014641; ENSG00000146701; ENSG00000164904; ENSG00000100889; ENSG00000285241; ENSG00000131828; ENSG00000168291; ENSG00000067225; ENSG00000170634; ENSG00000137106&quot;,&quot;ENSG00000117448; ENSG00000197894; ENSG00000074800; LRG_1165; ENSG00000111640; ENSG00000105220; LRG_1178; ENSG00000282019; ENSG00000134333; ENSG00000288299; ENSG00000111716; ENSG00000164904; ENSG00000100889; ENSG00000285241; ENSG00000131828; ENSG00000168291; ENSG00000141959; ENSG00000067057; ENSG00000171314; ENSG00000102144; ENSG00000079739; ENSG00000067225; ENSG00000169299; ENSG00000111669; ENSG00000159322; ENSG00000107789&quot;,&quot;ENSG00000118217; ENSG00000197930; ENSG00000134001; ENSG00000086232; ENSG00000116044; ENSG00000162909; ENSG00000014216; ENSG00000172071; ENSG00000138674; ENSG00000173960; ENSG00000182446; ENSG00000162191; ENSG00000088833; ENSG00000167671; ENSG00000152700; ENSG00000079332; ENSG00000155660; ENSG00000143870; ENSG00000089248; ENSG00000239264; ENSG00000050748; ENSG00000109339; ENSG00000107643; ENSG00000165280; LRG_657; ENSG00000150961; ENSG00000101310; LRG_1134; ENSG00000113615; ENSG00000138802; ENSG00000070010&quot;,&quot;ENSG00000122705; ENSG00000175416; ENSG00000111450; ENSG00000220205; ENSG00000161203; ENSG00000042753; ENSG00000196961; ENSG00000183020; ENSG00000280759; ENSG00000281385; ENSG00000079841&quot;,&quot;ENSG00000127884; ENSG00000198130; ENSG00000084754; ENSG00000278540; ENSG00000275176; ENSG00000134333; ENSG00000288299; ENSG00000111716; ENSG00000146085; ENSG00000175198; ENSG00000083123; ENSG00000087008; ENSG00000172340; ENSG00000163541; ENSG00000136143&quot;,&quot;ENSG00000129562; ENSG00000163527; ENSG00000179085; ENSG00000163902; ENSG00000118705; ENSG00000000419; ENSG00000214160; ENSG00000161013; ENSG00000284501; ENSG00000033170; ENSG00000086062; ENSG00000112893; ENSG00000131446; ENSG00000168282; ENSG00000152127; ENSG00000115275; LRG_1226; ENSG00000101901; ENSG00000104723; ENSG00000119523; ENSG00000117411; ENSG00000198162; ENSG00000177239; ENSG00000159063; ENSG00000111885&quot;,&quot;ENSG00000135823; ENSG00000100568; ENSG00000111450; ENSG00000168818; ENSG00000151532; ENSG00000103496; ENSG00000079950; ENSG00000265808; ENSG00000105829; ENSG00000106636; ENSG00000220205; ENSG00000049245; ENSG00000124222; ENSG00000170310; ENSG00000108433&quot;,&quot;ENSG00000136754; ENSG00000132970; ENSG00000055208; ENSG00000061676; ENSG00000254999; ENSG00000109320; LRG_1316; ENSG00000273749; ENSG00000288461; ENSG00000280618; ENSG00000104365; LRG_1241; ENSG00000130429; LRG_1188; ENSG00000241553; ENSG00000111229; ENSG00000143761; ENSG00000162704; ENSG00000163466; ENSG00000198121; ENSG00000120063; ENSG00000196914; ENSG00000181104; ENSG00000107643; ENSG00000112290; ENSG00000111640; ENSG00000146535; ENSG00000071051; ENSG00000026103; LRG_134; ENSG00000136238; ENSG00000115594; ENSG00000070831; ENSG00000175104; ENSG00000158092; ENSG00000135341; ENSG00000136950; ENSG00000198001; LRG_75; LRG_876; ENSG00000172115; ENSG00000076928; ENSG00000269335; LRG_70; ENSG00000137275; ENSG00000050748; ENSG00000120889; ENSG00000109339; ENSG00000184216; ENSG00000008256; ENSG00000108669; ENSG00000171475; ENSG00000115935; LRG_374; ENSG00000196954; ENSG00000128641; ENSG00000196586; ENSG00000145555; ENSG00000176658; ENSG00000157483; LRG_89; ENSG00000100906; ENSG00000033050; ENSG00000139644; ENSG00000176014; ENSG00000123416; ENSG00000167553; ENSG00000137267; ENSG00000075624; LRG_132; ENSG00000184009; ENSG00000164305; ENSG00000067900&quot;,&quot;ENSG00000138443; ENSG00000167193; ENSG00000061676; ENSG00000169398; ENSG00000254999; ENSG00000273749; ENSG00000288461; ENSG00000280618; ENSG00000117020; LRG_1396; ENSG00000275199; ENSG00000140285; ENSG00000138448; ENSG00000105855; ENSG00000131504; ENSG00000120063; ENSG00000137809; ENSG00000146648; LRG_304; ENSG00000138675; ENSG00000158186; ENSG00000115414; ENSG00000133818; ENSG00000138685; ENSG00000198121; ENSG00000150760; ENSG00000146535; ENSG00000102606; ENSG00000077782; LRG_993; ENSG00000091409; ENSG00000213949; ENSG00000067900; ENSG00000181104; ENSG00000136238; ENSG00000100485; LRG_92; ENSG00000213281; ENSG00000126458; ENSG00000107562; ENSG00000156299; ENSG00000112290; ENSG00000160293; LRG_309; ENSG00000134853; ENSG00000113721; ENSG00000142208; LRG_721; ENSG00000145431; ENSG00000196914; ENSG00000070831; ENSG00000076928; ENSG00000139734; ENSG00000050820; ENSG00000285460; ENSG00000115904; LRG_754; ENSG00000184009; ENSG00000075624; LRG_132; ENSG00000125503; ENSG00000084112; ENSG00000141298; ENSG00000115020; ENSG00000186111; ENSG00000150867; ENSG00000035403; LRG_383; ENSG00000143398; ENSG00000160007; ENSG00000065534; ENSG00000132155; LRG_413; ENSG00000126934; LRG_750; ENSG00000157764; LRG_299; ENSG00000136950; ENSG00000162704; ENSG00000241553; ENSG00000121879; LRG_310; ENSG00000108518; ENSG00000145675; ENSG00000101335; ENSG00000070087; ENSG00000111229; ENSG00000130429; LRG_1188; ENSG00000163466; ENSG00000140575; ENSG00000118680; ENSG00000101608; ENSG00000147065; ENSG00000137710; ENSG00000205542; ENSG00000165410; LRG_213; ENSG00000172757; ENSG00000130402; ENSG00000282844; LRG_886; ENSG00000072110; ENSG00000089159; ENSG00000148180&quot;,&quot;ENSG00000140443; ENSG00000145782; ENSG00000136560; ENSG00000155792; ENSG00000126775; ENSG00000110931; ENSG00000144848; ENSG00000157954; ENSG00000133818; ENSG00000070540; ENSG00000169047; ENSG00000189403; ENSG00000148572; ENSG00000197548; LRG_92; ENSG00000213281; ENSG00000164253; ENSG00000142875; ENSG00000185950; ENSG00000102710; ENSG00000168209; ENSG00000158186; ENSG00000126934; LRG_750; ENSG00000089289; ENSG00000172071; ENSG00000204673; ENSG00000183735; LRG_1306; ENSG00000110497; ENSG00000083290; ENSG00000125703; ENSG00000132155; LRG_413; ENSG00000175224; ENSG00000126458; ENSG00000135341; ENSG00000057663; ENSG00000175104; ENSG00000196455; ENSG00000062716; ENSG00000152348; ENSG00000099940; ENSG00000066739; ENSG00000121879; LRG_310; ENSG00000145675; ENSG00000100330; ENSG00000117020; LRG_1396; ENSG00000275199; ENSG00000142208; LRG_721; ENSG00000107643; ENSG00000050748; ENSG00000109339; ENSG00000176171; ENSG00000003402; ENSG00000170296; ENSG00000034713; ENSG00000140941; ENSG00000101460; ENSG00000161011; ENSG00000284099; ENSG00000134001; ENSG00000123395&quot;,&quot;ENSG00000154582; ENSG00000055208; ENSG00000082512; ENSG00000103363; ENSG00000184216; ENSG00000167004; ENSG00000117020; LRG_1396; ENSG00000275199; ENSG00000107758; ENSG00000172354; ENSG00000109320; LRG_1316; ENSG00000183735; LRG_1306; LRG_92; ENSG00000213281; ENSG00000142208; LRG_721; ENSG00000136238; ENSG00000121879; LRG_310; ENSG00000145675; ENSG00000088256; ENSG00000026103; LRG_134; ENSG00000135341; ENSG00000156052; ENSG00000107643; ENSG00000174021; ENSG00000242616; ENSG00000127920; LRG_876; ENSG00000172115; ENSG00000175104; ENSG00000198001; LRG_75; ENSG00000114125; ENSG00000050748; ENSG00000120910; ENSG00000221823; ENSG00000034152; ENSG00000126934; LRG_750; ENSG00000132155; LRG_413; ENSG00000109339; ENSG00000169398; ENSG00000137275; ENSG00000145041; ENSG00000231925; LRG_114; ENSG00000112493; ENSG00000236490; ENSG00000206208; ENSG00000206281; ENSG00000206503; ENSG00000229215; ENSG00000224320; ENSG00000235657; ENSG00000231834; ENSG00000227715; ENSG00000206505; ENSG00000223980; ENSG00000234745; ENSG00000223532; ENSG00000224608; ENSG00000232126; ENSG00000206450; ENSG00000228964; ENSG00000204592; ENSG00000233904; ENSG00000229252; ENSG00000236632; ENSG00000225201; ENSG00000206493; ENSG00000230254; ENSG00000126456; ENSG00000197943; LRG_376; ENSG00000100387; ENSG00000072736; ENSG00000104365; LRG_1241; ENSG00000269335; LRG_70; ENSG00000165410; LRG_213; ENSG00000164305; ENSG00000172757; ENSG00000244509; ENSG00000166747; ENSG00000106367; ENSG00000166710; LRG_1215; ENSG00000273686; ENSG00000100280; ENSG00000089159; ENSG00000167193; ENSG00000123104&quot;,&quot;ENSG00000160211; LRG_148; ENSG00000130313; ENSG00000105220; LRG_1178; ENSG00000282019; ENSG00000023697; ENSG00000141959; ENSG00000067057; ENSG00000142657; ENSG00000079739; ENSG00000169299; LRG_264; ENSG00000147224; ENSG00000177156; ENSG00000163931; ENSG00000049239&quot;,&quot;ENSG00000161203; ENSG00000122705; ENSG00000147010; ENSG00000042753; ENSG00000175416; ENSG00000063245; ENSG00000072134; ENSG00000196961; ENSG00000198873; ENSG00000141985; ENSG00000183020; ENSG00000280759; ENSG00000281385; ENSG00000085832; ENSG00000009844; ENSG00000153071; ENSG00000127527; ENSG00000107560; ENSG00000114423; LRG_1179; ENSG00000136631; LRG_1170; ENSG00000141971; ENSG00000196814; LRG_779; ENSG00000163513; ENSG00000181852; ENSG00000175104; ENSG00000123124; ENSG00000069869; ENSG00000108854; ENSG00000198742; ENSG00000284126; ENSG00000078747; LRG_354; ENSG00000140443; ENSG00000176783; ENSG00000284260; ENSG00000024422; ENSG00000146648; LRG_304; LRG_309; ENSG00000134853; ENSG00000115145; ENSG00000136738; ENSG00000175387; ENSG00000105974; ENSG00000105971&quot;,&quot;ENSG00000162704; ENSG00000241553; ENSG00000111229; ENSG00000130429; LRG_1188; ENSG00000163466; ENSG00000167193; ENSG00000109458; ENSG00000150760; ENSG00000115414; ENSG00000177105; ENSG00000089159; ENSG00000136238; ENSG00000121879; LRG_310; ENSG00000145675; ENSG00000169398; ENSG00000071205; ENSG00000136950; ENSG00000112290; ENSG00000035403; LRG_383; ENSG00000070831; ENSG00000050820; ENSG00000285460; ENSG00000105976; LRG_662; ENSG00000075624; LRG_132; ENSG00000184009; ENSG00000197959; ENSG00000044115&quot;,&quot;ENSG00000163281; ENSG00000108479; ENSG00000156958; LRG_787; ENSG00000198380; ENSG00000112699; ENSG00000105220; LRG_1178; ENSG00000282019; ENSG00000173540; ENSG00000144591; ENSG00000213614; ENSG00000049860; ENSG00000079739; ENSG00000013375; LRG_1312; ENSG00000100417; ENSG00000140650; ENSG00000095380; ENSG00000169299; ENSG00000124357; ENSG00000100522; ENSG00000109814; ENSG00000115652; ENSG00000131459&quot;,&quot;ENSG00000167193; ENSG00000055208; ENSG00000150760; ENSG00000104365; LRG_1241; ENSG00000184216; ENSG00000130429; LRG_1188; ENSG00000126934; LRG_750; ENSG00000162704; ENSG00000111229; ENSG00000241553; ENSG00000196914; ENSG00000089159; ENSG00000034152; ENSG00000171475; ENSG00000156052; ENSG00000115414; ENSG00000198001; LRG_75; ENSG00000107643; ENSG00000050748; ENSG00000109339; ENSG00000183735; LRG_1306; ENSG00000135341; LRG_89; ENSG00000100906; ENSG00000163466; ENSG00000160293; ENSG00000169398; ENSG00000177105; ENSG00000136238; ENSG00000076928; ENSG00000065559; ENSG00000050820; ENSG00000285460; ENSG00000175104; ENSG00000269335; LRG_70; ENSG00000115935; LRG_374; ENSG00000102606; ENSG00000136950; ENSG00000139436; ENSG00000070831; ENSG00000072736; ENSG00000126456; ENSG00000109320; LRG_1316; ENSG00000117020; LRG_1396; ENSG00000275199; ENSG00000142208; LRG_721; ENSG00000121879; LRG_310; ENSG00000145675; ENSG00000123143; ENSG00000071242; ENSG00000075624; LRG_132; ENSG00000184009; ENSG00000186111; ENSG00000143398; ENSG00000067900; ENSG00000005020&quot;,&quot;ENSG00000167315; ENSG00000106049; ENSG00000127884; ENSG00000198130; ENSG00000060971; ENSG00000072506; LRG_450; ENSG00000084754; ENSG00000128928; ENSG00000075239; LRG_1400; ENSG00000120437; ENSG00000146085; ENSG00000164904; ENSG00000175198; ENSG00000148090; LRG_449; ENSG00000078070; ENSG00000083123; ENSG00000131844; ENSG00000281742; ENSG00000275300; ENSG00000081760&quot;,&quot;ENSG00000167315; ENSG00000127884; ENSG00000206527; ENSG00000068366; ENSG00000060971; ENSG00000084754; ENSG00000278540; ENSG00000275176; ENSG00000072778; ENSG00000149084; ENSG00000074696; ENSG00000131238; LRG_690; ENSG00000170522; ENSG00000087008; ENSG00000165996; ENSG00000099797; ENSG00000110090; ENSG00000116171; ENSG00000075239; LRG_1400; ENSG00000120437&quot;,&quot;ENSG00000167315; ENSG00000127884; ENSG00000206527; ENSG00000084754; ENSG00000149084; ENSG00000074696; ENSG00000131238; LRG_690; ENSG00000012660; ENSG00000066322; ENSG00000170522; ENSG00000165996; ENSG00000099797; ENSG00000097021; ENSG00000159445&quot;,&quot;ENSG00000169398; ENSG00000138448; ENSG00000167193; ENSG00000082512; ENSG00000089159; ENSG00000167004; ENSG00000172354; ENSG00000120063; ENSG00000146648; LRG_304; ENSG00000117020; LRG_1396; ENSG00000275199; ENSG00000183735; LRG_1306; ENSG00000088256; ENSG00000146535; ENSG00000170776; ENSG00000121281; ENSG00000242616; ENSG00000162104; ENSG00000127920; ENSG00000156052; ENSG00000087460; ENSG00000145675; ENSG00000142208; LRG_721; ENSG00000196914; ENSG00000174021; ENSG00000142875; LRG_309; ENSG00000134853; ENSG00000026103; LRG_134; ENSG00000104365; LRG_1241; ENSG00000243646; LRG_152; LRG_92; ENSG00000213281; ENSG00000132155; LRG_413; ENSG00000120910; ENSG00000221823; LRG_876; ENSG00000172115; ENSG00000115594; ENSG00000107758; ENSG00000121879; LRG_310; ENSG00000126934; LRG_750; ENSG00000137275; ENSG00000100485; ENSG00000115904; LRG_754; ENSG00000067900; ENSG00000076928; ENSG00000107562; ENSG00000269335; LRG_70; ENSG00000231925; LRG_114; ENSG00000112493; ENSG00000236490; ENSG00000206208; ENSG00000206281; ENSG00000204592; ENSG00000233904; ENSG00000229252; ENSG00000236632; ENSG00000225201; ENSG00000206493; ENSG00000230254; ENSG00000234745; ENSG00000223532; ENSG00000224608; ENSG00000232126; ENSG00000206450; ENSG00000228964; ENSG00000206503; ENSG00000229215; ENSG00000224320; ENSG00000235657; ENSG00000231834; ENSG00000227715; ENSG00000206505; ENSG00000223980; LRG_89; ENSG00000100906; ENSG00000126456; ENSG00000109320; LRG_1316; ENSG00000168610; LRG_112; ENSG00000072736; ENSG00000101333; ENSG00000136238; ENSG00000050820; ENSG00000285460; ENSG00000107175; ENSG00000118260; ENSG00000182158; ENSG00000146592; ENSG00000164305; ENSG00000166710; LRG_1215; ENSG00000273686; ENSG00000112715; ENSG00000123104&quot;,&quot;ENSG00000170296; ENSG00000034713; ENSG00000083290; ENSG00000145782; ENSG00000123395; ENSG00000197548; ENSG00000089289; ENSG00000152348; ENSG00000144848; ENSG00000125703; ENSG00000057663; ENSG00000175224&quot;,&quot;ENSG00000177889; ENSG00000125826; ENSG00000150760; ENSG00000055208; ENSG00000167193; ENSG00000082512; ENSG00000014216; ENSG00000106100; ENSG00000109320; LRG_1316; ENSG00000213585; ENSG00000104365; LRG_1241; ENSG00000137076; ENSG00000145782; ENSG00000172175; LRG_1221; ENSG00000035403; LRG_383; ENSG00000244687; ENSG00000162909; ENSG00000162704; ENSG00000241553; ENSG00000111229; ENSG00000118680; ENSG00000117020; LRG_1396; ENSG00000275199; ENSG00000163466; ENSG00000146648; LRG_304; ENSG00000130429; LRG_1188; ENSG00000142208; LRG_721; ENSG00000109339; ENSG00000101335; ENSG00000136950; ENSG00000067900; ENSG00000169398; ENSG00000171914; ENSG00000130402; ENSG00000282844; ENSG00000131504; ENSG00000143761; ENSG00000101608; ENSG00000104312; ENSG00000115594; ENSG00000089159; ENSG00000136238; ENSG00000112290; ENSG00000175104; ENSG00000169139; ENSG00000108669; ENSG00000135341; ENSG00000183735; LRG_1306; ENSG00000196954; ENSG00000057663; ENSG00000269335; LRG_70; ENSG00000107643; ENSG00000050748; ENSG00000137275; ENSG00000050820; ENSG00000285460; ENSG00000100387; LRG_886; ENSG00000072110; ENSG00000176171; ENSG00000008256; ENSG00000161011; ENSG00000284099; ENSG00000070540; ENSG00000070831; ENSG00000140941; ENSG00000101460; ENSG00000157954; LRG_89; ENSG00000100906; ENSG00000122565; ENSG00000126456; ENSG00000145675; ENSG00000138193; ENSG00000197943; LRG_376; ENSG00000101333; ENSG00000187239; ENSG00000123104; ENSG00000137942; ENSG00000121879; LRG_310; ENSG00000126247; ENSG00000184009; ENSG00000075624; LRG_132; ENSG00000204673; LRG_876; ENSG00000172115; ENSG00000171497; ENSG00000168385; ENSG00000122545; ENSG00000070087; ENSG00000108518; ENSG00000150907; ENSG00000118689&quot;,&quot;ENSG00000184216; ENSG00000198001; LRG_75; ENSG00000090376; ENSG00000107263; LRG_92; ENSG00000213281; ENSG00000169047; ENSG00000117020; LRG_1396; ENSG00000275199; ENSG00000137764; ENSG00000197943; LRG_376; ENSG00000136238; LRG_260; ENSG00000134259; ENSG00000121879; LRG_310; ENSG00000142208; LRG_721; ENSG00000185652; ENSG00000100485; ENSG00000116473; ENSG00000107643; ENSG00000104365; LRG_1241; ENSG00000095015; ENSG00000050748; ENSG00000141522; ENSG00000126934; LRG_750; ENSG00000132155; LRG_413; ENSG00000145675; ENSG00000115904; LRG_754; ENSG00000157764; LRG_299; ENSG00000175104; ENSG00000071242; ENSG00000070831; ENSG00000109339; ENSG00000109458; ENSG00000111252; LRG_621; ENSG00000134313; ENSG00000166681; ENSG00000166225; ENSG00000167193; LRG_89; ENSG00000100906; ENSG00000118689; ENSG00000110851; ENSG00000104312; ENSG00000179222; ENSG00000109320; LRG_1316&quot;,&quot;ENSG00000188641; LRG_722; ENSG00000128951; ENSG00000025708; LRG_727; ENSG00000076685; ENSG00000143156; ENSG00000125458; ENSG00000164978; ENSG00000239672; ENSG00000243678; ENSG00000103024; ENSG00000103202; ENSG00000111906; ENSG00000122643; ENSG00000197594; ENSG00000162368; ENSG00000167900; ENSG00000166548; ENSG00000143179; ENSG00000114491; ENSG00000169093; ENSG00000197586; ENSG00000197217&quot;,&quot;ENSG00000188641; LRG_722; ENSG00000128951; ENSG00000025708; LRG_727; ENSG00000169919; ENSG00000143156; ENSG00000165704; ENSG00000106348; ENSG00000178035; ENSG00000125877; ENSG00000239672; ENSG00000243678; ENSG00000103024; ENSG00000103202; ENSG00000162368; ENSG00000167900; ENSG00000166548; ENSG00000137364; ENSG00000143179; ENSG00000114491; ENSG00000163655; ENSG00000134202; LRG_723; ENSG00000084207; ENSG00000008394; ENSG00000143198; ENSG00000148834&quot;,&quot;ENSG00000269858; ENSG00000167193; ENSG00000109458; ENSG00000107263; ENSG00000136238; ENSG00000105976; LRG_662; ENSG00000135766; ENSG00000126934; LRG_750; ENSG00000157764; LRG_299; ENSG00000070831; ENSG00000132155; LRG_413; ENSG00000005339; ENSG00000116016; ENSG00000121879; LRG_310; ENSG00000145675; ENSG00000117020; LRG_1396; ENSG00000275199; ENSG00000142208; LRG_721; ENSG00000115904; LRG_754; ENSG00000100485; ENSG00000116473; ENSG00000134954; ENSG00000117394; LRG_1132; ENSG00000112715&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n<\/tr>","caption":"<caption>\n  <em>KEGG pathways that were significantly enriced among differentially expressed genes detected using MAST. The DE column contains pathways genes that were also differentially expressed genes.<\/em>\n<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39"],["kegg.Huntington disease","kegg.Glyoxylate and dicarboxylate metabolism","kegg.Pentose phosphate pathway","kegg.Shigellosis","kegg.Fluid shear stress and atherosclerosis","kegg.Biosynthesis of nucleotide sugars","kegg.Pyruvate metabolism","kegg.Amino sugar and nucleotide sugar metabolism","kegg.2-Oxocarboxylic acid metabolism","kegg.Biosynthesis of amino acids","kegg.Protein processing in endoplasmic reticulum","kegg.Focal adhesion","kegg.Human immunodeficiency virus 1 infection","kegg.SNARE interactions in vesicular transport","kegg.Autophagy - animal","kegg.Yersinia infection","kegg.Pathogenic Escherichia coli infection","kegg.Nucleotide metabolism","kegg.Valine, leucine and isoleucine degradation","kegg.Human cytomegalovirus infection","kegg.Lysine degradation","kegg.Autophagy - other","kegg.Glycolysis / Gluconeogenesis","kegg.Fatty acid elongation","kegg.Pyrimidine metabolism","kegg.Synaptic vesicle cycle","kegg.N-Glycan biosynthesis","kegg.Apoptosis","kegg.Fatty acid metabolism","kegg.Epithelial cell signaling in Helicobacter pylori infection","kegg.Drug metabolism - other enzymes","kegg.Regulation of actin cytoskeleton","kegg.Neurotrophin signaling pathway","kegg.HIF-1 signaling pathway","kegg.Endocytosis","kegg.Renal cell carcinoma","kegg.Bacterial invasion of epithelial cells","kegg.Central carbon metabolism in cancer","kegg.Propanoate metabolism"],[159,33,34,296,172,41,54,50,16,79,60,257,317,25,161,176,198,95,57,341,70,24,79,35,63,20,56,180,57,51,92,282,148,137,121,72,71,98,34],[61,17,17,106,65,22,25,25,11,31,31,111,100,15,66,68,73,34,23,110,33,12,26,15,24,11,25,62,22,24,28,105,58,51,48,33,32,41,15],["8.55e-06","8.87e-06","9.36e-06","9.97e-06","1.06e-05","1.09e-05","1.23e-05","1.35e-05","1.46e-05","2.26e-05","5.84e-05","7.23e-05","0.0002","0.0002","0.0002","0.0003","0.0003","0.0003","0.0003","0.0005","0.0005","0.0006","0.0009","0.0010","0.0011","0.0013","0.0014","0.0019","0.0026","0.0026","0.0026","0.0029","0.0030","0.0032","0.0037","0.0045","0.0045","0.0056","0.0057"],["0.0027","0.0028","0.0029","0.0031","0.0033","0.0034","0.0038","0.0042","0.0046","0.0070","0.0182","0.0226","0.0580","0.0623","0.0747","0.0887","0.0949","0.1046","0.1078","0.1460","0.1554","0.1754","0.2824","0.3248","0.3525","0.4093","0.4231","0.5830","0.8157","0.8226","0.8256","0.9077","0.9472","0.9848","1.0000","1.0000","1.0000","1.0000","1.0000"],["0.0002","0.0002","0.0002","0.0002","0.0002","0.0002","0.0002","0.0002","0.0002","0.0003","0.0007","0.0008","0.0020","0.0021","0.0024","0.0028","0.0029","0.0031","0.0031","0.0041","0.0042","0.0046","0.0072","0.0081","0.0086","0.0097","0.0098","0.0132","0.0176","0.0176","0.0176","0.0189","0.0193","0.0197","0.0224","0.0267","0.0267","0.0323","0.0326"],["CREB3; AP2A1; ADRM1; AP2M1; AP2S1; CLTA; CLTB; CREB1; AP2A2; AP2A2; AP2A2; HDAC1; HDAC2; ATG14; HIP1; RCOR1; IFT57; SIN3A; NRBF2; PIK3R4; CREB3L2; SEM1; ULK2; ATG101; ATG13; CYCS; CYCS; GNAQ; AMBRA1; APAF1; ATG2B; PLCB4; MAPK9; MAPK8; MAPK10; CREB5; REST; CASP3; WIPI1; WIPI2; MT-CYB; UQCR10; SDHA; UQCRB; KIF5B; CYC1; KLC1; UQCRQ; UQCRC2; UQCRH; SDHB; SDHB; SDHC; SDHC; UQCRC1; UQCRFS1; SOD2; NRF1; GPX8; SOD1; SOD1","CS; GCSH; GCSH; PGP; MDH1; MDH2; ACO1; ACO2; PCCA; SHMT2; HYI; CAT; GRHPR; ACAT1; ACAT1; ACAT2; MMUT","G6PD; G6PD; PGLS; GPI; GPI; GPI; DERA; PFKL; PFKP; PGD; PGM1; PGM2; PRPS1; PRPS1; TALDO1; TKT; H6PD","UBE2N; RBCK1; DOCK1; TAB2; CRK; TRAF5; CAPN1; NOD1; NFKB1; NFKB1; VDAC1; IKBKB; IKBKB; TLN1; ATG12; MALT1; MALT1; VCL; VCL; UBE2V1; CAPN2; ARPC5; ARPC4; ARPC3; MYL12B; AKT3; AKT3; AKT3; ARPC2; EGFR; EGFR; ARPC1B; ARPC1B; AKT1; AKT1; MAPK10; MYL9; ARPC5L; ROCK1; PTK2; TLN2; ACTN4; ACTN4; DIAPH1; ARF1; MYL12A; RIPK2; IL1R1; PXN; RAC1; WASF1; TRAF6; UBE2V2; CYTH1; MAP3K7; TBK1; TBK1; CASP4; ATG5; IKBKG; IKBKG; MAPK8; MAPK9; RIPK1; BCAR1; BCAR1; RBX1; ACTN1; ACTN1; BNIP3; CYTH3; SQSTM1; SQSTM1; WIPI1; CDC42; MAP1LC3B; MAP1LC3A; WIPI2; NFKBIA; NFKBIA; CBX3; IRF3; PIK3R1; PLCE1; PLCG2; PLCG2; PLCB4; FNBP1; ITPR2; FNBP1L; PIK3CA; PIK3CA; CAPNS1; ACTG1; ACTB; ACTB; AKT1S1; CYCS; CYCS; PPID; SEPTIN2; SEPTIN7; PFN2; PFN1; FOXO1; FOXO3","CYBA; CYBA; HSP90AB1; SUMO2; SUMO3; HSP90B1; CAV2; RAC1; CAV1; GPC1; NFE2L2; SDC1; PIK3R1; ITGAV; MAPK8; PTK2; MAP2K4; AKT1; AKT1; AKT3; AKT3; AKT3; BMPR1B; ACVR2A; BMPR2; BMPR2; IKBKB; IKBKB; IL1R1; MAP3K7; SDC4; MAPK9; MAPK10; IKBKG; IKBKG; PIK3CA; PIK3CA; BMPR1A; BMPR1A; MAP2K5; KEAP1; MEF2A; NFKB1; NFKB1; SUMO1; ACTB; ACTB; ACTG1; SQSTM1; SQSTM1; GSTO1; PLAT; PLAT; CTSL; GSTP1; GSTP1; TXN; HMOX1; MGST1; GSTM3; MGST3; VEGFA; NQO1; TXN2; MMP2","GALK1; GALK2; GFPT1; GFPT1; GMDS; GPI; GPI; GPI; GMPPB; GMPPA; PGM1; PGM3; PGM3; PMM1; PMM2; NANS; PGM2; NAGK; GNPNAT1; UGDH; GFPT2; UXS1","AKR1A1; ME3; ADH5; FH; FH; GLO1; HAGH; ACACA; ACACA; ACAT1; ACAT1; ACAT2; LDHA; LDHA; LDHB; MDH1; MDH2; ALDH7A1; PCK2; PCK2; PDHA1; PDHB; PKM; ACYP2; GRHPR","GNPDA2; GALK1; GALK2; GFPT1; GFPT1; GMDS; GPI; GPI; GPI; GMPPB; GMPPA; HEXA; HEXB; PGM1; PGM3; PGM3; PMM1; PMM2; NANS; PGM2; NAGK; GNPNAT1; UGDH; UXS1; GFPT2","CS; GOT2; IDH1; IDH1; IDH2; IDH2; IDH3A; IDH3B; IDH3G; ACO1; ACO2","CS; ENO1; ENO1; GAPDH; GOT2; PSAT1; IDH1; IDH1; IDH2; IDH2; IDH3A; IDH3B; IDH3G; ASL; MTR; ACO1; ACO2; PFKL; PFKP; PGAM1; PGK1; PKM; SHMT2; TKT; GLUL; TPI1; PRPS1; PRPS1; MAT2B; MAT2A; TALDO1","ATF6; ERO1A; EIF2S1; EIF2AK1; NFE2L2; CAPN2; CAPN1; EIF2AK3; SEC31A; UBXN2A; NPLOC4; UBXN1; NSFL1C; UBXN6; SAR1B; SAR1A; PDIA4; PDIA6; ERP29; TXNDC5; MAPK9; MAPK10; MAPK8; VCP; VCP; SEC24D; SEC23B; SEC23B; SEC24A; SEC24B; UFD1","ITGA6; ITGA11; ITGAV; ITGA1; ITGB8; COL4A1; COL6A1; COL6A1; MYL12B; MYL9; COL4A2; COL6A3; COL6A3; COL6A2; COL6A2; COL1A1; COL1A1; COL1A2; COL1A2; COMP; CRK; DOCK1; MYL12A; MET; MET; MYLK; RAP1A; FN1; FYN; LAMB1; RAPGEF1; LAMB2; LAMA2; LAMA2; TNC; EGFR; EGFR; PTK2; VEGFA; LAMC1; THBS2; PDGFRA; PDGFRA; PDGFRB; IGF1R; VAV2; LAMA4; LAMA4; CDC42; RAC1; VEGFB; VEGFC; BCAR1; BCAR1; TNXB; TNXB; TNXB; TNXB; TNXB; TNXB; TNXB; TNXB; SOS1; SOS1; PDGFC; CAV2; SOS2; CAV1; FLNC; FLNC; PARVB; TLN2; FLNB; PARVA; VCL; VCL; EMP2; PXN; ACTN4; ACTN4; ACTN1; ACTN1; VASP; ZYX; ZYX; TLN1; PPP1R12C; DIAPH1; PIP5K1C; PIK3CA; PIK3CA; PIK3R1; PIP5K1A; AKT3; AKT3; AKT3; ROCK1; CAPN2; ARHGAP35; AKT1; AKT1; RAF1; RAF1; MAPK8; MAPK9; BRAF; BRAF; MAPK10; ACTB; ACTB; ACTG1","ELOC; TAB2; TRAF5; ELOB; IRAK1; PDIA3; AKT3; AKT3; AKT3; PPP3CB; GNB2; NFKB1; NFKB1; TBK1; TBK1; NRAS; NRAS; AKT1; AKT1; RAC1; PIK3CA; PIK3CA; PIK3R1; GNA11; FAS; FAS; MAP3K7; GNAQ; MAPK8; GNG5; GNG10; GNG11; CYCS; CYCS; TRAF6; IRAK4; IRAK4; RNF7; MAPK9; PPP3CC; PPP3R1; MAP2K3; MAP2K2; MAP2K2; RAF1; RAF1; MAPK10; PTK2; RIPK1; DCAF1; TAPBP; TAPBP; TAPBP; TAPBP; TAPBP; TAPBP; HLA-A; HLA-A; HLA-A; HLA-A; HLA-A; HLA-A; HLA-A; HLA-A; HLA-B; HLA-B; HLA-B; HLA-B; HLA-B; HLA-B; HLA-E; HLA-E; HLA-E; HLA-E; HLA-E; HLA-E; HLA-E; IRF3; PLCG2; PLCG2; RBX1; NFATC3; IKBKB; IKBKB; IKBKG; IKBKG; CFL2; CFL2; CASP3; CFL1; APOBEC3C; AP1G1; AP1S1; B2M; B2M; B2M; AP1B1; PXN; CRK; ITPR2","STX6; VTI1B; STX2; STX18; VTI1A; STX4; STX7; SEC22B; BET1; YKT6; VAMP2; VAMP3; STX16; STX8; GOSR2","IGF1R; ATG12; TANK; DEPTOR; ATG14; CAMKK2; ATG3; WIPI2; RRAS2; WIPI1; IRS1; HMGB1; NRBF2; ATG7; NRAS; NRAS; WDR41; PRKACB; IRS2; SUPT20H; DDIT4; MRAS; MAP2K2; MAP2K2; IGBP1; EIF2AK3; AKT1S1; TBK1; TBK1; AMBRA1; ULK2; ATG4C; RAF1; RAF1; ATG13; RRAS; MAP3K7; ATG5; TRAF6; PIK3R4; VMP1; ATG10; SNAP29; ATG2B; PIK3CA; PIK3CA; PIK3R1; MTMR3; AKT3; AKT3; AKT3; AKT1; AKT1; MAPK8; MAPK9; MAPK10; BNIP3; CFLAR; GABARAP; GABARAPL2; MAP1LC3B; MAP1LC3A; SQSTM1; SQSTM1; EIF2S1; ATG101","CRK; TAB2; DOCK1; IKBKB; IKBKB; IRAK1; ARPC1B; ARPC1B; MAP2K2; MAP2K2; ARPC5; ARPC3; ARPC4; ARHGEF12; PXN; MAP2K3; WIPF2; GNAQ; FN1; IRAK4; IRAK4; MAPK8; MAPK9; MAPK10; TBK1; TBK1; MAP3K7; NFKBIA; NFKBIA; ARPC2; VAV2; PTK2; RHOG; RAC1; ARHGEF1; MAP2K4; BCAR1; BCAR1; TRAF6; IKBKG; IKBKG; WIPF1; WIPF1; ARHGEF7; ARPC5L; GIT2; CDC42; NFATC3; IRF3; NFKB1; NFKB1; AKT3; AKT3; AKT3; AKT1; AKT1; PIK3CA; PIK3CA; PIK3R1; PKN1; RPS6KA2; ACTB; ACTB; ACTG1; PIP5K1C; PIP5K1A; ROCK1; SKAP2","ABI1; WASF3; TAB2; NCKAP1; BRK1; NFKB1; NFKB1; CYFIP1; CYFIP1; CYFIP1; IKBKB; IKBKB; ARPC1B; ARPC1B; ARPC4; ARPC3; ARF1; ARPC5; ARPC2; LPAR1; GNA13; ARHGEF12; F2R; MAPK8; WASF1; GAPDH; GNA12; NCK2; FAS; FAS; RAC1; IL1R1; CDC42; TRAF6; NCK1; MAP3K7; ARPC5L; IRAK4; IRAK4; CYCS; CYCS; ARHGEF1; IKBKG; IKBKG; RIPK1; MAPK9; TNFRSF10B; MAPK10; IRAK1; CYTH3; CYTH1; WIPF2; WIPF1; WIPF1; CASP4; MYO1B; MYO6; MYO10; MYO1D; MYO1E; NFKBIA; NFKBIA; ABCF2; TMBIM6; TUBB6; TUBA1B; TUBA1C; TUBB2A; ACTB; ACTB; ACTG1; CASP3; ROCK1","AK6; AK6; ADK; ADSL; DUT; TYMP; TYMP; NT5C2; AK5; GUK1; NME7; NT5C; HPRT1; APRT; IMPDH1; IMPDH2; ITPA; NME1; NME2; NME3; NME4; PNP; PNP; HDDC2; NT5C3A; ENPP1; CMPK1; TK1; TK2; UCK2; ASMTL; GMPS; ENTPD6; ENTPD4","ACAA2; HIBADH; ECHS1; HIBCH; ACAA1; HSD17B10; HSD17B10; HADHA; IVD; ACAT1; ACAT1; ACAT2; MMUT; ALDH7A1; PCCA; AUH; AUH; MCCC1; BCKDHB; MCCC2; MCCC2; MCCC2; AACS","PTK2; ITGAV; CRK; TRAF5; PXN; PDIA3; GNB2; GNA13; EGFR; EGFR; AKT3; AKT3; AKT3; TBK1; TBK1; GNA11; GNA12; AKAP13; ADCY7; GNG10; ADCY9; GNG11; GNAQ; GNAS; PIK3R1; AKT1; AKT1; ARHGEF12; GNG5; PRKACB; PDGFRA; PDGFRA; FAS; FAS; IKBKB; IKBKB; IL10RB; IL10RB; NRAS; NRAS; RAF1; RAF1; PPP3CC; PPP3R1; CYCS; CYCS; IL1R1; PPP3CB; PIK3CA; PIK3CA; MAP2K2; MAP2K2; RIPK1; SOS2; SOS1; SOS1; ROCK1; ARHGEF1; CXCL12; IKBKG; IKBKG; TAPBP; TAPBP; TAPBP; TAPBP; TAPBP; TAPBP; HLA-E; HLA-E; HLA-E; HLA-E; HLA-E; HLA-E; HLA-E; HLA-B; HLA-B; HLA-B; HLA-B; HLA-B; HLA-B; HLA-A; HLA-A; HLA-A; HLA-A; HLA-A; HLA-A; HLA-A; HLA-A; NFKBIA; NFKBIA; IRF3; NFKB1; NFKB1; STAT3; STAT3; NFATC3; PLCB4; RAC1; BCAR1; BCAR1; CREB3; CREB1; CREB3L2; CREB5; CASP3; B2M; B2M; B2M; VEGFA; ITPR2","AASS; ECHS1; EZH1; SETD2; SETD2; HADHA; KMT5A; KMT2A; KMT2A; ALDH7A1; KMT5B; NSD3; ASH1L; KMT2E; SMYD2; KMT2C; NSD1; NSD1; SMYD3; NSD2; PRDM2; EHMT1; CAMKMT; KMT2D; SETD7; DOT1L; PHYKPL; PLOD2; PLOD3; ACAT1; ACAT1; ACAT2; SETDB1","GABARAP; GABARAPL2; ULK2; ATG12; ATG101; ATG7; IGBP1; ATG10; ATG3; ATG4C; ATG5; ATG13","AKR1A1; ADH5; ENO1; ENO1; GAPDH; GPI; GPI; GPI; LDHA; LDHA; LDHB; ALDH7A1; PCK2; PCK2; PDHA1; PDHB; PFKL; PFKP; PGAM1; PGK1; PGM1; PKM; PGM2; TPI1; ADPGK; MINPP1","ACAA2; ECHS1; HACD2; HADHA; HSD17B12; HACD3; PPT1; PPT1; ELOVL5; ELOVL1; ELOVL6; HACD1; TECR; ACOT7; THEM4","DPYD; DPYD; DUT; TYMP; TYMP; NT5C2; NME7; NT5C; NUDT2; NME1; NME2; NME3; NME4; HDDC2; NT5C3A; ENPP1; CMPK1; TK1; TK2; UCK2; UMPS; ASMTL; ENTPD6; ENTPD4","CLTA; CLTB; STX2; VAMP2; AP2M1; AP2S1; AP2A1; AP2A2; AP2A2; AP2A2; RIMS1","DAD1; STT3B; DPM3; RPN1; RPN2; DPM1; ALG3; MGAT4B; MGAT4B; FUT8; B4GALT1; MAN2A1; MGAT1; MGAT2; MGAT5; MOGS; MOGS; ALG13; TUSC3; ALG2; B4GALT2; MAN1A2; MAN1B1; ALG8; MAN1A1","FAS; FAS; MAPK9; APAF1; CTSC; CTSC; MAPK10; CTSL; AKT1; AKT1; RAF1; RAF1; CAPN1; AKT3; AKT3; AKT3; TNFRSF10B; MAP2K2; MAP2K2; MAP3K14; MAP3K14; MAP3K14; CYCS; CYCS; NGF; NGF; NRAS; NRAS; EIF2S1; MAPK8; CTSB; CTSB; CTSD; CAPN2; IKBKB; IKBKB; EIF2AK3; IKBKG; IKBKG; NFKBIA; NFKBIA; NFKB1; NFKB1; PIK3CA; PIK3CA; PIK3R1; CASP3; MCL1; CFLAR; RIPK1; PTPN13; GADD45B; PARP4; PARP1; ACTG1; DFFA; TUBA1C; LMNB2; ACTB; ACTB; TUBA1B; LMNB1","ACAA2; ECHS1; HACD2; ACSL4; ACAA1; HADHA; ACACA; ACACA; ACADVL; HSD17B12; HACD3; PPT1; PPT1; ELOVL6; ACOX3; HACD1; TECR; CPT1A; SCP2; ACAT1; ACAT1; ACAT2","IKBKB; IKBKB; ADAM10; MAPK10; ADAM17; ADAM17; MAPK8; MAP3K14; MAP3K14; MAP3K14; MAPK9; MAP2K4; JAM3; NFKBIA; NFKBIA; NFKB1; NFKB1; NOD1; IKBKG; IKBKG; EGFR; EGFR; TJP1; TJP1","DPYD; DPYD; DUT; TYMP; TYMP; GUSB; NME7; HPRT1; IMPDH1; IMPDH2; ITPA; NME1; NME2; NME3; NME4; CMPK1; TK1; TK2; TPMT; UCK2; UMPS; GMPS; GSTM3; GSTP1; GSTP1; MGST1; MGST3; GSTO1","ABI2; CRK; NCKAP1; PTK2; BRK1; CYFIP1; CYFIP1; CYFIP1; AKT3; AKT3; AKT3; FGF7; ITGAV; ITGB8; DIAPH1; GNA13; ITGA11; EGFR; EGFR; FGF5; MRAS; FN1; RRAS2; FGF2; LPAR1; DOCK1; GNA12; ARHGEF7; FGFR1; FGFR1; ITGA6; ITGA1; ROCK1; F2R; RAC1; SOS2; NRAS; NRAS; RRAS; CXCL12; TIAM1; WASF1; VAV2; PDGFRA; PDGFRA; PDGFRB; AKT1; AKT1; PDGFC; ARHGEF12; CDC42; ARHGEF1; DIAPH3; BCAR1; BCAR1; SOS1; SOS1; ACTG1; ACTB; ACTB; PPP1R12C; SSH1; SSH2; PIKFYVE; PIP5K1C; PIP4K2A; VCL; VCL; PIP5K1A; ARHGAP35; MYLK; RAF1; RAF1; MAP2K2; MAP2K2; BRAF; BRAF; ARPC5L; ARPC5; ARPC4; PIK3CA; PIK3CA; PFN1; PIK3R1; MYL9; PFN2; ARPC3; ARPC1B; ARPC1B; ARPC2; IQGAP1; MYL12B; MYL12A; MSN; RDX; TMSB4X; CFL2; CFL2; CFL1; ACTN4; ACTN4; ACTN1; ACTN1; PXN; GSN","IRAK1; IRAK4; IRAK4; IRAK3; RAPGEF1; NRAS; NRAS; IRS1; AKT3; AKT3; AKT3; MAP2K5; PLCG2; PLCG2; RAC1; NGF; NGF; PIK3CA; PIK3CA; AKT1; AKT1; NTF3; SOS2; RAP1A; MAPK8; IKBKB; IKBKB; MAP3K1; MAPK9; ARHGDIA; MAP2K2; MAP2K2; RAF1; RAF1; PIK3R1; SOS1; SOS1; BRAF; BRAF; TRAF6; RPS6KA2; CDC42; MAPK10; GAB1; SH2B3; SH2B3; KIDINS220; BEX3; FRS2; CRK; NFKBIA; NFKBIA; FOXO3; PRDM4; RIPK2; MAGED1; NFKB1; NFKB1","CREBBP; ELOB; ELOC; AKT3; AKT3; AKT3; EGLN2; INSR; IGF1R; AKT1; AKT1; PIK3CA; PIK3CA; EGFR; EGFR; MKNK2; PIK3R1; IFNGR2; IFNGR2; IFNGR2; MAP2K2; MAP2K2; EGLN1; NFKB1; NFKB1; STAT3; STAT3; PLCG2; PLCG2; PDK1; RBX1; EIF4E2; PFKL; VEGFA; PFKP; PGK1; LTBR; SLC2A1; SLC2A1; ANGPT1; GAPDH; ANGPT2; HMOX1; LDHA; LDHA; LDHB; ENO1; ENO1; TEK; PDHA1; PDHB","AP2M1; CLTA; SH3KBP1; AP2S1; CLTB; EPN1; EPN2; AP2A1; GRK5; SH3GL1; AP2A2; AP2A2; AP2A2; EPS15; VTA1; DAB2; EPS15L1; RAB11FIP2; CBLB; CBLB; VPS45; VPS45; MVB12A; MVB12B; TGFBR2; TGFBR2; RNF41; TRAF6; WWP1; NEDD4; SMURF2; SMURF1; SMURF1; ITCH; ITCH; IGF1R; RUFY1; RUFY1; EHD2; EGFR; EGFR; PDGFRA; PDGFRA; STAM2; STAM; SMAD2; CAV1; CAV2","EGLN2; CRK; GAB1; RAPGEF1; RAC1; MET; MET; EGLN1; MAP2K2; MAP2K2; BRAF; BRAF; CDC42; RAF1; RAF1; CREBBP; EPAS1; PIK3CA; PIK3CA; PIK3R1; AKT3; AKT3; AKT3; AKT1; AKT1; SOS1; SOS1; SOS2; RAP1A; ETS1; SLC2A1; SLC2A1; VEGFA","ARPC5; ARPC4; ARPC3; ARPC1B; ARPC1B; ARPC2; CRK; GAB1; DOCK1; FN1; RHOG; PXN; RAC1; PIK3CA; PIK3CA; PIK3R1; PTK2; ARHGAP10; ARPC5L; WASF1; VCL; VCL; CDC42; BCAR1; BCAR1; MET; MET; ACTB; ACTB; ACTG1; DNM3; CTNNA1","AKT3; AKT3; AKT3; EGFR; EGFR; MAP2K2; MAP2K2; RAF1; RAF1; AKT1; AKT1; PDGFRA; PDGFRA; PDGFRB; FGFR1; FGFR1; MET; MET; PIK3CA; PIK3CA; PIK3R1; NRAS; NRAS; PDK1; SCO2; SLC7A5; LDHA; LDHA; PGAM1; PKM; LDHB; SLC2A1; SLC2A1; PFKP; SLC1A5; GLS; PFKL; PDHA1; PDHB; G6PD; G6PD","ECHS1; HIBCH; HADHA; ACACA; ACACA; LDHA; LDHA; LDHB; MMUT; PCCA; BCKDHB; ACOX3; SUCLG2; SUCLG1; SUCLA2"],["ENSG00000107175; ENSG00000196961; ENSG00000130706; ENSG00000161203; ENSG00000042753; ENSG00000122705; ENSG00000175416; ENSG00000118260; ENSG00000183020; ENSG00000280759; ENSG00000281385; ENSG00000116478; ENSG00000196591; ENSG00000126775; ENSG00000127946; ENSG00000089902; ENSG00000114446; ENSG00000169375; ENSG00000148572; ENSG00000196455; ENSG00000182158; ENSG00000127922; ENSG00000083290; ENSG00000123395; ENSG00000175224; LRG_876; ENSG00000172115; ENSG00000156052; ENSG00000110497; ENSG00000120868; ENSG00000066739; ENSG00000101333; ENSG00000050748; ENSG00000107643; ENSG00000109339; ENSG00000146592; ENSG00000084093; ENSG00000164305; ENSG00000070540; ENSG00000157954; ENSG00000198727; ENSG00000184076; ENSG00000073578; ENSG00000156467; ENSG00000170759; ENSG00000179091; ENSG00000126214; ENSG00000164405; ENSG00000140740; ENSG00000173660; ENSG00000117118; LRG_316; LRG_317; ENSG00000143252; ENSG00000010256; ENSG00000169021; ENSG00000112096; ENSG00000106459; ENSG00000164294; LRG_652; ENSG00000142168","ENSG00000062485; ENSG00000140905; LRG_541; ENSG00000184207; ENSG00000014641; ENSG00000146701; ENSG00000122729; ENSG00000100412; ENSG00000175198; ENSG00000182199; ENSG00000178922; ENSG00000121691; ENSG00000137106; ENSG00000075239; LRG_1400; ENSG00000120437; ENSG00000146085","ENSG00000160211; LRG_148; ENSG00000130313; ENSG00000105220; LRG_1178; ENSG00000282019; ENSG00000023697; ENSG00000141959; ENSG00000067057; ENSG00000142657; ENSG00000079739; ENSG00000169299; LRG_264; ENSG00000147224; ENSG00000177156; ENSG00000163931; ENSG00000049239","ENSG00000177889; ENSG00000125826; ENSG00000150760; ENSG00000055208; ENSG00000167193; ENSG00000082512; ENSG00000014216; ENSG00000106100; ENSG00000109320; LRG_1316; ENSG00000213585; ENSG00000104365; LRG_1241; ENSG00000137076; ENSG00000145782; ENSG00000172175; LRG_1221; ENSG00000035403; LRG_383; ENSG00000244687; ENSG00000162909; ENSG00000162704; ENSG00000241553; ENSG00000111229; ENSG00000118680; ENSG00000117020; LRG_1396; ENSG00000275199; ENSG00000163466; ENSG00000146648; LRG_304; ENSG00000130429; LRG_1188; ENSG00000142208; LRG_721; ENSG00000109339; ENSG00000101335; ENSG00000136950; ENSG00000067900; ENSG00000169398; ENSG00000171914; ENSG00000130402; ENSG00000282844; ENSG00000131504; ENSG00000143761; ENSG00000101608; ENSG00000104312; ENSG00000115594; ENSG00000089159; ENSG00000136238; ENSG00000112290; ENSG00000175104; ENSG00000169139; ENSG00000108669; ENSG00000135341; ENSG00000183735; LRG_1306; ENSG00000196954; ENSG00000057663; ENSG00000269335; LRG_70; ENSG00000107643; ENSG00000050748; ENSG00000137275; ENSG00000050820; ENSG00000285460; ENSG00000100387; LRG_886; ENSG00000072110; ENSG00000176171; ENSG00000008256; ENSG00000161011; ENSG00000284099; ENSG00000070540; ENSG00000070831; ENSG00000140941; ENSG00000101460; ENSG00000157954; LRG_89; ENSG00000100906; ENSG00000122565; ENSG00000126456; ENSG00000145675; ENSG00000138193; ENSG00000197943; LRG_376; ENSG00000101333; ENSG00000187239; ENSG00000123104; ENSG00000137942; ENSG00000121879; LRG_310; ENSG00000126247; ENSG00000184009; ENSG00000075624; LRG_132; ENSG00000204673; LRG_876; ENSG00000172115; ENSG00000171497; ENSG00000168385; ENSG00000122545; ENSG00000070087; ENSG00000108518; ENSG00000150907; ENSG00000118689","ENSG00000051523; LRG_52; ENSG00000096384; ENSG00000188612; ENSG00000184900; ENSG00000166598; ENSG00000105971; ENSG00000136238; ENSG00000105974; ENSG00000063660; ENSG00000116044; ENSG00000115884; ENSG00000145675; ENSG00000138448; ENSG00000107643; ENSG00000169398; ENSG00000065559; ENSG00000142208; LRG_721; ENSG00000117020; LRG_1396; ENSG00000275199; ENSG00000138696; ENSG00000121989; ENSG00000204217; LRG_712; ENSG00000104365; LRG_1241; ENSG00000115594; ENSG00000135341; ENSG00000124145; ENSG00000050748; ENSG00000109339; ENSG00000269335; LRG_70; ENSG00000121879; LRG_310; ENSG00000107779; LRG_298; ENSG00000137764; ENSG00000079999; ENSG00000068305; ENSG00000109320; LRG_1316; ENSG00000116030; ENSG00000075624; LRG_132; ENSG00000184009; ENSG00000161011; ENSG00000284099; ENSG00000148834; ENSG00000104368; LRG_570; ENSG00000135047; LRG_723; ENSG00000084207; ENSG00000136810; ENSG00000100292; ENSG00000008394; ENSG00000134202; ENSG00000143198; ENSG00000112715; ENSG00000181019; ENSG00000100348; ENSG00000087245","ENSG00000108479; ENSG00000156958; LRG_787; ENSG00000198380; ENSG00000112699; ENSG00000105220; LRG_1178; ENSG00000282019; ENSG00000173540; ENSG00000144591; ENSG00000079739; ENSG00000013375; LRG_1312; ENSG00000100417; ENSG00000140650; ENSG00000095380; ENSG00000169299; ENSG00000124357; ENSG00000100522; ENSG00000109814; ENSG00000131459; ENSG00000115652","ENSG00000117448; ENSG00000151376; ENSG00000197894; LRG_504; ENSG00000091483; ENSG00000124767; ENSG00000063854; ENSG00000278540; ENSG00000275176; ENSG00000075239; LRG_1400; ENSG00000120437; ENSG00000134333; ENSG00000288299; ENSG00000111716; ENSG00000014641; ENSG00000146701; ENSG00000164904; ENSG00000100889; ENSG00000285241; ENSG00000131828; ENSG00000168291; ENSG00000067225; ENSG00000170634; ENSG00000137106","ENSG00000163281; ENSG00000108479; ENSG00000156958; LRG_787; ENSG00000198380; ENSG00000112699; ENSG00000105220; LRG_1178; ENSG00000282019; ENSG00000173540; ENSG00000144591; ENSG00000213614; ENSG00000049860; ENSG00000079739; ENSG00000013375; LRG_1312; ENSG00000100417; ENSG00000140650; ENSG00000095380; ENSG00000169299; ENSG00000124357; ENSG00000100522; ENSG00000109814; ENSG00000115652; ENSG00000131459","ENSG00000062485; ENSG00000125166; LRG_610; ENSG00000138413; ENSG00000182054; LRG_611; ENSG00000166411; ENSG00000101365; ENSG00000067829; ENSG00000122729; ENSG00000100412","ENSG00000062485; ENSG00000074800; LRG_1165; ENSG00000111640; ENSG00000125166; ENSG00000135069; LRG_610; ENSG00000138413; ENSG00000182054; LRG_611; ENSG00000166411; ENSG00000101365; ENSG00000067829; ENSG00000126522; ENSG00000116984; ENSG00000122729; ENSG00000100412; ENSG00000141959; ENSG00000067057; ENSG00000171314; ENSG00000102144; ENSG00000067225; ENSG00000182199; ENSG00000163931; ENSG00000135821; ENSG00000111669; LRG_264; ENSG00000147224; ENSG00000038274; ENSG00000168906; ENSG00000177156","ENSG00000118217; ENSG00000197930; ENSG00000134001; ENSG00000086232; ENSG00000116044; ENSG00000162909; ENSG00000014216; ENSG00000172071; ENSG00000138674; ENSG00000173960; ENSG00000182446; ENSG00000162191; ENSG00000088833; ENSG00000167671; ENSG00000152700; ENSG00000079332; ENSG00000155660; ENSG00000143870; ENSG00000089248; ENSG00000239264; ENSG00000050748; ENSG00000109339; ENSG00000107643; ENSG00000165280; LRG_657; ENSG00000150961; ENSG00000101310; LRG_1134; ENSG00000113615; ENSG00000138802; ENSG00000070010","ENSG00000091409; ENSG00000137809; ENSG00000138448; ENSG00000213949; ENSG00000105855; ENSG00000187498; LRG_475; ENSG00000142156; ENSG00000118680; ENSG00000101335; ENSG00000134871; ENSG00000163359; LRG_473; ENSG00000142173; LRG_476; LRG_1; ENSG00000108821; LRG_2; ENSG00000164692; ENSG00000105664; ENSG00000167193; ENSG00000150760; ENSG00000101608; ENSG00000105976; LRG_662; ENSG00000065534; ENSG00000116473; ENSG00000115414; ENSG00000010810; ENSG00000091136; ENSG00000107263; ENSG00000172037; ENSG00000196569; LRG_409; ENSG00000041982; ENSG00000146648; LRG_304; ENSG00000169398; ENSG00000112715; ENSG00000135862; ENSG00000186340; LRG_309; ENSG00000134853; ENSG00000113721; ENSG00000140443; ENSG00000160293; ENSG00000112769; LRG_433; ENSG00000070831; ENSG00000136238; ENSG00000173511; ENSG00000150630; ENSG00000050820; ENSG00000285460; ENSG00000168477; ENSG00000229341; ENSG00000233323; ENSG00000231608; ENSG00000236221; ENSG00000206258; ENSG00000229353; ENSG00000236236; ENSG00000115904; LRG_754; ENSG00000145431; ENSG00000105971; ENSG00000100485; ENSG00000105974; ENSG00000128591; LRG_870; ENSG00000188677; ENSG00000171914; ENSG00000136068; ENSG00000197702; ENSG00000035403; LRG_383; ENSG00000213853; ENSG00000089159; ENSG00000130402; ENSG00000282844; LRG_886; ENSG00000072110; ENSG00000125753; ENSG00000159840; ENSG00000285443; ENSG00000137076; ENSG00000125503; ENSG00000131504; ENSG00000186111; ENSG00000121879; LRG_310; ENSG00000145675; ENSG00000143398; ENSG00000117020; LRG_1396; ENSG00000275199; ENSG00000067900; ENSG00000162909; ENSG00000160007; ENSG00000142208; LRG_721; ENSG00000132155; LRG_413; ENSG00000107643; ENSG00000050748; ENSG00000157764; LRG_299; ENSG00000109339; ENSG00000075624; LRG_132; ENSG00000184009","ENSG00000154582; ENSG00000055208; ENSG00000082512; ENSG00000103363; ENSG00000184216; ENSG00000167004; ENSG00000117020; LRG_1396; ENSG00000275199; ENSG00000107758; ENSG00000172354; ENSG00000109320; LRG_1316; ENSG00000183735; LRG_1306; LRG_92; ENSG00000213281; ENSG00000142208; LRG_721; ENSG00000136238; ENSG00000121879; LRG_310; ENSG00000145675; ENSG00000088256; ENSG00000026103; LRG_134; ENSG00000135341; ENSG00000156052; ENSG00000107643; ENSG00000174021; ENSG00000242616; ENSG00000127920; LRG_876; ENSG00000172115; ENSG00000175104; ENSG00000198001; LRG_75; ENSG00000114125; ENSG00000050748; ENSG00000120910; ENSG00000221823; ENSG00000034152; ENSG00000126934; LRG_750; ENSG00000132155; LRG_413; ENSG00000109339; ENSG00000169398; ENSG00000137275; ENSG00000145041; ENSG00000231925; LRG_114; ENSG00000112493; ENSG00000236490; ENSG00000206208; ENSG00000206281; ENSG00000206503; ENSG00000229215; ENSG00000224320; ENSG00000235657; ENSG00000231834; ENSG00000227715; ENSG00000206505; ENSG00000223980; ENSG00000234745; ENSG00000223532; ENSG00000224608; ENSG00000232126; ENSG00000206450; ENSG00000228964; ENSG00000204592; ENSG00000233904; ENSG00000229252; ENSG00000236632; ENSG00000225201; ENSG00000206493; ENSG00000230254; ENSG00000126456; ENSG00000197943; LRG_376; ENSG00000100387; ENSG00000072736; ENSG00000104365; LRG_1241; ENSG00000269335; LRG_70; ENSG00000165410; LRG_213; ENSG00000164305; ENSG00000172757; ENSG00000244509; ENSG00000166747; ENSG00000106367; ENSG00000166710; LRG_1215; ENSG00000273686; ENSG00000100280; ENSG00000089159; ENSG00000167193; ENSG00000123104","ENSG00000135823; ENSG00000100568; ENSG00000111450; ENSG00000168818; ENSG00000151532; ENSG00000103496; ENSG00000079950; ENSG00000265808; ENSG00000105829; ENSG00000106636; ENSG00000220205; ENSG00000049245; ENSG00000124222; ENSG00000170310; ENSG00000108433","ENSG00000140443; ENSG00000145782; ENSG00000136560; ENSG00000155792; ENSG00000126775; ENSG00000110931; ENSG00000144848; ENSG00000157954; ENSG00000133818; ENSG00000070540; ENSG00000169047; ENSG00000189403; ENSG00000148572; ENSG00000197548; LRG_92; ENSG00000213281; ENSG00000164253; ENSG00000142875; ENSG00000185950; ENSG00000102710; ENSG00000168209; ENSG00000158186; ENSG00000126934; LRG_750; ENSG00000089289; ENSG00000172071; ENSG00000204673; ENSG00000183735; LRG_1306; ENSG00000110497; ENSG00000083290; ENSG00000125703; ENSG00000132155; LRG_413; ENSG00000175224; ENSG00000126458; ENSG00000135341; ENSG00000057663; ENSG00000175104; ENSG00000196455; ENSG00000062716; ENSG00000152348; ENSG00000099940; ENSG00000066739; ENSG00000121879; LRG_310; ENSG00000145675; ENSG00000100330; ENSG00000117020; LRG_1396; ENSG00000275199; ENSG00000142208; LRG_721; ENSG00000107643; ENSG00000050748; ENSG00000109339; ENSG00000176171; ENSG00000003402; ENSG00000170296; ENSG00000034713; ENSG00000140941; ENSG00000101460; ENSG00000161011; ENSG00000284099; ENSG00000134001; ENSG00000123395","ENSG00000167193; ENSG00000055208; ENSG00000150760; ENSG00000104365; LRG_1241; ENSG00000184216; ENSG00000130429; LRG_1188; ENSG00000126934; LRG_750; ENSG00000162704; ENSG00000111229; ENSG00000241553; ENSG00000196914; ENSG00000089159; ENSG00000034152; ENSG00000171475; ENSG00000156052; ENSG00000115414; ENSG00000198001; LRG_75; ENSG00000107643; ENSG00000050748; ENSG00000109339; ENSG00000183735; LRG_1306; ENSG00000135341; LRG_89; ENSG00000100906; ENSG00000163466; ENSG00000160293; ENSG00000169398; ENSG00000177105; ENSG00000136238; ENSG00000076928; ENSG00000065559; ENSG00000050820; ENSG00000285460; ENSG00000175104; ENSG00000269335; LRG_70; ENSG00000115935; LRG_374; ENSG00000102606; ENSG00000136950; ENSG00000139436; ENSG00000070831; ENSG00000072736; ENSG00000126456; ENSG00000109320; LRG_1316; ENSG00000117020; LRG_1396; ENSG00000275199; ENSG00000142208; LRG_721; ENSG00000121879; LRG_310; ENSG00000145675; ENSG00000123143; ENSG00000071242; ENSG00000075624; LRG_132; ENSG00000184009; ENSG00000186111; ENSG00000143398; ENSG00000067900; ENSG00000005020","ENSG00000136754; ENSG00000132970; ENSG00000055208; ENSG00000061676; ENSG00000254999; ENSG00000109320; LRG_1316; ENSG00000273749; ENSG00000288461; ENSG00000280618; ENSG00000104365; LRG_1241; ENSG00000130429; LRG_1188; ENSG00000241553; ENSG00000111229; ENSG00000143761; ENSG00000162704; ENSG00000163466; ENSG00000198121; ENSG00000120063; ENSG00000196914; ENSG00000181104; ENSG00000107643; ENSG00000112290; ENSG00000111640; ENSG00000146535; ENSG00000071051; ENSG00000026103; LRG_134; ENSG00000136238; ENSG00000115594; ENSG00000070831; ENSG00000175104; ENSG00000158092; ENSG00000135341; ENSG00000136950; ENSG00000198001; LRG_75; LRG_876; ENSG00000172115; ENSG00000076928; ENSG00000269335; LRG_70; ENSG00000137275; ENSG00000050748; ENSG00000120889; ENSG00000109339; ENSG00000184216; ENSG00000008256; ENSG00000108669; ENSG00000171475; ENSG00000115935; LRG_374; ENSG00000196954; ENSG00000128641; ENSG00000196586; ENSG00000145555; ENSG00000176658; ENSG00000157483; LRG_89; ENSG00000100906; ENSG00000033050; ENSG00000139644; ENSG00000176014; ENSG00000123416; ENSG00000167553; ENSG00000137267; ENSG00000075624; LRG_132; ENSG00000184009; ENSG00000164305; ENSG00000067900","ENSG00000085231; ENSG00000279247; ENSG00000156110; ENSG00000239900; ENSG00000128951; ENSG00000025708; LRG_727; ENSG00000076685; ENSG00000154027; ENSG00000143774; ENSG00000143156; ENSG00000125458; ENSG00000165704; ENSG00000198931; ENSG00000106348; ENSG00000178035; ENSG00000125877; ENSG00000239672; ENSG00000243678; ENSG00000103024; ENSG00000103202; ENSG00000198805; LRG_91; ENSG00000111906; ENSG00000122643; ENSG00000197594; ENSG00000162368; ENSG00000167900; ENSG00000166548; ENSG00000143179; ENSG00000169093; ENSG00000163655; ENSG00000197586; ENSG00000197217","ENSG00000167315; ENSG00000106049; ENSG00000127884; ENSG00000198130; ENSG00000060971; ENSG00000072506; LRG_450; ENSG00000084754; ENSG00000128928; ENSG00000075239; LRG_1400; ENSG00000120437; ENSG00000146085; ENSG00000164904; ENSG00000175198; ENSG00000148090; LRG_449; ENSG00000078070; ENSG00000083123; ENSG00000131844; ENSG00000281742; ENSG00000275300; ENSG00000081760","ENSG00000169398; ENSG00000138448; ENSG00000167193; ENSG00000082512; ENSG00000089159; ENSG00000167004; ENSG00000172354; ENSG00000120063; ENSG00000146648; LRG_304; ENSG00000117020; LRG_1396; ENSG00000275199; ENSG00000183735; LRG_1306; ENSG00000088256; ENSG00000146535; ENSG00000170776; ENSG00000121281; ENSG00000242616; ENSG00000162104; ENSG00000127920; ENSG00000156052; ENSG00000087460; ENSG00000145675; ENSG00000142208; LRG_721; ENSG00000196914; ENSG00000174021; ENSG00000142875; LRG_309; ENSG00000134853; ENSG00000026103; LRG_134; ENSG00000104365; LRG_1241; ENSG00000243646; LRG_152; LRG_92; ENSG00000213281; ENSG00000132155; LRG_413; ENSG00000120910; ENSG00000221823; LRG_876; ENSG00000172115; ENSG00000115594; ENSG00000107758; ENSG00000121879; LRG_310; ENSG00000126934; LRG_750; ENSG00000137275; ENSG00000100485; ENSG00000115904; LRG_754; ENSG00000067900; ENSG00000076928; ENSG00000107562; ENSG00000269335; LRG_70; ENSG00000231925; LRG_114; ENSG00000112493; ENSG00000236490; ENSG00000206208; ENSG00000206281; ENSG00000204592; ENSG00000233904; ENSG00000229252; ENSG00000236632; ENSG00000225201; ENSG00000206493; ENSG00000230254; ENSG00000234745; ENSG00000223532; ENSG00000224608; ENSG00000232126; ENSG00000206450; ENSG00000228964; ENSG00000206503; ENSG00000229215; ENSG00000224320; ENSG00000235657; ENSG00000231834; ENSG00000227715; ENSG00000206505; ENSG00000223980; LRG_89; ENSG00000100906; ENSG00000126456; ENSG00000109320; LRG_1316; ENSG00000168610; LRG_112; ENSG00000072736; ENSG00000101333; ENSG00000136238; ENSG00000050820; ENSG00000285460; ENSG00000107175; ENSG00000118260; ENSG00000182158; ENSG00000146592; ENSG00000164305; ENSG00000166710; LRG_1215; ENSG00000273686; ENSG00000112715; ENSG00000123104","ENSG00000008311; ENSG00000127884; ENSG00000108799; LRG_775; ENSG00000181555; ENSG00000084754; ENSG00000183955; ENSG00000118058; LRG_613; ENSG00000164904; ENSG00000110066; ENSG00000147548; ENSG00000116539; ENSG00000005483; ENSG00000143499; ENSG00000055609; ENSG00000165671; LRG_512; ENSG00000185420; ENSG00000109685; ENSG00000116731; ENSG00000181090; ENSG00000143919; ENSG00000167548; ENSG00000145391; ENSG00000104885; ENSG00000175309; ENSG00000152952; ENSG00000106397; ENSG00000075239; LRG_1400; ENSG00000120437; ENSG00000143379","ENSG00000170296; ENSG00000034713; ENSG00000083290; ENSG00000145782; ENSG00000123395; ENSG00000197548; ENSG00000089289; ENSG00000152348; ENSG00000144848; ENSG00000125703; ENSG00000057663; ENSG00000175224","ENSG00000117448; ENSG00000197894; ENSG00000074800; LRG_1165; ENSG00000111640; ENSG00000105220; LRG_1178; ENSG00000282019; ENSG00000134333; ENSG00000288299; ENSG00000111716; ENSG00000164904; ENSG00000100889; ENSG00000285241; ENSG00000131828; ENSG00000168291; ENSG00000141959; ENSG00000067057; ENSG00000171314; ENSG00000102144; ENSG00000079739; ENSG00000067225; ENSG00000169299; ENSG00000111669; ENSG00000159322; ENSG00000107789","ENSG00000167315; ENSG00000127884; ENSG00000206527; ENSG00000084754; ENSG00000149084; ENSG00000074696; ENSG00000131238; LRG_690; ENSG00000012660; ENSG00000066322; ENSG00000170522; ENSG00000165996; ENSG00000099797; ENSG00000097021; ENSG00000159445","ENSG00000188641; LRG_722; ENSG00000128951; ENSG00000025708; LRG_727; ENSG00000076685; ENSG00000143156; ENSG00000125458; ENSG00000164978; ENSG00000239672; ENSG00000243678; ENSG00000103024; ENSG00000103202; ENSG00000111906; ENSG00000122643; ENSG00000197594; ENSG00000162368; ENSG00000167900; ENSG00000166548; ENSG00000143179; ENSG00000114491; ENSG00000169093; ENSG00000197586; ENSG00000197217","ENSG00000122705; ENSG00000175416; ENSG00000111450; ENSG00000220205; ENSG00000161203; ENSG00000042753; ENSG00000196961; ENSG00000183020; ENSG00000280759; ENSG00000281385; ENSG00000079841","ENSG00000129562; ENSG00000163527; ENSG00000179085; ENSG00000163902; ENSG00000118705; ENSG00000000419; ENSG00000214160; ENSG00000161013; ENSG00000284501; ENSG00000033170; ENSG00000086062; ENSG00000112893; ENSG00000131446; ENSG00000168282; ENSG00000152127; ENSG00000115275; LRG_1226; ENSG00000101901; ENSG00000104723; ENSG00000119523; ENSG00000117411; ENSG00000198162; ENSG00000177239; ENSG00000159063; ENSG00000111885","ENSG00000026103; LRG_134; ENSG00000050748; ENSG00000120868; ENSG00000109861; LRG_50; ENSG00000109339; ENSG00000135047; ENSG00000142208; LRG_721; ENSG00000132155; LRG_413; ENSG00000014216; ENSG00000117020; LRG_1396; ENSG00000275199; ENSG00000120889; ENSG00000126934; LRG_750; ENSG00000006062; LRG_1222; ENSG00000282637; LRG_876; ENSG00000172115; LRG_260; ENSG00000134259; LRG_92; ENSG00000213281; ENSG00000134001; ENSG00000107643; ENSG00000164733; ENSG00000285132; ENSG00000117984; ENSG00000162909; ENSG00000104365; LRG_1241; ENSG00000172071; ENSG00000269335; LRG_70; LRG_89; ENSG00000100906; ENSG00000109320; LRG_1316; ENSG00000121879; LRG_310; ENSG00000145675; ENSG00000164305; ENSG00000143384; ENSG00000003402; ENSG00000137275; ENSG00000163629; ENSG00000099860; ENSG00000102699; ENSG00000143799; ENSG00000184009; ENSG00000160049; ENSG00000167553; ENSG00000176619; ENSG00000075624; LRG_132; ENSG00000123416; ENSG00000113368","ENSG00000167315; ENSG00000127884; ENSG00000206527; ENSG00000068366; ENSG00000060971; ENSG00000084754; ENSG00000278540; ENSG00000275176; ENSG00000072778; ENSG00000149084; ENSG00000074696; ENSG00000131238; LRG_690; ENSG00000170522; ENSG00000087008; ENSG00000165996; ENSG00000099797; ENSG00000110090; ENSG00000116171; ENSG00000075239; LRG_1400; ENSG00000120437","ENSG00000104365; LRG_1241; ENSG00000137845; ENSG00000109339; ENSG00000151694; LRG_1203; ENSG00000107643; ENSG00000006062; LRG_1222; ENSG00000282637; ENSG00000050748; ENSG00000065559; ENSG00000166086; LRG_89; ENSG00000100906; ENSG00000109320; LRG_1316; ENSG00000106100; ENSG00000269335; LRG_70; ENSG00000146648; LRG_304; ENSG00000104067; ENSG00000277401","ENSG00000188641; LRG_722; ENSG00000128951; ENSG00000025708; LRG_727; ENSG00000169919; ENSG00000143156; ENSG00000165704; ENSG00000106348; ENSG00000178035; ENSG00000125877; ENSG00000239672; ENSG00000243678; ENSG00000103024; ENSG00000103202; ENSG00000162368; ENSG00000167900; ENSG00000166548; ENSG00000137364; ENSG00000143179; ENSG00000114491; ENSG00000163655; ENSG00000134202; LRG_723; ENSG00000084207; ENSG00000008394; ENSG00000143198; ENSG00000148834","ENSG00000138443; ENSG00000167193; ENSG00000061676; ENSG00000169398; ENSG00000254999; ENSG00000273749; ENSG00000288461; ENSG00000280618; ENSG00000117020; LRG_1396; ENSG00000275199; ENSG00000140285; ENSG00000138448; ENSG00000105855; ENSG00000131504; ENSG00000120063; ENSG00000137809; ENSG00000146648; LRG_304; ENSG00000138675; ENSG00000158186; ENSG00000115414; ENSG00000133818; ENSG00000138685; ENSG00000198121; ENSG00000150760; ENSG00000146535; ENSG00000102606; ENSG00000077782; LRG_993; ENSG00000091409; ENSG00000213949; ENSG00000067900; ENSG00000181104; ENSG00000136238; ENSG00000100485; LRG_92; ENSG00000213281; ENSG00000126458; ENSG00000107562; ENSG00000156299; ENSG00000112290; ENSG00000160293; LRG_309; ENSG00000134853; ENSG00000113721; ENSG00000142208; LRG_721; ENSG00000145431; ENSG00000196914; ENSG00000070831; ENSG00000076928; ENSG00000139734; ENSG00000050820; ENSG00000285460; ENSG00000115904; LRG_754; ENSG00000184009; ENSG00000075624; LRG_132; ENSG00000125503; ENSG00000084112; ENSG00000141298; ENSG00000115020; ENSG00000186111; ENSG00000150867; ENSG00000035403; LRG_383; ENSG00000143398; ENSG00000160007; ENSG00000065534; ENSG00000132155; LRG_413; ENSG00000126934; LRG_750; ENSG00000157764; LRG_299; ENSG00000136950; ENSG00000162704; ENSG00000241553; ENSG00000121879; LRG_310; ENSG00000108518; ENSG00000145675; ENSG00000101335; ENSG00000070087; ENSG00000111229; ENSG00000130429; LRG_1188; ENSG00000163466; ENSG00000140575; ENSG00000118680; ENSG00000101608; ENSG00000147065; ENSG00000137710; ENSG00000205542; ENSG00000165410; LRG_213; ENSG00000172757; ENSG00000130402; ENSG00000282844; LRG_886; ENSG00000072110; ENSG00000089159; ENSG00000148180","ENSG00000184216; ENSG00000198001; LRG_75; ENSG00000090376; ENSG00000107263; LRG_92; ENSG00000213281; ENSG00000169047; ENSG00000117020; LRG_1396; ENSG00000275199; ENSG00000137764; ENSG00000197943; LRG_376; ENSG00000136238; LRG_260; ENSG00000134259; ENSG00000121879; LRG_310; ENSG00000142208; LRG_721; ENSG00000185652; ENSG00000100485; ENSG00000116473; ENSG00000107643; ENSG00000104365; LRG_1241; ENSG00000095015; ENSG00000050748; ENSG00000141522; ENSG00000126934; LRG_750; ENSG00000132155; LRG_413; ENSG00000145675; ENSG00000115904; LRG_754; ENSG00000157764; LRG_299; ENSG00000175104; ENSG00000071242; ENSG00000070831; ENSG00000109339; ENSG00000109458; ENSG00000111252; LRG_621; ENSG00000134313; ENSG00000166681; ENSG00000166225; ENSG00000167193; LRG_89; ENSG00000100906; ENSG00000118689; ENSG00000110851; ENSG00000104312; ENSG00000179222; ENSG00000109320; LRG_1316","ENSG00000005339; ENSG00000103363; ENSG00000154582; ENSG00000117020; LRG_1396; ENSG00000275199; ENSG00000269858; ENSG00000171105; ENSG00000140443; ENSG00000142208; LRG_721; ENSG00000121879; LRG_310; ENSG00000146648; LRG_304; ENSG00000099875; ENSG00000145675; LRG_67; ENSG00000159128; ENSG00000262795; ENSG00000126934; LRG_750; ENSG00000135766; ENSG00000109320; LRG_1316; ENSG00000168610; LRG_112; ENSG00000197943; LRG_376; ENSG00000152256; ENSG00000100387; ENSG00000135930; ENSG00000141959; ENSG00000112715; ENSG00000067057; ENSG00000102144; ENSG00000111321; ENSG00000117394; LRG_1132; ENSG00000154188; ENSG00000111640; ENSG00000091879; ENSG00000100292; ENSG00000134333; ENSG00000288299; ENSG00000111716; ENSG00000074800; LRG_1165; ENSG00000120156; ENSG00000131828; ENSG00000168291","ENSG00000161203; ENSG00000122705; ENSG00000147010; ENSG00000042753; ENSG00000175416; ENSG00000063245; ENSG00000072134; ENSG00000196961; ENSG00000198873; ENSG00000141985; ENSG00000183020; ENSG00000280759; ENSG00000281385; ENSG00000085832; ENSG00000009844; ENSG00000153071; ENSG00000127527; ENSG00000107560; ENSG00000114423; LRG_1179; ENSG00000136631; LRG_1170; ENSG00000141971; ENSG00000196814; LRG_779; ENSG00000163513; ENSG00000181852; ENSG00000175104; ENSG00000123124; ENSG00000069869; ENSG00000108854; ENSG00000198742; ENSG00000284126; ENSG00000078747; LRG_354; ENSG00000140443; ENSG00000176783; ENSG00000284260; ENSG00000024422; ENSG00000146648; LRG_304; LRG_309; ENSG00000134853; ENSG00000115145; ENSG00000136738; ENSG00000175387; ENSG00000105974; ENSG00000105971","ENSG00000269858; ENSG00000167193; ENSG00000109458; ENSG00000107263; ENSG00000136238; ENSG00000105976; LRG_662; ENSG00000135766; ENSG00000126934; LRG_750; ENSG00000157764; LRG_299; ENSG00000070831; ENSG00000132155; LRG_413; ENSG00000005339; ENSG00000116016; ENSG00000121879; LRG_310; ENSG00000145675; ENSG00000117020; LRG_1396; ENSG00000275199; ENSG00000142208; LRG_721; ENSG00000115904; LRG_754; ENSG00000100485; ENSG00000116473; ENSG00000134954; ENSG00000117394; LRG_1132; ENSG00000112715","ENSG00000162704; ENSG00000241553; ENSG00000111229; ENSG00000130429; LRG_1188; ENSG00000163466; ENSG00000167193; ENSG00000109458; ENSG00000150760; ENSG00000115414; ENSG00000177105; ENSG00000089159; ENSG00000136238; ENSG00000121879; LRG_310; ENSG00000145675; ENSG00000169398; ENSG00000071205; ENSG00000136950; ENSG00000112290; ENSG00000035403; LRG_383; ENSG00000070831; ENSG00000050820; ENSG00000285460; ENSG00000105976; LRG_662; ENSG00000075624; LRG_132; ENSG00000184009; ENSG00000197959; ENSG00000044115","ENSG00000117020; LRG_1396; ENSG00000275199; ENSG00000146648; LRG_304; ENSG00000126934; LRG_750; ENSG00000132155; LRG_413; ENSG00000142208; LRG_721; LRG_309; ENSG00000134853; ENSG00000113721; ENSG00000077782; LRG_993; ENSG00000105976; LRG_662; ENSG00000121879; LRG_310; ENSG00000145675; LRG_92; ENSG00000213281; ENSG00000152256; ENSG00000284194; ENSG00000103257; ENSG00000134333; ENSG00000288299; ENSG00000171314; ENSG00000067225; ENSG00000111716; ENSG00000117394; LRG_1132; ENSG00000067057; ENSG00000105281; ENSG00000115419; ENSG00000141959; ENSG00000131828; ENSG00000168291; ENSG00000160211; LRG_148","ENSG00000127884; ENSG00000198130; ENSG00000084754; ENSG00000278540; ENSG00000275176; ENSG00000134333; ENSG00000288299; ENSG00000111716; ENSG00000146085; ENSG00000175198; ENSG00000083123; ENSG00000087008; ENSG00000172340; ENSG00000163541; ENSG00000136143"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>gs_name<\/th>\n      <th>nExpressed<\/th>\n      <th>nDE<\/th>\n      <th>PValue<\/th>\n      <th>adjP<\/th>\n      <th>FDR<\/th>\n      <th>DE<\/th>\n      <th>gene_id<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollY":"450px","columnDefs":[{"className":"dt-right","targets":[2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</div>
<div id="de-on-pseudo-bulk" class="section level1">
<h1>DE on pseudo-bulk</h1>
<p>Raw counts of cells belonging to the same cell type from the same
donor were summed into pseudo-bulk profiles.</p>
<pre class="r"><code>seurat_count &lt;- GetAssayData(object = fibroblast, slot = &#39;counts&#39;)</code></pre>
<pre class="r"><code>minCPM &lt;- 1.5
minSample &lt;- 2</code></pre>
<p>The criteria for a gene to be considered as detected, <span
class="math inline">\(&gt;\)</span> 1.5 counts per million (CPM) were
required to observed in <span class="math inline">\(\geq\)</span> 2
samples.Additionally, only genes with known average transcript length
and GC content were kept.</p>
<pre class="r"><code>genesGR &lt;- genesGR %&gt;%
     as.data.frame() %&gt;%
    .[!duplicated(.$gene_name),]
rownames(genesGR) &lt;- genesGR[,&quot;gene_name&quot;]</code></pre>
<pre class="r"><code>pseudo_by &lt;-  fibroblast[[]] %&gt;%
    droplevels() %&gt;%
    split(f = .$Sample) %&gt;%
    lapply(rownames)
pseudo_bulk &lt;- sapply(names(pseudo_by), function(x){
    seurat_count %&gt;%
        .[,colnames(.) %in% pseudo_by[[x]]] %&gt;%
        rowSums()
}, simplify = FALSE) %&gt;%
    do.call(rbind,.) %&gt;%
    t()
gene2keep &lt;- pseudo_bulk %&gt;%
  edgeR::cpm(log = TRUE) %&gt;%
    is_greater_than(minCPM) %&gt;%
    rowSums() %&gt;%
    is_weakly_greater_than(minSample)
sample_meta &lt;- fibroblast[[]] %&gt;%
    dplyr::select(Sample, Treatment, patient) %&gt;%
    unique() %&gt;%
    set_rownames(NULL) %&gt;%
    droplevels()</code></pre>
<p>Of the 27,381 genes contained in the original
<code>seurat</code>object, 13,682 genes were removed as failing this
criteria for detection, leaving 13,699 genes for downstream
analysis.</p>
<pre class="r"><code>p1 &lt;- pseudo_bulk %&gt;%
  edgeR::cpm(log = TRUE) %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;gene_id&quot;) %&gt;%
  as_tibble() %&gt;%
  pivot_longer(
    cols = -&quot;gene_id&quot;,
    names_to = &quot;Sample&quot;,
    values_to = &quot;logCPM&quot;
  ) %&gt;%
  left_join(sample_meta) %&gt;%
  ggplot(aes(logCPM, stat(density), colour = Sample, linetype = Treatment)) +
  geom_density() +
  labs(
    y = &quot;Density&quot;,
    colour = &quot;Sample&quot;,
    linetype = &quot;Treatment&quot;
  ) +
    scale_color_manual(values = sample_col)
p2 &lt;- pseudo_bulk[gene2keep, ] %&gt;%
  edgeR::cpm(log = TRUE) %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;gene_id&quot;) %&gt;%
  as_tibble() %&gt;%
  pivot_longer(
    cols = -&quot;gene_id&quot;,
    names_to = &quot;Sample&quot;,
    values_to = &quot;logCPM&quot;
  ) %&gt;%
  left_join(sample_meta) %&gt;%
  ggplot(aes(logCPM, stat(density), colour = Sample, linetype = Treatment)) +
  geom_density() +
  labs(
    y = &quot;Density&quot;,
    colour = &quot;Sample&quot;,
    linetype = &quot;Treatment&quot;
  ) +
    scale_color_manual(values = sample_col)
plot_grid(
    plot_grid(
    p1 + theme(legend.position = &quot;none&quot;), 
    p2 + theme(legend.position = &quot;none&quot;), 
    labels = c(&quot;A&quot;, &quot;B&quot;),
    nrow = 1), 
    get_legend(p1), 
    nrow = 1, 
    rel_widths = c(6,1))</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/Differential_Expression.Rmd/plotDensities-1.png" alt="*Distributions of logCPM values on pseudo-bulk counts, A) before and B) after filtering of undetectable genes.*" width="1344" />
<p class="caption">
<em>Distributions of logCPM values on pseudo-bulk counts, A) before and
B) after filtering of undetectable genes.</em>
</p>
</div>
<pre class="r"><code>pseudo_bulk &lt;- pseudo_bulk[gene2keep, ]
geneWithInfo &lt;- rownames(pseudo_bulk) %in% rownames(genesGR)
pseudo_bulk &lt;- pseudo_bulk[geneWithInfo, ]</code></pre>
<p>A <code>DGEList</code> was created using the filtered counts and
sample metadata.</p>
<pre class="r"><code>dge &lt;- pseudo_bulk %&gt;%
        DGEList(
            samples = sample_meta %&gt;%
                set_rownames(.$Sample) %&gt;%
                .[match(colnames(pseudo_bulk), rownames(.)),],
            genes = genesGR[rownames(.),]
        ) %&gt;%
  calcNormFactors()</code></pre>
<div id="library-sizes" class="section level3">
<h3>Library Sizes</h3>
<p>Due to the differences in the numbers of cells derived from each
sample seen earlier, there were large differences in the library sizes
of pseudo-bulk samples too.</p>
<pre class="r"><code>dge$samples %&gt;%
    ggplot(aes(Sample, lib.size, fill = Sample)) +
    geom_col() +
    geom_hline(yintercept = 1e7, linetype = 2) +
    scale_y_continuous(
        labels = comma, expand = expansion(c(0, 0.05))
    ) +
    scale_fill_manual(values= sample_col) +
    labs(x = &quot;Sample Name&quot;, y = &quot;Library Size&quot;, fill = &quot;Sample&quot;)  +
    theme(legend.position = &quot;top&quot;)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/Differential_Expression.Rmd/plotLibSizes-1.png" alt="*Library sizes of all samples after removal of undetectable genes. The common-use minimum library size of 10 million reads is shown as a dashed line.*" width="1152" />
<p class="caption">
<em>Library sizes of all samples after removal of undetectable genes.
The common-use minimum library size of 10 million reads is shown as a
dashed line.</em>
</p>
</div>
</div>
<div id="pca-pre-normalisation" class="section level3">
<h3>PCA: Pre Normalisation</h3>
<pre class="r"><code>pca &lt;- dge %&gt;%
  edgeR::cpm(log = TRUE) %&gt;%
  t() %&gt;%
  prcomp() </code></pre>
<pre class="r"><code>broom::tidy(pca) %&gt;% 
  dplyr::filter(PC %in% 1:2) %&gt;% 
  dplyr::rename(Sample = row) %&gt;% 
  left_join(dge$samples) %&gt;% 
  pivot_wider(names_from = &quot;PC&quot;, values_from = &quot;value&quot;, names_prefix = &quot;PC&quot;) %&gt;%
  ggplot(
    aes(PC1, PC2, colour = Treatment)
  ) +
  geom_point(size = 2) +
  geom_segment(
    aes(x = `PC1_VEH`, y = `PC2_VEH`, xend = `PC1_DHT`, yend = `PC2_DHT`),
    data = . %&gt;% 
      dplyr::select(patient, Treatment,  PC1, PC2) %&gt;%
      pivot_wider(names_from = &quot;Treatment&quot;, values_from = c(&quot;PC1&quot;, &quot;PC2&quot;)) ,
    inherit.aes = FALSE,
    arrow = arrow(length = unit(0.02, &quot;npc&quot;), type = &quot;closed&quot;),
    colour = &quot;grey30&quot;
  ) +
  geom_label_repel(
    aes(label = patient),
    color = &quot;black&quot;,
    fill = rgb(1, 1, 1, 0.5),
    size = 5,
    data = . %&gt;% 
      group_by(patient) %&gt;% 
      summarise(PC1 = mean(PC1), PC2 = mean(PC2), .groups = &quot;drop&quot;)
  ) +
  labs(
    x = glue(&quot;PC1 ({percent(summary(pca)$importance[2, 1], 0.1)})&quot;),
    y = glue(&quot;PC2 ({percent(summary(pca)$importance[2, 2], 0.1)})&quot;)
  ) +
  scale_colour_manual(values = treat_col) </code></pre>
<div class="figure" style="text-align: center">
<img src="figure/Differential_Expression.Rmd/unnamed-chunk-24-1.png" alt="*Pre-normalisation PCA plot with arrows pointing from treated to corresponding vehicle sample.The 2 arrows were pointing at completely opposite directions.*" width="960" />
<p class="caption">
<em>Pre-normalisation PCA plot with arrows pointing from treated to
corresponding vehicle sample.The 2 arrows were pointing at completely
opposite directions.</em>
</p>
</div>
</div>
<div id="normalisation" class="section level3">
<h3>Normalisation</h3>
<pre class="r"><code>cqn &lt;-  with(
    dge,
    cqn(
        counts= counts,
        x = genes$gc_content,
        lengths = genes$ave_tx_len
    )
)
dge$offset &lt;- cqn$glm.offset
logCPM &lt;- cqn$y + cqn$offset
pcaPost &lt;- logCPM %&gt;%
    t() %&gt;%
    prcomp() </code></pre>
<pre class="r"><code>pca_post_plot &lt;- broom::tidy(pcaPost) %&gt;% 
  dplyr::filter(PC %in% 1:2) %&gt;% 
  dplyr::rename(Sample = row) %&gt;% 
  left_join(dge$samples) %&gt;% 
  pivot_wider(names_from = &quot;PC&quot;, values_from = &quot;value&quot;, names_prefix = &quot;PC&quot;) %&gt;%
  ggplot(
    aes(PC1, PC2, colour = Treatment)
  ) +
  geom_point(size = 4) +
  geom_segment(
    aes(x = `PC1_VEH`, y = `PC2_VEH`, xend = `PC1_DHT`, yend = `PC2_DHT`),
    data = . %&gt;% 
      dplyr::select(patient, Treatment,  PC1, PC2) %&gt;%
      pivot_wider(names_from = &quot;Treatment&quot;, values_from = c(&quot;PC1&quot;, &quot;PC2&quot;)) ,
    inherit.aes = FALSE,
    arrow = arrow(length = unit(0.05, &quot;npc&quot;), type = &quot;closed&quot;),
    colour = &quot;grey30&quot;
  ) +
  geom_label_repel(
    aes(label = patient),
    color = &quot;black&quot;,
    fill = rgb(1, 1, 1, 0.5),
    size = 5,
    data = . %&gt;% 
      group_by(patient) %&gt;% 
      summarise(PC1 = mean(PC1), PC2 = mean(PC2), .groups = &quot;drop&quot;)
  ) +
  labs(
    x = glue(&quot;PC1 ({percent(summary(pca)$importance[2, 1], 0.1)})&quot;),
    y = glue(&quot;PC2 ({percent(summary(pca)$importance[2, 2], 0.1)})&quot;)
  ) +
  scale_colour_manual(values = treat_col) 
pca_post_plot</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/Differential_Expression.Rmd/unnamed-chunk-25-1.png" alt="*Post-normalisation PCA plot with arrows pointing from treated to corresponding vehicle sample.The 2 arrows were still pointing at completely opposite directions even after normalisation.*" width="960" />
<p class="caption">
<em>Post-normalisation PCA plot with arrows pointing from treated to
corresponding vehicle sample.The 2 arrows were still pointing at
completely opposite directions even after normalisation.</em>
</p>
</div>
</div>
<div id="edger" class="section level3">
<h3>edgeR</h3>
<p>Average treatment response was tested through fitting
quasi-likelihood negative binomial GLMs through <code>edgeR</code>.
Model matrix in the form of
<code>model.matrix(~ 0 + patient + Treatment, data = dge$samples)</code>
was constructed.</p>
<pre class="r"><code>X &lt;- model.matrix(~ 0 + patient + Treatment,
                  data = dge$samples %&gt;%
                      mutate(Treatment = factor(Treatment, levels = c(&quot;VEH&quot;, &quot;DHT&quot;)))
) %&gt;%
    set_colnames(str_remove_all(colnames(.), &quot;patient|Treatment&quot;)) %&gt;%
    .[,colSums(.) != 0]  
dge &lt;- estimateDisp(dge, design = X, robust = TRUE)
fit &lt;- glmQLFit(dge)</code></pre>
<pre class="r"><code>alpha &lt;- 0.05
topTable &lt;- glmQLFTest(fit, coef = &quot;DHT&quot;) %&gt;%
      topTags(n = Inf) %&gt;%
      .[[&quot;table&quot;]] %&gt;%
      as_tibble() %&gt;%
      mutate(
        location = paste0(seqnames, &quot;:&quot;, start, &quot;-&quot;, end, &quot;:&quot;, strand),
        rankingStat = -sign(logFC)*log10(PValue),
        signedRank = rank(rankingStat),
        DE = FDR &lt; alpha
      ) %&gt;%
      dplyr::select(
        gene_id, gene_name, logCPM, logFC, PValue, FDR, 
        location, gene_biotype, entrezid, ave_tx_len, gc_content,
        rankingStat, signedRank, DE
      )
DEGs &lt;- topTable %&gt;%
    dplyr::filter(DE)</code></pre>
<p>With this pseudo-bulk + edgeR approach, only 1 gene had a FDR &lt;
0.05, out of the 13544 genes tested.</p>
<pre class="r"><code>topTable %&gt;%
    dplyr::filter(DE) %&gt;%
    mutate_at(vars(c(&quot;PValue&quot;, &quot;FDR&quot;)), formatP) %&gt;%
    mutate_at(vars(c(&quot;logCPM&quot;, &quot;logFC&quot;)), round, 4) %&gt;%
    dplyr::select(`Gene name` = gene_name, 
                  `Gene ID` = gene_id, 
                  logCPM, logFC, PValue, 
                  FDR , DE) %&gt;%
    mutate_if(is.character, as.factor) %&gt;%
    datatable(
       filter = &quot;top&quot;, 
       caption = htmltools::tags$caption(
            htmltools::em(
                &quot;Gene that was differentially expressed post-treatment.&quot;)
        )
    ) %&gt;% 
    formatStyle(
        &#39;logFC&#39;, 
        color = styleInterval(0, c(&#39;blue&#39;, &quot;red&quot;))
    )</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-307bf623007724e05f1e" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-307bf623007724e05f1e">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td><\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\" disabled=\"\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;RASD2&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\" disabled=\"\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;ENSG00000100302&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\" disabled=\"\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0\" data-max=\"1\" data-scale=\"3\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\" disabled=\"\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0\" data-max=\"1\" data-scale=\"4\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\" disabled=\"\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;2.91e-06&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\" disabled=\"\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;0.0395&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"logical\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\" disabled=\"\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;true&quot;,&quot;false&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n<\/tr>","caption":"<caption>\n  <em>Gene that was differentially expressed post-treatment.<\/em>\n<\/caption>","data":[["1"],["RASD2"],["ENSG00000100302"],[2.103],[1.4008],["2.91e-06"],["0.0395"],[true]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Gene name<\/th>\n      <th>Gene ID<\/th>\n      <th>logCPM<\/th>\n      <th>logFC<\/th>\n      <th>PValue<\/th>\n      <th>FDR<\/th>\n      <th>DE<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'color':isNaN(parseFloat(value)) ? '' : value <= 0 ? \"blue\" : \"red\"});\n}"},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
<p>This gene RASD2 was not an AR signature.</p>
<pre class="r"><code># AR_sig %&gt;%
#     dplyr::filter(gene_name == DEGs$gene_name)</code></pre>
<pre class="r"><code>bp_basd2 &lt;- logCPM %&gt;%
    .[&quot;RASD2&quot;,] %&gt;%
    enframe(
        name = &quot;Sample&quot;, 
        value = &quot;logCPM&quot;
    ) %&gt;%
    left_join(sample_meta) %&gt;%
    ggplot(
        aes(Treatment, logCPM, color = Treatment )
    ) +
    geom_boxplot() +
    geom_point(size = 6) +
    scale_color_manual(values = treat_col) 
bp_basd2</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/Differential_Expression.Rmd/unnamed-chunk-28-1.png" alt="*LogCPM of the only differentially expressed gene detected using the pseudo-bulk approach RASD2*" width="960" />
<p class="caption">
<em>LogCPM of the only differentially expressed gene detected using the
pseudo-bulk approach RASD2</em>
</p>
</div>
</div>
<div id="knott-pseudo-bulk-signature" class="section level3">
<h3>Knott pseudo-bulk signature</h3>
<p>Differential expression analysis using the pseudo-bulk + edgeR
approach was also performed on the cis-female and trans-female breast
snRNA-seq data .</p>
<pre class="r"><code>knott_pseudo_de &lt;- readRDS(&quot;~/20131906_HickeyT_JC_NormalBreast/output/knott_pseudo_de.rds&quot;)
knott_pseudo_de &lt;- knott_pseudo_de$Fibroblast </code></pre>
<p>53 DEGs were detected with a FDR cut-off of 0.05.</p>
<pre class="r"><code>fibro_2RS_pseudo &lt;- topTable %&gt;%
    dplyr::select(gene_name, pilot = rankingStat) %&gt;%
    left_join(
        knott_pseudo_de %&gt;%
            dplyr::select(gene_name , knott = rankingStat)
    ) %&gt;%
    drop_na()
fibro_2RS_cor_pseudo &lt;- cor.test(fibro_2RS_pseudo$pilot, fibro_2RS_pseudo$knott)</code></pre>
<p>Correlation between the ranking statistics derived from the two
independent studies was 0.085 with a p-value of 1.0207975^{-22}.</p>
<pre class="r"><code>cor_plot_pseudo &lt;- fibro_2RS_pseudo %&gt;%
    mutate(aveRS = (pilot + knott)/2) %&gt;%
    .[order(abs(.$aveRS), decreasing = TRUE),] %&gt;%
    mutate(RANK = row_number()) %&gt;%
    ggplot(
        aes(pilot , knott)
    ) +
    geom_point() +
    geom_text_repel(
        aes(label = gene_name), 
        data = . %&gt;% 
            dplyr::filter(
                RANK &lt; 50
            ), 
        color = &quot;red&quot;
    ) +
  geom_vline(
    xintercept = 0,
    linetype = 2,
    size = 1.5,
    colour = &quot;red&quot;
  ) +
  geom_hline(
    yintercept = 0,
    linetype = 2,
    size = 1.5,
    colour = &quot;red&quot;
  ) +
    labs(
        x = &quot;In vitro fibroblast DHT vs Veh ranking stat&quot;, 
        y = &quot;TM vs CF fibroblast ranking stat&quot;, 
        # title = paste(&quot;Correlation between logFCs between the 2 study:&quot;, 
        #                 round(cor(fibro_2R$pilot, fibro_2FC$knott), digits = 2), 
        #                 sep = &quot; &quot;)
    ) +
    theme(
        axis.title  = element_text(size = 16)
    )
cor_plot_pseudo</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/Differential_Expression.Rmd/unnamed-chunk-31-1.png" alt="*Comparison of ranking statistics derived from the knott fibroblast cells against this pilot study. Genes with top 50 ranked mean absolute ranking statistics were labelled.*" width="1344" />
<p class="caption">
<em>Comparison of ranking statistics derived from the knott fibroblast
cells against this pilot study. Genes with top 50 ranked mean absolute
ranking statistics were labelled.</em>
</p>
</div>
</div>
</div>
<div id="thesis-figure" class="section level1">
<h1>Thesis figure</h1>
<p>Code for reproducing figures used in the theis:</p>
<pre class="r"><code># thesis chapter 5 table
zlm_summary_mixed %&gt;%
    dplyr::filter(
        FDR &lt; 0.01, 
        gene_name %in% AR_sig$gene_name
    )  %&gt;%
    left_join(
        AR_sig 
    ) %&gt;%
    mutate(
        Align = case_when(
            logFC &lt; 0 &amp; AR_sig == &quot;Down&quot; ~ &quot;\\color{teal}✓&quot;,
            logFC &gt; 0 &amp; AR_sig == &quot;Up&quot; ~ &quot;\\color{teal}✓&quot;,
            logFC &lt; 0 &amp; AR_sig == &quot;Up&quot; ~ &quot;✗&quot;,
            logFC &gt; 0 &amp; AR_sig == &quot;Down&quot; ~ &quot;✗&quot;
        ), 
        AR_sig = ifelse(
            AR_sig == &quot;Down&quot;, &quot;\\color{blue}Down&quot;, &quot;\\color{red}Up&quot;
        )
    ) %&gt;%
    mutate_if(is.character, as.factor) %&gt;%
    .[order(.$FDR),] %&gt;%
    mutate_at(vars(c(&quot;Pr(&gt;Chisq)&quot;, &quot;FDR&quot;)), formatP) %&gt;%
    mutate(logFC = as.character(round(logFC, 4))) %&gt;%
    dplyr::select(
        `Gene Name` = gene_name,
        `Expected Direction` = AR_sig,
        logFC,FDR, Concordant = Align
    ) %&gt;%
    xtable::xtable(
        caption = &quot;AR signatures defined in ER-positive BC\\cite{hickey_androgen_2021} that were also differentially expressed upon DHT treatment in mammary fibroblasts. The concordant column reflects whether the signs of logFCs in fibroblasts agreed with the signature types in malignant epithelial cells (ie. expected direction).&quot;, 
        label = &quot;chap5_table3&quot;
    )  %&gt;%
     xtable::print.xtable(
        &quot;~/PhD_thesis/draft_figure/chapter_05/table3.tex&quot;, type = &quot;latex&quot;,
        caption.placement = &quot;top&quot;,
        include.rownames = FALSE,
        sanitize.text.function = function(x) x)</code></pre>
<pre class="r"><code>pca_post_plot &lt;- broom::tidy(pcaPost) %&gt;% 
  dplyr::filter(PC %in% 1:2) %&gt;% 
  dplyr::rename(Sample = row) %&gt;% 
  left_join(dge$samples) %&gt;% 
  pivot_wider(names_from = &quot;PC&quot;, values_from = &quot;value&quot;, names_prefix = &quot;PC&quot;) %&gt;%
  ggplot(
    aes(PC1, PC2, colour = Treatment)
  ) +
  geom_point(size = 8) +
  geom_segment(
    aes(x = `PC1_VEH`, y = `PC2_VEH`, xend = `PC1_DHT`, yend = `PC2_DHT`),
    data = . %&gt;% 
      dplyr::select(patient, Treatment,  PC1, PC2) %&gt;%
      pivot_wider(names_from = &quot;Treatment&quot;, values_from = c(&quot;PC1&quot;, &quot;PC2&quot;)) ,
    inherit.aes = FALSE,
    arrow = arrow(length = unit(0.05, &quot;npc&quot;), type = &quot;closed&quot;),
    colour = &quot;grey30&quot;
  ) +
  geom_label_repel(
    aes(label = patient),
    color = &quot;black&quot;,
    fill = rgb(1, 1, 1, 0.5),
    size = 18,
    data = . %&gt;% 
      group_by(patient) %&gt;% 
      summarise(PC1 = mean(PC1), PC2 = mean(PC2), .groups = &quot;drop&quot;)
  ) +
  labs(
    x = glue(&quot;PC1 ({percent(summary(pca)$importance[2, 1], 0.1)})&quot;),
    y = glue(&quot;PC2 ({percent(summary(pca)$importance[2, 2], 0.1)})&quot;)
  ) +
  scale_colour_manual(values = treat_col) </code></pre>
<pre class="r"><code>chap5_fig7 &lt;- cor_plot &amp;
    theme(
        text = element_text(size = 50)
    )
# png(
#     &quot;/Users/wenjunliu/PhD_thesis/Images/Chapter_05/chap5_fig7.png&quot;,
#     height = 1600,
#     width = 2000
# )
#chap5_fig7
# dev.off()</code></pre>
<pre class="r"><code>chap5_fig8 &lt;- (pca_post_plot | bp_basd2 + theme(legend.position = &quot;none&quot;)) + 
    plot_layout(guides = &quot;collect&quot;) +
    plot_annotation(tag_levels = &#39;A&#39;) &amp;
    theme(
        plot.margin = unit(rep(1, 4), &quot;cm&quot;),
        text = element_text(size = 50),
        # axis.text.x = element_text(size = 50),
        plot.tag = element_text(size = 55, face = &quot;bold&quot; ), 
        legend.key.size = unit(2, &quot;cm&quot;)
    )
# png(
#     &quot;/Users/wenjunliu/PhD_thesis/Images/Chapter_05/chap5_fig8.png&quot;,
#     height = 1200,
#     width = 2400
# )
#chap5_fig8
# dev.off()</code></pre>
<pre class="r"><code># # supp_table3
# library(xlsx)</code></pre>
<pre class="r"><code>## Table 4 (top 10 most enriched pathways)
goseq_kg %&gt;%
    dplyr::filter(FDR &lt; 0.05) %&gt;%
    mutate(
        gs_name = str_remove_all(gs_name, &quot;kegg.&quot;)
    ) %&gt;%
    dplyr::select(
        `Gene-set name` = gs_name,
        FDR = adjP,
        `Number of DE pathway gene` = nDE,
        `Number of expressed pathway gene` = nExpressed
    ) %&gt;%
    .[1:10,] %&gt;%
    xtable::xtable(
        caption = &quot;Top 10 KEGG pathways significantly enriched among DEGs detected among DHT-treated mammary fibroblasts. Top 10 KEGG pathways significantly enriched among DEGs detected using GOseq between DHT-treated vs vehicle fibroblasts. Gene length was used as an offset term to account for any bias in the enrichment analysis. Pathways with a FDR \\textless 0.05 were considered to be significantly enriched.&quot;, 
        label = &quot;chap5_table4&quot;
    )  %&gt;%
     xtable::print.xtable(
        &quot;~/PhD_thesis/draft_figure/chapter_05/table4.tex&quot;, type = &quot;latex&quot;,
        caption.placement = &quot;top&quot;,
        include.rownames = FALSE,
        sanitize.text.function = function(x) x)</code></pre>
<pre class="r"><code>source(here::here(&quot;analysis/plot_commu_temp.R&quot;))
library(reshape2)
library(ggraph)
library(igraph)</code></pre>
<pre class="r"><code>set.seed(123)
chap5_fig9 &lt;- goseq_kg %&gt;%
    dplyr::filter(FDR &lt; 0.05) %&gt;%
    mutate(pvalue = as.numeric(PValue)) %&gt;%
    plot_community_temp(
        normalisedScores = .,
        gsTopology = gsTopology,
        colorBy = &quot;pvalue&quot;,
        color_lg_title = &quot;-log10(P-value)&quot;,
        lb_size = 16,
        scale_nodeSize = c(10, 14),
        label.fontsize = 50,
         label.buffer = unit(60, &quot;mm&quot;),
    ) +
    theme(
        panel.border = element_blank(),
        legend.key.size = unit(20, &quot;mm&quot;),
        text = element_text(size = 40)
    )
# png(
#     &quot;/Users/wenjunliu/PhD_thesis/Images/Chapter_05/chap5_fig9.png&quot;,
#     height = 3200,
#     width = 3800
# )
#chap5_fig9
# dev.off()</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21)
Platform: x86_64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.3.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Australia/Adelaide
tzcode source: internal

attached base packages:
[1] splines   stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] igraph_1.4.2                ggraph_2.1.0               
 [3] reshape2_1.4.4              sSNAPPY_1.4.1              
 [5] goseq_1.52.0                geneLenDataBase_1.36.0     
 [7] BiasedUrn_2.0.9             msigdbr_7.5.1              
 [9] UpSetR_1.4.0                BiocParallel_1.34.1        
[11] pheatmap_1.0.12             MAST_1.26.0                
[13] SingleCellExperiment_1.22.0 SummarizedExperiment_1.30.1
[15] MatrixGenerics_1.12.0       matrixStats_0.63.0         
[17] ggforce_0.4.1               randomcoloR_1.1.0.1        
[19] ggpubr_0.6.0                colorspace_2.1-0           
[21] patchwork_1.1.2             plotly_4.10.1              
[23] corrplot_0.92               SeuratObject_4.1.3         
[25] Seurat_4.3.0                DT_0.27                    
[27] ggrepel_0.9.3               cqn_1.46.0                 
[29] quantreg_5.95               SparseM_1.81               
[31] preprocessCore_1.62.1       nor1mix_1.3-0              
[33] mclust_6.0.0                magrittr_2.0.3             
[35] ggfortify_0.4.16            cowplot_1.1.1              
[37] ensembldb_2.24.0            AnnotationFilter_1.24.0    
[39] GenomicFeatures_1.52.0      AnnotationDbi_1.62.1       
[41] Biobase_2.60.0              GenomicRanges_1.52.0       
[43] GenomeInfoDb_1.36.0         IRanges_2.34.0             
[45] S4Vectors_0.38.1            AnnotationHub_3.8.0        
[47] BiocFileCache_2.8.0         dbplyr_2.3.2               
[49] BiocGenerics_0.46.0         edgeR_3.42.2               
[51] limma_3.56.1                glue_1.6.2                 
[53] pander_0.6.5                scales_1.2.1               
[55] yaml_2.3.7                  lubridate_1.9.2            
[57] forcats_1.0.0               stringr_1.5.0              
[59] dplyr_1.1.2                 purrr_1.0.1                
[61] readr_2.1.4                 tidyr_1.3.0                
[63] tibble_3.2.1                ggplot2_3.4.2              
[65] tidyverse_2.0.0             workflowr_1.7.0            

loaded via a namespace (and not attached):
  [1] fs_1.6.2                      ProtGenerics_1.32.0          
  [3] spatstat.sparse_3.0-1         bitops_1.0-7                 
  [5] httr_1.4.6                    RColorBrewer_1.1-3           
  [7] backports_1.4.1               tools_4.3.0                  
  [9] sctransform_0.3.5             utf8_1.2.3                   
 [11] R6_2.5.1                      mgcv_1.8-42                  
 [13] lazyeval_0.2.2                uwot_0.1.14                  
 [15] withr_2.5.0                   sp_1.6-0                     
 [17] graphite_1.46.0               prettyunits_1.1.1            
 [19] gridExtra_2.3                 progressr_0.13.0             
 [21] cli_3.6.1                     spatstat.explore_3.1-0       
 [23] labeling_0.4.2                sass_0.4.6                   
 [25] spatstat.data_3.0-1           ggridges_0.5.4               
 [27] pbapply_1.7-0                 Rsamtools_2.16.0             
 [29] parallelly_1.35.0             rstudioapi_0.14              
 [31] RSQLite_2.3.1                 generics_0.1.3               
 [33] BiocIO_1.10.0                 crosstalk_1.2.0              
 [35] ica_1.0-3                     spatstat.random_3.1-4        
 [37] car_3.1-2                     GO.db_3.17.0                 
 [39] Matrix_1.5-4                  fansi_1.0.4                  
 [41] abind_1.4-5                   lifecycle_1.0.3              
 [43] whisker_0.4.1                 carData_3.0-5                
 [45] Rtsne_0.16                    grid_4.3.0                   
 [47] blob_1.2.4                    promises_1.2.0.1             
 [49] crayon_1.5.2                  miniUI_0.1.1.1               
 [51] lattice_0.21-8                KEGGREST_1.40.0              
 [53] pillar_1.9.0                  knitr_1.42                   
 [55] rjson_0.2.21                  future.apply_1.10.0          
 [57] codetools_0.2-19              leiden_0.4.3                 
 [59] getPass_0.2-2                 V8_4.3.0                     
 [61] data.table_1.14.8             vctrs_0.6.2                  
 [63] png_0.1-8                     gtable_0.3.3                 
 [65] cachem_1.0.8                  xfun_0.39                    
 [67] S4Arrays_1.0.4                mime_0.12                    
 [69] tidygraph_1.2.3               survival_3.5-5               
 [71] statmod_1.5.0                 interactiveDisplayBase_1.38.0
 [73] ellipsis_0.3.2                fitdistrplus_1.1-11          
 [75] ROCR_1.0-11                   nlme_3.1-162                 
 [77] bit64_4.0.5                   progress_1.2.2               
 [79] filelock_1.0.2                RcppAnnoy_0.0.20             
 [81] rprojroot_2.0.3               bslib_0.4.2                  
 [83] irlba_2.3.5.1                 KernSmooth_2.23-21           
 [85] DBI_1.1.3                     tidyselect_1.2.0             
 [87] processx_3.8.1                bit_4.0.5                    
 [89] compiler_4.3.0                curl_5.0.0                   
 [91] git2r_0.32.0                  graph_1.78.0                 
 [93] xml2_1.3.4                    DelayedArray_0.26.2          
 [95] rtracklayer_1.60.0            lmtest_0.9-40                
 [97] callr_3.7.3                   rappdirs_0.3.3               
 [99] digest_0.6.31                 goftest_1.2-3                
[101] spatstat.utils_3.0-3          rmarkdown_2.21               
[103] XVector_0.40.0                htmltools_0.5.5              
[105] pkgconfig_2.0.3               highr_0.10                   
[107] fastmap_1.1.1                 rlang_1.1.1                  
[109] htmlwidgets_1.6.2             shiny_1.7.4                  
[111] farver_2.1.1                  jquerylib_0.1.4              
[113] zoo_1.8-12                    jsonlite_1.8.4               
[115] RCurl_1.98-1.12               GenomeInfoDbData_1.2.10      
[117] munsell_0.5.0                 Rcpp_1.0.10                  
[119] viridis_0.6.3                 babelgene_22.9               
[121] reticulate_1.28               stringi_1.7.12               
[123] zlibbioc_1.46.0               MASS_7.3-60                  
[125] org.Hs.eg.db_3.17.0           plyr_1.8.8                   
[127] parallel_4.3.0                listenv_0.9.0                
[129] deldir_1.0-6                  graphlayouts_1.0.0           
[131] Biostrings_2.68.0             tensor_1.5                   
[133] hms_1.1.3                     locfit_1.5-9.7               
[135] ps_1.7.5                      spatstat.geom_3.2-1          
[137] ggsignif_0.6.4                biomaRt_2.56.0               
[139] BiocVersion_3.17.1            XML_3.99-0.14                
[141] evaluate_0.21                 BiocManager_1.30.20          
[143] tweenr_2.0.2                  tzdb_0.3.0                   
[145] httpuv_1.6.10                 MatrixModels_0.5-1           
[147] RANN_2.6.1                    polyclip_1.10-4              
[149] future_1.32.0                 scattermore_1.0              
[151] broom_1.0.4                   xtable_1.8-4                 
[153] restfulr_0.0.15               rstatix_0.7.2                
[155] later_1.3.1                   viridisLite_0.4.2            
[157] memoise_2.0.1                 GenomicAlignments_1.36.0     
[159] cluster_2.1.4                 timechange_0.2.0             
[161] globals_0.16.2                here_1.0.1                   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
