---
title: "Differential_Expression"
output: html_document
author: "Wenjun Liu<br>Dame Roma Mitchell Cancer Research Laboratories<br>Adelaide Medical School<br>University of Adelaide"
date: "2023-01-18"
---
```{r setup, echo=FALSE}
knitr::opts_chunk$set(
  echo = TRUE,
  message = FALSE,
  warning = FALSE,
  fig.align = "center",
  fig.width = 10
)
```

```{r packages}
library(tidyverse)
library(yaml)
library(scales)
library(pander)
library(glue)
library(edgeR)
library(AnnotationHub)
library(ensembldb)
library(cowplot)
library(ggfortify)
library(magrittr)
library(cqn)
library(ggrepel)
library(DT)
library(Seurat)
library(corrplot)
library(plotly)
library(patchwork)
library(colorspace)
library(ggpubr)
library(randomcoloR)
library(ggforce)
library(MAST)
library(pheatmap)
library(BiocParallel)
library(UpSetR)
library(msigdbr)
library(goseq)
library(sSNAPPY)
library(AUCell)
library(grid)
```

```{r options}
panderOptions("table.split.table", Inf)
panderOptions("big.mark", ",")
theme_set(theme_bw())
```

```{r formatP}
formatP <- function(p, m = 0.0001){
out <- rep("", length(p))
out[p < m] <- sprintf("%.2e", p[p<m])
out[p >= m] <- sprintf("%.4f", p[p>=m])
out
}
```

## Gene Annotations

Firstly, gene annotations were retrieved from Ensembl Release 101. 

```{r ah}
ah <- AnnotationHub() %>%
  subset(rdataclass == "EnsDb") %>%
  subset(str_detect(description, "101")) %>%
  subset(genome == "GRCh38")
stopifnot(length(ah) == 1)
```

```{r ensDb}
ensDb <- ah[[1]]
genesGR <- genes(ensDb)
transGR <- transcripts(ensDb)
```

```{r addTxLen}
mcols(transGR) <- mcols(transGR) %>%
  cbind(
    transcriptLengths(ensDb)[rownames(.), c("nexon", "tx_len")]
  )
```

```{r addGcLen2Genes}
mcols(genesGR) <- mcols(genesGR) %>%
  as.data.frame() %>%
  dplyr::select(
    gene_id, gene_name, gene_biotype, entrezid
  ) %>%
  left_join(
    mcols(transGR) %>%
      as.data.frame() %>%
      mutate(
        tx_support_level = case_when(
          is.na(tx_support_level) ~ 1L, 
          TRUE ~ tx_support_level
        )
      ) %>%
      group_by(gene_id) %>%
      summarise(
        n_tx = n(),
        longest_tx = max(tx_len),
        ave_tx_len = mean(tx_len),
        gc_content = sum(tx_len*gc_content) / sum(tx_len)
      ) %>%
      mutate(
        bin_length = cut(
          x = ave_tx_len,
          labels = seq_len(10),
          breaks = quantile(ave_tx_len, probs = seq(0, 1, length.out = 11)),
          include.lowest = TRUE
        ),
        bin_gc = cut(
          x = gc_content,
          labels = seq_len(10),
          breaks = quantile(gc_content, probs = seq(0, 1, length.out = 11)),
          include.lowest = TRUE
        ),
        bin = paste(bin_gc, bin_length, sep = "_")
      ),
    by = "gene_id"
  ) %>%
  set_rownames(.$gene_id) %>%
  as("DataFrame")
```

# Read in seurat object

The pre-processed `seurat` object saved in the pre-processing step was loaded in. 

```{r}
fibroblast <- readRDS(here::here("output/fibroblast_seurat.rds"))
# fibroblast[[]] %>%
#     rownames_to_column("cell") %>%
#     mutate(
#         label = vapply(cell, function(x){
#             str_split(x, "-")[[1]][2]
#         }, character(1))
#     ) %>%
#     split(f = .$orig.ident) %>%
#     lapply(pull, label) %>%
#     lapply(unique)
```

Color palettes for the patient and treatment levels are set up. 

```{r}
# Color palettes used by most scRNA-seq methods could be found [here](https://rdrr.io/bioc/scater/src/R/plot_colours.R). 
patient_col <- c(
    "MF61" = "#A71B4B", 
    "MF43" = "#584B9F"
)
# scanpy colors were found in https://github.com/scverse/scanpy/blob/034ca2823804645e0d4874c9b16ba2eb8c13ac0f/scanpy/plotting/palettes.py
treat_col <- c(
  VEH = rgb(0.7, 0.7, 0.7),
  DHT = rgb(0.8, 0.2, 0.2))
sample_col <- hcl.colors(n = 4) %>%
    set_names(c("mf43DHT", "mf43VEH", "mf61DHT", "mf61VEH"))
```

There are two different strategies that can be used to perform differential expression analysis in scRNA-seq data: 
1. Global DE: performing DE on all DHT-treated cells against all vehicle treated cells. 
2. Cluster-specific DE: performing DE on DHT-treated cells against vehicle treated cells within each cluster. 

Both strategies will be applied to this dataset in this analysis. 

# DE using MAST

## Global DE

Firstly, differential expression analysis was perform using `MAST` to compare all DHT-treated samples against the all vehicle while nested within patients. 

`sca` objects were created using `MAST::FromMatrix()` function and the actual linear mixed hurdle model fittings were done on the DRMCRL workstation.

Note that the continuous part of the hurdle models is conditional of a gene being expressed, so if a gene had non-zero count in only one Cell Type + Type group, the models won't work. The `FindMarkers()` function in `Seruat` solves this by including a `min.pct` parameter so only genes that were detected in a minimum fraction of `min.pct` cells in either of the two populations would be tested. [The developer of `MAST` package also recommended this strategy](https://github.com/RGLab/MAST/issues/146) so only genes that had non-zero log-norm counts in at least 10% of vehicle or DHT-treated cells were tested. 

```{r sca}
# create SCA object
seurat_data <- GetAssayData(object = fibroblast, slot = 'data')
VEH_cell <- fibroblast[[]] %>%
        dplyr::filter(Treatment == "VEH") %>%
        rownames()
gene2keep_VEH <- rowSums(seurat_data[, colnames(seurat_data) %in% VEH_cell] > 0) > (0.1 * length(VEH_cell))
DHT_cell <- fibroblast[[]] %>%
        dplyr::filter(Treatment == "DHT") %>%
        rownames()
gene2keep_DHT <- rowSums(seurat_data[, colnames(seurat_data) %in% DHT_cell] > 0) > (0.1 * length(DHT_cell))
seurat_data <- seurat_data[gene2keep_DHT|gene2keep_VEH,]
# MF43_cell <- fibroblast[[]] %>%
#         dplyr::filter(patient == "MF43") %>%
#         rownames()
# gene2keep_MF43 <- rowSums(seurat_data[, colnames(seurat_data) %in% MF43_cell] > 0) > (0.1 * length(MF43_cell))
# MF61_cell <- fibroblast[[]] %>%
#         dplyr::filter(Treatment == "MF61") %>%
#         rownames()
# gene2keep_MF61 <- rowSums(seurat_data[, colnames(seurat_data) %in% MF61_cell] > 0) > (0.1 * length(MF61_cell))
# seurat_data <- seurat_data[gene2keep_MF61|gene2keep_MF43,]
sca <- MAST::FromMatrix(
    exprsArray = as.matrix(seurat_data), 
    cData = fibroblast[[]], 
    fData = data.frame(primerid = rownames(seurat_data)) %>%
            set_rownames(rownames(seurat_data))
)
```

Cell detection rate was calculated for each gene and added to the `sca` object..

```{r}
cdr <- colSums(SummarizedExperiment::assay(sca)>0)
SummarizedExperiment::colData(sca)$ngeneson <- scale(cdr)
```

The reference level of treatment was set to VEH. 

```{r}
SummarizedExperiment::colData(sca)$Treatment <- factor(SummarizedExperiment::colData(sca)$Treatment, levels = c("VEH", "DHT"))
SummarizedExperiment::colData(sca)$patient <- factor(SummarizedExperiment::colData(sca)$patient)
SummarizedExperiment::colData(sca)$patient <- droplevels(SummarizedExperiment::colData(sca)$patient)
```

```{r}
# saveRDS(sca, here::here("data/sca.rds"))
```

Hurdle model was fitted with the patient as a random effect to nest the comparisons in patient using linear mixed modelling. (This step was performed on the DRMCRL workstation).

```{r}
zlm_summary_mixed <- readRDS(here::here("output/zlm_summary_mixed.rds"))$datatable
zlm_summary_mixed <- zlm_summary_mixed %>%
        dplyr::filter(contrast=='TreatmentDHT' & component=='H') %>%
        dplyr::select(primerid, `Pr(>Chisq)`) %>%
        left_join(
            # logFC
            zlm_summary_mixed %>%
                dplyr::filter(contrast=='TreatmentDHT' & component=='logFC') %>%
                dplyr::select(primerid, logFC = coef, ci.hi, ci.lo) 
        )  %>%
        mutate(FDR = p.adjust(`Pr(>Chisq)`, "fdr"),
               DE = ifelse(FDR < 0.01, TRUE, FALSE)
               ) %>%
        dplyr::rename(gene_name = primerid)
```

To my surprise, with the linear mixed model, even with a stringent cut-off of FDR < 0.01, There were still `r nrow(dplyr::filter(zlm_summary_mixed, FDR < 0.01))` genes passing the significance threshold and being considered differentially expressed. 

### Knott MAST results

In the study by [Rath et al. (2023)](https://www.cell.com/cell-genomics/pdf/S2666-979X(23)00032-0.pdf), breast tissues from 9 trans-male donors who have been undertaking gender-affirming therapy (masculinizing dose of androgen) were compared against 9 cis-females' breast through snRNA-seq. 

The pre-processed `Seurat` object for the snRNA-seq data in that study was shared with us by the authors. In the original study, differential expression analyses were performed for each detected cell type without accounting for the confounding effect of patients. Therefore, I re-analysed the data using MAST with donor as a random effect as described above. Only genes that had non-zero log-norm counts in at least 10% of CF or TF cells were tested. Results derived from fibroblast cells were loaded. 


```{r}
knott_mast_de <- readRDS(
    "~/20131906_HickeyT_JC_NormalBreast/output/knott_fibro_mast.rds"
) %>%
    mutate(knott = -sign(coef) * log10(`Pr(>Chisq)`))
```

`r nrow(dplyr::filter(knott_mast_de, DE))` out of the `r nrow(knott_mast_de)` tested genes were considered as significantly differentially expressed with a FDR cut-off of 0.05. 

Ranking statistics for genes in both datasets were defined as $$-sign(logFC) \times \log_{10} (p{\text -}value)$$.

```{r}
knott_mast_de <- readRDS(
    "~/20131906_HickeyT_JC_NormalBreast/output/knott_fibro_mast.rds"
) %>%
    mutate(knott = -sign(coef) * log10(`Pr(>Chisq)`))
```

```{r}
fibro_2RS <- zlm_summary_mixed %>%
    mutate(pilot = -sign(logFC) * log10(`Pr(>Chisq)`)) %>%
    dplyr::select(gene_name, pilot) %>%
    left_join(
        knott_mast_de%>%
            dplyr::select(gene_name , knott)
    ) %>%
    drop_na()
fibro_2RS_cor <- cor.test(fibro_2RS$pilot, fibro_2RS$knott)
```

Correlation between the ranking statistics derived from the two independent studies was found to be `r round( fibro_2RS_cor$estimate, digits = 3)` with a p-value of `r round(fibro_2RS_cor$p.value, digit = 3)`.

```{r}
pilotRK <- zlm_summary_mixed %>%
    mutate(pilot = -sign(logFC) * log10(`Pr(>Chisq)`))  %>%
    dplyr::filter(DE) %>%
    pull(pilot)
knottRK <- knott_mast_de  %>%
    dplyr::filter(DE) %>%
    pull(knott)
```

```{r, fig.height=10, fig.width=14, fig.cap="*Correlation between genes’ ranking statistics when (y-axis) comparing trans-male breast fibroblasts against cis-females’ vs (x-axis) comparing DHT-treated fibroblasts against vehicle-treated ones. The red dashed lines highlight the ranking statistic of zero. The signs of ranking statistics reflect genes’ directions of change. Blue dashed lines indicate the significance threshold for each study*"}
cor_plot <- fibro_2RS %>%
    mutate(aveRS = (pilot + knott)/2) %>%
    .[order(abs(.$aveRS), decreasing = TRUE),] %>%
    mutate(RANK = row_number()) %>%
    ggplot(
        aes(pilot , knott)
    ) +
    geom_point() +
  geom_vline(
    xintercept = 0,
    linetype = 2,
    size = 1.5,
    colour = "red"
  ) +
    geom_vline(
    xintercept = c(max(knottRK[knottRK < 0]), min(knottRK[knottRK > 0])),
    linetype = 2,
    size = 1.5,
    colour = "blue"
  ) +
  geom_hline(
    yintercept = 0,
    linetype = 2,
    size = 1.5,
    colour = "red"
  ) +
    geom_hline(
    yintercept = c(max(pilotRK[pilotRK < 0]), min(pilotRK[pilotRK > 0])),
    linetype = 2,
    size = 1.5,
    colour = "blue"
  ) +
    geom_text_repel(
        aes(label = gene_name), 
        data = . %>% 
            dplyr::filter(
                RANK < 50
            ), 
        color = "red", 
        size = 4
    ) +
    labs(
        x = "Ranking Statistic (DHT vs Veh)", 
        y = "Ranking Statistic (trans-male vs cis-female)", 
        # title = paste("Correlation between logFCs between the 2 study:", 
        #                 round(cor(fibro_2R$pilot, fibro_2FC$knott), digits = 2), 
        #                 sep = " ")
    ) 
cor_plot
```


### AR signature

```{r}
AR_sig <- read.csv(here::here("data/AR_sig_down.csv")) %>%
    mutate(AR_sig = "Down") %>%
    rbind(
        read.csv(here::here("data/AR_sig_up.csv")) %>%
            mutate(AR_sig = "Up")
    )
AR_sig_col <- AR_sig %>%
    mutate(
        color = ifelse(AR_sig == "Down",'blue', "red")
    )
```

A list of `r nrow(AR_sig)` AR signatures that were identified in ER-positive breast cancer cell lines and xenograft models were loaded and the following ones were considered differentially expressed in DHT-treated fibroblasts:

```{r}
zlm_summary_mixed %>%
    dplyr::filter(
        FDR < 0.01, 
        gene_name %in% AR_sig$gene_name
    )  %>%
    left_join(
        AR_sig 
    ) %>%
    mutate(
        Align = case_when(
           logFC < 0 & AR_sig == "Down" ~ "\u2714 Yes",
           logFC > 0 & AR_sig == "Up" ~ "\u2714 Yes",
           logFC < 0 & AR_sig == "Up" ~ "\u2716 No",
           logFC > 0 & AR_sig == "Down" ~ "\u2716 No"
        )
    ) %>%
    mutate_if(is.character, as.factor) %>%
    mutate_at(vars(c("Pr(>Chisq)", "FDR")), formatP) %>%
    mutate(logFC = round(logFC, 3)) %>%
    dplyr::select(
        `Gene Name` = gene_name,
        `Signature type` = AR_sig,
        logFC,
        PValue = `Pr(>Chisq)`,
        FDR, Align
    ) %>%
    datatable(
        filter = "top", 
        caption = htmltools::tags$caption(
            htmltools::em(
                "AR signatures that were differentially expressed in DHT-treated fibroblast cells. The align column shows whether directions of changes in this study align with AR signature type")
        ), 
        options = list(scrollY = "450px")
    ) %>%
    formatStyle(
        'logFC',
        color = styleInterval(0, c('blue', "red"))
    ) %>%
    formatStyle(
        'Signature type',
            color = styleEqual(c("Down", "Up"), c('blue', "red"))
    ) %>%
    formatStyle(
        'Align',
            color = styleEqual(c( "\u2714 Yes", "\u2716 No"), c("#008000", "#e00000"))
    )

```


Not all of the AR signatures that were differentially expressed in DHT-treated cells showed directions of changes that aligned with their signature types in ER-pos breast cancer. 

## DE in each cluster

To perform DE in each cluster, the exact same methods described above were followed, except that now one DE analysis was performed for each clsuter. 

```{r sca_ls}
# create SCA object
seurat_data <- GetAssayData(object = fibroblast, slot = 'data')
sca_ls <-  sapply(levels(fibroblast$cluster), function(x){
    cluster_sub <- fibroblast[[]] %>%
        dplyr::filter(cluster == x)
    VEH_cell <- cluster_sub %>%
        dplyr::filter(Treatment == "VEH") %>%
        rownames()
    gene2keep_VEH <- rowSums(seurat_data[, colnames(seurat_data) %in% VEH_cell] > 0) > (0.1 * length(VEH_cell))
    DHT_cell <- cluster_sub %>%
        dplyr::filter(Treatment == "DHT") %>%
        rownames()
    gene2keep_DHT <- rowSums(seurat_data[, colnames(seurat_data) %in% DHT_cell] > 0) > (0.1 * length(DHT_cell))
    count <- seurat_data[gene2keep_DHT|gene2keep_VEH, colnames(seurat_data) %in% rownames(cluster_sub)]
    MAST::FromMatrix(
    exprsArray = as.matrix(count), 
    cData = cluster_sub, 
    fData = data.frame(primerid = rownames(count)) %>%
            set_rownames(rownames(count))
)
}, simplify = FALSE)


```

Cell detection rate was calculated for each gene within each clsuter and added to the `sca` object..

```{r}
sca_ls <-  sapply(levels(fibroblast$cluster), function(x){
    temp <- sca_ls[[x]]
    cdr <- colSums(SummarizedExperiment::assay(temp)>0)
    SummarizedExperiment::colData(temp)$ngeneson <- scale(cdr)
    SummarizedExperiment::colData(temp)$Treatment <- factor(SummarizedExperiment::colData(temp)$Treatment, levels = c("VEH", "DHT"))
    SummarizedExperiment::colData(temp)$patient <- factor(SummarizedExperiment::colData(temp)$patient)
    SummarizedExperiment::colData(temp)$patient <- droplevels(SummarizedExperiment::colData(temp)$patient)
    temp
}, simplify = FALSE)

```


```{r}
# saveRDS(sca_ls, here::here("data/sca_ls.rds"))
```

```{r, echo=FALSE}
# param <- SnowParam(
#     workers = 7,
#     type = "SOCK"
# )
# zlm_list <- bplapply(sca_ls, function(x){
#     suppressMessages(MAST::zlm(~ ngeneson + Treatment + (1 | patient),
#                                x, method='glmer',ebayes = F,
#                                strictConvergence = FALSE))
# }, BPPARAM = param)
# summary_ls <- bplapply(zlm_list, function(x){
#     temp <- suppressMessages(MAST::summary(x,
#                                           doLRT='TreatmentDHT', logFC = TRUE))
#     temp$datatable
# }, BPPARAM = param)
# saveRDS(summary_ls, file = here::here("output/cluster_summary_ls.rds"))
```

Hurdle model was fitted with the patient as a random effect to nest the comparisons in patient using linear mixed modelling within each cluster, leveraging `Biocparrel` for parallisation. 

```{r}
cluster_summary_ls <- readRDS(here::here("output/cluster_summary_ls.rds")) 
cluster_summary_ls <- sapply(names(cluster_summary_ls), function(x){
    cluster_summary_ls[[x]] %>%
        dplyr::filter(contrast=='TreatmentDHT' & component=='H') %>%
        dplyr::select(primerid, `Pr(>Chisq)`) %>%
        left_join(
            # logFC
            cluster_summary_ls[[x]] %>%
                dplyr::filter(contrast=='TreatmentDHT' & component=='logFC') %>%
                dplyr::select(primerid, logFC = coef, ci.hi, ci.lo) 
        )  %>%
        mutate(FDR = p.adjust(`Pr(>Chisq)`, "fdr"),
               DE = ifelse(FDR < 0.01, TRUE, FALSE), 
               cluster = x
               ) %>%
        dplyr::rename(gene_name = primerid)
}, simplify = FALSE)
```


The numbers of DEGs detected in each cluster using a FDR cut-off of 0.01 are:

```{r}
cluster_summary_ls %>%
    lapply(dplyr::filter, DE) %>%
    vapply(nrow, numeric(1)) %>%
    pander()
```

Intersections between DEGs detected in each cluster were visualised using an Upset plot. 32 consistent DEGs were detected. 

```{r, fig.width=12, fig.height=6, fig.cap="*Upset plot showing the intersections between DHT-induced DEGs that were detected in each cluster using MAST. Genes with an FDR \textless 0.01 were considered to be differentially expressed. 32 genes that were consistently differentially expressed in all clusters were identified.*"}
library(ComplexUpset)
cluster_summary_ls %>%
    lapply(dplyr::filter, DE) %>%
    lapply(dplyr::select, cluster, gene_name) %>%
    bind_rows() %>%
    mutate(Present = 1) %>%
    pivot_wider(names_from = cluster, values_from = Present, values_fill = 0) %>%
    column_to_rownames("gene_name") %>%
    set_colnames(paste("Cluster", colnames(.), sep = " ")) %>%
    ComplexUpset::upset(
        intersect = colnames(.), 
        group_by = 'degree', 
        sort_sets = FALSE,
        # set_sizes = FALSE, 
        queries=list(
            upset_query(intersect=colnames(.), color='orange', fill = 'orange')), 
        base_annotations=list(
            'Intersection size'=intersection_size(
                text_mapping=aes(
                    label=ifelse(
                    !!get_size_mode('exclusive_intersection') == 32,
                    !!get_size_mode('exclusive_intersection'), NA), 
                    colour=ifelse(
                    !!get_size_mode('exclusive_intersection') == 32,
                    'on_bar', 'on_background'),
                ), 
                text_colors=c(
                     on_bar = 'orange', on_background="white"
                ), 
                text = list(
                    size = 5
                )
            )
            + ylab('Intersection size')
        ), 
        # min_size = 2, 
        themes=upset_default_themes(
            text = element_text(size = 12), 
            axis.title.x = element_blank())
    )
```

LogFCs of the 32 consistent DEGs in each cluster were visaulised and strong inter-patient heterogeneity was again evident. 

```{r, fig.cap="*LogFC of the 32 consistent DEGs in each cluster. Opposite directions of changes between MF43- and MF61-dominant clusters were observed in all genes.*"}
all_clu <- cluster_summary_ls %>%
    lapply(dplyr::filter, DE) %>%
    lapply(pull, gene_name) %>%
    Reduce(intersect,.)
deg32_df <- cluster_summary_ls %>% 
    bind_rows() %>%
    dplyr::filter(gene_name %in% all_clu) %>%
    dplyr::select(gene_name, logFC, cluster) %>%
    mutate(cluster = paste("Cluster", cluster, sep = " "))
anno_df <- data.frame(
    cluster = unique(deg32_df$cluster)
) %>%
    mutate(
        `Dominant Patient` = ifelse(
        str_detect(cluster, "2|5"), "MF43", "MF61"
    )
    )
deg32_df %>%
    pivot_wider(
        names_from = "cluster", 
        values_from = "logFC"
    ) %>%
    column_to_rownames("gene_name") %>%
    pheatmap(
        annotation_col = anno_df %>%
            column_to_rownames("cluster"), 
        annotation_colors = list(
           `Dominant Patient` = patient_col
        ),
        cutree_cols = 2, 
        cutree_rows = 2,
        legend_breaks = c(seq(-0.4, 0.4, length.out = 5), max(deg32_df$logFC)), 
        legend_labels = c(seq(-0.4, 0.4, length.out = 5), "LogFC\n")
    )
```

The top 10 global DEGs detected by comparing all DHT-treated cells against vehicle-treated cells are inspected in cluster-specific DE analysis results. Differences in those genes' logFCs in MF-43 dominant and MF61-dominant clusters are again observed, which was masked for global DE analysis. These observations demonstrate that performing DE analysis solely at global levels may obscure cluster-specific responses. 
Conducting DE analysis within each cluster, on the other hand, is a more refined approach that helps unveil cluster-specific transcriptional changes. 

```{r, fig.cap="*LogFC of the 10 most significant global DEGs that were detected by comparing all DHT-treated cells against all vehicle-treated cells in cluster-specific DE analyses. Grey cell indicates that the gene is not differentially expressed in a specific cluster*"}
global_top10 <- zlm_summary_mixed %>%
    dplyr::filter(
        FDR < 0.01
    ) %>%
    arrange(FDR) %>%
    .[1:10, ] %>%
    pull(gene_name)
global_top10_df <- cluster_summary_ls %>% 
    bind_rows() %>%
    dplyr::filter(
        gene_name %in% global_top10, 
        DE) %>%
    dplyr::select(gene_name, logFC, cluster) %>%
    mutate(cluster = paste("Cluster", cluster)) %>%
    pivot_wider(
        names_from = "cluster", 
        values_from = "logFC"
    ) %>%
    column_to_rownames("gene_name") 
global_top10_df %>%
    pheatmap(
        annotation_col = anno_df %>%
            column_to_rownames("cluster"), 
        annotation_colors = list(
           `Dominant Patient` = patient_col
        ),
        cutree_cols = 2, 
        legend_breaks = c(seq(-0.5, 0.5, length.out = 5), max(global_top10_df)), 
        legend_labels = c(seq(-0.5, 0.5, length.out = 5), "LogFC\n"), 
        # display_numbers = sig_anno
    )
```

## Pathway Analysis

### Using `GOseq`

Enrichment of KEGG pathways among cluster-specific DEGs was tested using `GOseq`.

```{r}
gsTopology <- retrieve_topology(
    database = "kegg", species = "hsapiens")
# names(gsTopology) <- str_remove_all(names(gsTopology), "kegg.")
source(here::here("analysis/make_gsNetwork.R"))
kg <- get_GSgenelist(gsTopology)
genesGR <- genesGR %>%
    as.data.frame() %>%
    unnest(entrezid)
kg <- kg %>%
    mutate(gene_id = str_remove_all(gene_id, "ENTREZID:")) %>%
    dplyr::rename(entrezid = gene_id) %>%
    left_join(genesGR %>%
                  dplyr::select(entrezid, gene_name, gene_id) %>%
                  mutate(entrezid = as.character(entrezid)))
gsTokeep <- readRDS(here::here("~/20131906_HickeyT_JC_NormalBreast/output/permutedscore_normal.rds")) %>%
    names()
kg <- kg %>%
    dplyr::filter(gs_name %in% gsTokeep)
kgByGene <- kg  %>%
  split(f = .$gene_id) %>%
  lapply(extract2, "gs_name")
kgByGs <- kg  %>%
  split(f = .$gs_name) %>%
  lapply(extract2, "entrezid")
load(system.file("extdata", "entrez2name.rda", package = "sSNAPPY"))
```

`r length(kgByGs)` KEGG pathways are tested for enrichment. 

Enrichment of KEGG pathways among DEGs found with MAST in each cluster was tested with `goseq`, using gene length as an offset. 

```{r lenPwf_ls}
de_ls <- sapply(names(cluster_summary_ls), function(x){
    genesGR %>%
        dplyr::filter(gene_name %in% (cluster_summary_ls[[x]] %>%
                                          dplyr::filter(DE) %>%
                                          pull(gene_name))) %>%
        pull(gene_id)
}, simplify = FALSE)
lenPwf_ls <- sapply(names(cluster_summary_ls), function(x){
    lenPwf <-  transGR %>%
        as.data.frame() %>%
        distinct(gene_id, .keep_all = TRUE) %>%
        dplyr::select(gene_id, tx_len) %>%
        mutate(Status = case_when(gene_id %in% de_ls[[x]] ~ 1, !gene_id %in% de_ls[[x]]   ~ 0)) %>%
        with(
            nullp(
                DEgenes = structure(
                    Status, names = gene_id
                ),
                genome = "GRCh38.p16",
                id = "ensGene",
                bias.data = tx_len,
                plot.fit = FALSE
            )
        )
}, simplify = FALSE)
```

```{r}
goseq_kg_ls <-  sapply(names(cluster_summary_ls), function(x){
    goseq(lenPwf_ls[[x]],
          gene2cat = kgByGene) %>%
        as_tibble %>%
        dplyr::filter(numDEInCat > 0) %>%
        mutate(
            adjP = p.adjust(over_represented_pvalue, method = "bonf"),
            FDR = as.numeric(p.adjust(over_represented_pvalue, method = "fdr"))
        ) %>%
        dplyr::select(-contains("under")) %>%
        dplyr::rename(
            gs_name = category,
            PValue = over_represented_pvalue,
            nDE = numDEInCat,
            nExpressed = numInCat
        ) %>%
        left_join(kg) %>%
        dplyr::select(
            gs_name, nExpressed, nDE, 
            contains("P", ignore.case = FALSE),
            FDR,
            gene_name, gene_id
        ) %>%
        dplyr::filter(
            gene_id %in% de_ls[[x]]
        ) %>%
        chop(c("gene_name", "gene_id")) %>%
        mutate(
            gene_name = vapply(.$gene_name, function(x){
                paste(x,collapse = "; ")
            }, character(1)),
            gene_id = vapply(.$gene_id, function(x){
                paste(x,collapse = "; ")
            }, character(1)), 
            cluster = x
        ) %>%
        dplyr::rename(DE = gene_name) %>%
        mutate_at(
            vars(one_of(c("PValue", "adjP", "FDR"))),
            formatP
        )
}, simplify = FALSE)
```

Using an FDR cut-off of 0.05, the number of pathways that were significantly enriched in each pathway and their intersections are displaced in the upset plot below: 
```{r, fig.cap="*Upset plot of KEGG pathways significantly over-represented among DEGs detected in each cluster.*"}
goseq_kg_ls %>%
    lapply(dplyr::filter, FDR < 0.05) %>%
    lapply(pull, gs_name) %>%
    fromList() %>%
    set_colnames(paste("Cluster", colnames(.), sep = " ")) %>%
    UpSetR::upset(
        sets = colnames(.), 
        nintersects = NA, 
        keep.order = TRUE,
        queries = list(
          list(
             query = intersects, 
             params = list(setdiff(colnames(.), "Cluster 6")),
             color = "orange",
             active = T),
          list(
             query = intersects, 
             params = list("Cluster 1"),
             color = 'forestgreen',
             active = T), 
          list(
             query = intersects, 
             params = list(setdiff(colnames(.), c("Cluster 2", "Cluster 6"))),
             color = '#9565CD',
             active = T)
        ), 
        text.scale = 2
    )
```

```{r}
all_but6 <- goseq_kg_ls[c("0","2", "5", "1", "3", "4")] %>%
    lapply(dplyr::filter, FDR < 0.05) %>%
    lapply(pull, gs_name) %>%
    Reduce(intersect, .) 
both_pat_gs <-  goseq_kg_ls[c("0", "5", "1", "3", "4")] %>%
    lapply(dplyr::filter, FDR < 0.05) %>%
    lapply(pull, gs_name) %>%
    Reduce(intersect, .) %>%
    setdiff(., all_but6)
```

```{r}
clu1_specific_gs <- setdiff(
    goseq_kg_ls$`1` %>%
        dplyr::filter(FDR < 0.05) %>%
        pull(gs_name), 
    goseq_kg_ls[setdiff(names(goseq_kg_ls), "1")] %>%
        lapply(dplyr::filter, FDR < 0.05) %>%
        lapply(pull, gs_name) %>%
        Reduce(union,.)
)
```

The pathway that was significantly enriched in all clusters but cluster 6 (highlighted in orange) is `r all_but6`, and the gene-set that was enriched in 5 clusters (hightlighted in purple), both MF43-dominant and MF61-dominant ones, was `r both_pat_gs`. 

There were also 8 pathways that were uniquely enriched in cluster 1, which was predicted to be more stem-like than cells in other clusters. The 8 cluster1-specific pathways were `r str_remove_all(clu1_specific_gs, "kegg.")`, among which the WNT pathway was identified. 

The WNT Signalling Pathway is known to be a critical mediator of cell stemness. Enrichment of this pathway exclusively in cluster 1 suggests that AR activation might have an impact on the stemness of fibroblasts assigned to cluster 1. 

### Using `AUCell`

To further explore whether cells in cluster 1 represent a more stem-like state of fibroblasts and if AR activation can change the stemness of cells, we performed a pathway scoring analysis using `AUCell`, which is a method specifically designed for sc/snRNA-seq data that allows scoring of pathway activity at the individual cell level. 

A list of 164 human stem cell signature was retrieved from the study by Barata et al. 

Using `AUCell`, genes are ranked based on their expression within each cell, and one stemness score was computed for each cell based on the ranking of the stemness genes. 

```{r}
stem_gs <- list(
    "stemness" = read_csv(
        here::here("data/stem_sig.txt"), col_names = "gene_name") %>%
        pull(gene_name)
)
```

```{r}
seurat_count <- GetAssayData(object = fibroblast, slot = 'counts')
AUC_stemness <- AUCell_run(
    geneSets = stem_gs, 
    exprMat = seurat_count, 
    aucMaxRank = ceiling(0.2 * nrow(seurat_count)),
    BPPARAM=BiocParallel::MulticoreParam(8)
)
AUC_matrix_stemness <- getAUC(AUC_stemness)
```

Distributions of the computed AUC stemness scores in cells of different clusters and treatments were visualized. While the untreated cells in cluster 5 had the highest median stemness scores, the median stemness score of the vehicle-treated cells in cluster 1 was higher than all the remaining clusters. The DHT-treated cells in cluster 1 also exhibited the highest median stemness score when compared to the DHT-treated cells in other clusters. 

```{r, fig.cap="*Cell stemness scores that were computed by scoring individual cells against a list of 164 human stem cell signature genes using the AUCell method. Cells are grouped by their treatment and cluster allocation.*"}
AUC_matrix_stemness %>%
        t() %>%
    as.data.frame() %>%
    rownames_to_column("cell") %>%
    left_join(
        fibroblast[[]] %>%
            rownames_to_column("cell")
    ) %>%
    mutate(
        Treatment = factor(Treatment, levels = c("VEH", "DHT")), 
        cluster = paste("Cluster", cluster, sep = " ")
    ) %>%
    ggplot(
        aes(cluster, stemness, fill = Treatment)
    ) +
    geom_boxplot() +
    scale_fill_manual(values = treat_col) +
    labs(
        x = "", 
        y = "AUC Stemness Score"
    ) +
    theme(axis.text.x = element_text(
        colour = c("black", "orange", rep("black", 5))
    ))
```

```{r}
# ECM_gs <- msigdbr("Homo sapiens", category = "C2", subcategory = "CP") %>%
#     dplyr::filter(
#         str_detect(gs_name, "MATRISOME|ECM")
#     ) %>%
#   dplyr::rename(gene_id = ensembl_gene) %>%
#   dplyr::filter(!is.na(gene_id)) %>%
#   distinct(gs_name, gene_id, .keep_all = TRUE)
# ecmByGene <- ECM_gs  %>%
#   split(f = .$gene_id) %>%
#   lapply(extract2, "gs_name")
# ecmByID <- ECM_gs  %>%
#   split(f = .$gs_name) %>%
#   lapply(extract2, "gene_symbol")
```


# Thesis figure

Code for reproducing figures used in the theis:

```{r}
library(writexl)
# save global DEGs to excel spreadsheet
zlm_summary_mixed %>%
    dplyr::filter(
        FDR < 0.01
    ) %>%
    dplyr::select(gene_name, Pvalue = `Pr(>Chisq)`,
           logFC, FDR) %>%
    write_xlsx(here::here("output/DE_global.xlsx"))
```

```{r}
# Table 5.3
zlm_summary_mixed %>%
    dplyr::filter(
        FDR < 0.01,
        gene_name %in% AR_sig$gene_name
    )  %>%
    left_join(
        AR_sig
    ) %>%
    mutate(
        Align = case_when(
            logFC < 0 & AR_sig == "Down" ~ "✓",
            logFC > 0 & AR_sig == "Up" ~ "✓",
            logFC < 0 & AR_sig == "Up" ~ "✗",
            logFC > 0 & AR_sig == "Down" ~ "✗"
        ),
        AR_sig = ifelse(
            AR_sig == "Down", "\\color{blue}Down", "\\color{red}Up"
        ),
        gene_name = paste("\\textit{", gene_name, "}", sep = "")
    ) %>%
    mutate_if(is.character, as.factor) %>%
    .[order(.$FDR),] %>%
    mutate_at(vars(c("Pr(>Chisq)", "FDR")), formatP) %>%
    mutate(logFC = ifelse(
        abs(logFC) < 0.01, sprintf("%.2e", logFC), sprintf("%.2f", logFC))) %>%
    # mutate(logFC = as.character(round(logFC, 4))) %>%
    dplyr::select(
        `Gene Name` = gene_name,
        `Expected Direction` = AR_sig,
        logFC,FDR, Concordant = Align
    ) %>%
    xtable::xtable(
        caption = "AR signatures defined in ER-positive BC\\cite{hickey_androgen_2021} that were also differentially expressed upon DHT treatment in mammary fibroblasts. The concordant column reflects whether the signs of logFCs in fibroblasts agreed with the signature types in malignant epithelial cells (ie. expected direction).",
        label = "chap5_table3"
    )  %>%
     xtable::print.xtable(
        "~/PhD_thesis/draft_figure/chapter_05/table3.tex", type = "latex",
        caption.placement = "top",
        include.rownames = FALSE,
        sanitize.text.function = function(x) x)
```

```{r}
chap5_fig7 <- cor_plot &
    theme(
        text = element_text(size = 50)
    )
# png(
#     "/Users/wenjunliu/PhD_thesis/Images/Chapter_05/chap5_fig7.png",
#     height = 1600,
#     width = 2000
# )
#chap5_fig7
# dev.off()
```

```{r}
# Table of num of DEGs in each cluster
cluster_summary_ls %>%
    lapply(dplyr::filter, DE) %>%
    vapply(nrow, numeric(1)) %>%
    enframe(
        name = "Cluster",
        value = "Number of DEGs"
    ) %>%
    left_join(
        cluster_summary_ls %>%
            vapply(nrow, numeric(1)) %>%
            enframe(
                name = "Cluster",
                value = "Number of tested genes"
            )
    ) %>%
    xtable::xtable(
        caption = "The numbers of tested genes and detected DEGs in each cluster. DE analyses were performed using MAST with patient as a random effect. Only genes that were expressed in at least 10% of either DHT- or vehicle-treated cells in each cluster were testde for differential expression. Genes with an FDR smaller than 0.01 were considered to be DEGs",
        label = "chap5_tableDE",
        digits = 0
    )  %>%
    xtable::print.xtable(
        "~/PhD_thesis/draft_figure/chapter_05/clsuter_DE.tex", type = "latex",
        caption.placement = "top",
        include.rownames = FALSE,
        sanitize.text.function = function(x) x)
```

```{r}
#DE upset plot
DE_upset <- cluster_summary_ls %>%
    lapply(dplyr::filter, DE) %>%
    lapply(dplyr::select, cluster, gene_name) %>%
    bind_rows() %>%
    mutate(Present = 1) %>%
    pivot_wider(names_from = cluster, values_from = Present, values_fill = 0) %>%
    column_to_rownames("gene_name") %>%
    set_colnames(paste("Cluster", colnames(.), sep = " ")) %>%
    ComplexUpset::upset(
        intersect = colnames(.),
        group_by = 'degree',
        sort_sets = FALSE,
        set_sizes = FALSE,
        queries=list(
            upset_query(intersect=colnames(.), color='orange', fill = 'orange')),
        base_annotations=list(
            'Intersection size'=intersection_size(
                text_mapping=aes(
                    label=ifelse(
                    !!get_size_mode('exclusive_intersection') == 32,
                    !!get_size_mode('exclusive_intersection'), NA),
                    colour=ifelse(
                    !!get_size_mode('exclusive_intersection') == 32,
                    'on_bar', 'on_background'),
                ),
                text_colors=c(
                     on_bar = 'orange', on_background="white"
                ),
                text = list(
                    size = 5
                )
            )
            + ylab('Intersection size')
        ),
        # min_size = 2,
        themes=upset_default_themes(
            text = element_text(size = 12),
            axis.title.x = element_blank())
    )
# png(
#     "/Users/wenjunliu/PhD_thesis/Images/chapter_05/DE_upset.png",
#     width = 210, height =130, units='mm', res = 300
# )
# DE_upset
# dev.off()
```

```{r}
# png(
#     "/Users/wenjunliu/PhD_thesis/Images/chapter_05/hp_32DEG.png",
#     width = 190, height =140, units='mm', res = 300
# )
# deg32_df %>%
#     pivot_wider(
#         names_from = "cluster",
#         values_from = "logFC"
#     ) %>%
#     column_to_rownames("gene_name") %>%
#     pheatmap(
#         annotation_col = anno_df %>%
#             column_to_rownames("cluster"),
#         annotation_colors = list(
#            `Dominant Patient` = patient_col
#         ),
#         cutree_cols = 2,
#         cutree_rows = 2,
#         legend_breaks = c(seq(-0.4, 0.4, length.out = 5), max(deg32_df$logFC)),
#         legend_labels = c(seq(-0.4, 0.4, length.out = 5), "LogFC\n")
#     )
# dev.off()
```

```{r}
# png(
#     "/Users/wenjunliu/PhD_thesis/Images/chapter_05/hp_global_top10.png",
#     width = 190, height =140, units='mm', res = 300
# )
# global_top10_df %>%
#     pheatmap(
#         annotation_col = anno_df %>%
#             column_to_rownames("cluster"),
#         annotation_colors = list(
#            `Dominant Patient` = patient_col
#         ),
#         cutree_cols = 2,
#         legend_breaks = c(seq(-0.5, 0.5, length.out = 5), max(global_top10_df)),
#         legend_labels = c(seq(-0.5, 0.5, length.out = 5), "LogFC\n")
#     )
# dev.off()
```

```{r}
#save cluster-specific DE results to spreadsheet
library(writexl)
DE_suppTable <- cluster_summary_ls %>%
    lapply(dplyr::filter, DE) %>%
    lapply(dplyr::select, gene_name, Pvalue = `Pr(>Chisq)`,
           logFC, FDR,  cluster)
names(DE_suppTable) <- paste("Cluster", names(DE_suppTable), sep = "_")
# write_xlsx(DE_suppTable, here::here("output/DE_suppTable.xlsx"))
```

```{r}
goseq_kg_ls %>%
    lapply(dplyr::filter, FDR < 0.05) %>%
    vapply(nrow, numeric(1)) %>%
    enframe(
        name = "Cluster",
        value = "Number of significantly\nenriched pathways"
    ) %>%
    xtable::xtable(
        caption = "The numbers of KEGG pathways that were significantly over-repsented in each cluster. The enrichment analysis was performed using GOseq\\cite{young_mgeethnode_2010}, with gene length as an offset term to account for any bias. Pathways with an FDR \\textless 0.05 were considered to be significantly enriched.",
        label = "path_numb",
        digits = 0
    )  %>%
    xtable::print.xtable(
        "~/PhD_thesis/draft_figure/chapter_05/path_num.tex", type = "latex",
        caption.placement = "top",
        include.rownames = FALSE,
        sanitize.text.function = function(x) x)
```

```{r}
## Supp Table: significantly enriched pathways in each cluster
path_suppTable <- goseq_kg_ls %>%
    lapply(dplyr::filter, FDR < 0.05) %>%
    lapply(function(x){mutate(x, gs_name = str_remove_all(x$gs_name, "kegg."))}) %>%
    lapply(dplyr::select, gs_name, nExpressed, nDE, FDR)
names(path_suppTable) <- paste("Cluster", names(path_suppTable), sep = "_")
# write_xlsx(path_suppTable, here::here("output/path_suppTable.xlsx"))
```

```{r}
# png(
#     "/Users/wenjunliu/PhD_thesis/Images/chapter_05/pathway_upset.png",
#     width = 230, height =150, units='mm', res = 300
# )
# goseq_kg_ls %>%
#     lapply(dplyr::filter, FDR < 0.05) %>%
#     lapply(pull, gs_name) %>%
#     fromList() %>%
#     set_colnames(paste("Cluster", colnames(.), sep = " ")) %>%
#     UpSetR::upset(
#         sets = colnames(.),
#         nintersects = NA,
#         keep.order = TRUE,
#         queries = list(
#           list(
#              query = intersects,
#              params = list(setdiff(colnames(.), "Cluster 6")),
#              color = "orange",
#              active = T),
#           list(
#              query = intersects,
#              params = list("Cluster 1"),
#              color = 'forestgreen',
#              active = T),
#           list(
#              query = intersects,
#              params = list(setdiff(colnames(.), c("Cluster 2", "Cluster 6"))),
#              color = '#9565CD',
#              active = T)
#         ),
#         text.scale = 2
#     )
# dev.off()
```

```{r}
goseq_kg_ls[["1"]] %>%
    dplyr::filter(gs_name %in% clu1_specific_gs) %>%
    mutate(
        gs_name = str_remove_all(gs_name, "kegg.")
    ) %>%
    dplyr::select(
        `Gene-set name` = gs_name,
        nDE,  nExpressed, FDR
    ) %>%
    xtable::xtable(
        caption = "KEGG pathways that were uniquely over-represented due to the DHT treatment in cluster 1",
        label = "clu1_unique",
        digits = 0
    )  %>%
    xtable::print.xtable(
        "~/PhD_thesis/draft_figure/chapter_05/clu1_unique.tex", type = "latex",
        caption.placement = "top",
        include.rownames = FALSE,
        sanitize.text.function = function(x) x)
```

```{r}
stem_boxplot <- AUC_matrix_stemness %>%
        t() %>%
    as.data.frame() %>%
    rownames_to_column("cell") %>%
    left_join(
        fibroblast[[]] %>%
            rownames_to_column("cell")
    ) %>%
    mutate(
        Treatment = factor(Treatment, levels = c("VEH", "DHT"))
    ) %>%
    ggplot(
        aes(cluster, stemness, fill = Treatment)
    ) +
    geom_boxplot() +
    scale_fill_manual(values = treat_col) +
    labs(
        x = "Cluster",
        y = "Cell Stemness Score"
    ) +
    theme(
        text = element_text(size = 12),
        axis.text.x = element_text(
        colour = c("black","orange", rep("black", 5)),
        size = c(12, 20, rep(12, 5))
    ))
# png(
#     "/Users/wenjunliu/PhD_thesis/Images/chapter_05/stem_boxplot.png",
#     width = 180, height =100, units='mm', res = 300
# )
# stem_boxplot
# dev.off()
```

